<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • liana</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_intracell.html">LIANA with Intracellular Omnipath</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/saezlab/liana/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org" class="external-link">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/saezlab/liana/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.7" id="liana-017-131022">LIANA 0.1.7 (13.10.22)<a class="anchor" aria-label="anchor" href="#liana-017-131022"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-7">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-7"></a></h3>
<ul><li><p>Changed the way ties are handles in liana_aggregate. Namely, I previously assigned the minimum rank, but this resulted in ties getting lower p-values than they should, particularly for scores with a lot of ties (e.g. CPDB p-value).</p></li>
<li><p>Fixed an issue where some subunits of 0 <code>expr_prop</code> would not get filtered. This was observed due to previous changes to <code>.filt_liana_pipe</code> in 0.1.6, where some subunits were filtered before <code>recomplexifying</code>.</p></li>
<li><p>Fixed an issue in which some NATMI complexes would be missing due to <code>recomplexify</code> being done on both .expr and .sum columns. These are now seperated into <code>columns</code>, which are the ones for which I account for complexes, and <code>add_cols</code>, the ones that are additional - no need to account for complexes (e.g. also <code>global_mean</code>).</p></li>
</ul></div>
<div class="section level3">
<h3 id="minor-changes-0-1-7">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-7"></a></h3>
<ul><li>I now refer to <a href="https://enblacar.github.io/SCpubr-book/index.html" class="external-link">SCPubr</a> in liana’s tutorial.</li></ul></div>
<div class="section level3">
<h3 id="minor-changes-0-1-7-1">Minor changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-7-1"></a></h3>
<ul><li>I now refer to SCpubr in the tutorial for more advanced plots.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.6" id="liana-016-110822">LIANA 0.1.6 (11.08.22)<a class="anchor" aria-label="anchor" href="#liana-016-110822"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-6">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-6"></a></h3>
<ul><li><p>Fixed an issue where interactions with complexes will not filtered be according to <code>expr_prop</code> for some methods. I now filter twice - once via <code>.filt_liana_pipe</code> for computational speed, and once after <code>recomplexify</code> to also remove the complexes with <code>expr_prop</code> &lt;= X. Will now also filter <code>crosstalk_scores</code> to <code>expr_prop</code>.</p></li>
<li><p>In the edgecase of complexes with subunits with equal expression, LIANA’s internal methods will not arbitrarily discard duplicate complex interactions.</p></li>
<li><p>Will now return <code>expr_prop</code> for each method. Note that this information is discarded by <code>liana_aggregate</code>.</p></li>
<li><p><code>liana_doplot</code> function is now more explicit in the way interactions are selected. Will now take <code>topn</code> and return the highest ranked interactions. Size of dots is also more distinguishable by default and the user can now pass a customizable value for the size range.</p></li>
<li><p>Added a <code>rank_method</code> helper function to rank single methods according to <code>specificity</code> and magnitude.</p></li>
<li><p>Removed ~20 bad quality interactions from the <code>Consensus</code> resource.</p></li>
<li><p>Minor changes on filtering SCE object in <code>liana_pipe</code> to ensure all complex subunits are present in the sce</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.5" id="liana-015-040722">LIANA 0.1.5 (04.07.22)<a class="anchor" aria-label="anchor" href="#liana-015-040722"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-5">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-5"></a></h3>
<ul><li><p>Re-implemented the <code>RRA</code> method from Kolde et al., 2012, as a consequence of the removal of the <code>RobustRankAggregate</code> package from CRAN.</p></li>
<li><p>Integrate <code>generate_homologs</code> with OmniPath’s <code>homologene</code> database. This allows homology conversion by simply passing an organism ID. Also, handles complicated cases, such as complex subunits with one-to-many mapping homologs.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.4" id="liana-014-180622">LIANA 0.1.4 (18.06.22)<a class="anchor" aria-label="anchor" href="#liana-014-180622"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-4">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-4"></a></h3>
<ul><li><p>Add <code>prod_weight</code> to NATMI’s score. This is the weight that both Connectome and NATMI suggest for between-condition comparisons. Add NATMI to the housekeeping aggregate ranking.</p></li>
<li><p>Enable weighing of interactions by cell pairs (using a DF in which each cell pair has an assigned weight). This would typically be done by spatial constraints, etc. These weights can also be used to mask any cell-pair interactions which are not relevant (by assigning weights of 0). This currently assumes that the weights would be between 0 to 1 - to be extended. Tutorial on this /w appropriate spatial weight generation to be written.</p></li>
</ul></div>
<div class="section level3">
<h3 id="minor-changes-0-1-4">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-4"></a></h3>
<ul><li>By default, the base for logFC will now be automatically assigned depending on the object passed to LIANA, i.e. <code>.antilog1m</code> for SCE will use 2 as base, and Euler’s number for Seurat. One could also pass the base they wish to use via <code>liana_wrap</code>.</li>
<li>Automate website deployment to gh-pages and run R checks on push.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.3" id="liana-013-150522">LIANA 0.1.3 (15.05.22)<a class="anchor" aria-label="anchor" href="#liana-013-150522"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-3">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-3"></a></h3>
<ul><li><p>Changed the aggregation columns of <code>liana_aggregate</code>, as in some cases methods would assign different subunits as the minimum, which results in redundancies for the same complex. As such, <code>liana_aggregate</code> will now return only the complex columns, nevertheless, the methods will still return both the minimum (lowest expressed subunit) and it’s corresponding complex.</p></li>
<li><p><code>base</code> used to calculate logFC (via <code>get_log2FC</code>) can now be passed as a parameter to <code>liana_wrap</code> via <code>liana_pipe.params</code>. Passing <code>NaN</code> to base would result in log2FC calculation using the raw counts without any pre-processing (e.g. no batch correction, etc).</p></li>
</ul><p>The base is by default set to 2, assuming that log2 transformation is performed following library size normalization, and thus preserving the normalization, while reverting back to ~counts.</p>
</div>
<div class="section level3">
<h3 id="minor-changes-0-1-3">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-3"></a></h3>
<ul><li>Extended the heatmap to allow filtering down to certain cell types.</li>
<li>Removed redundant/leftover code</li>
</ul></div>
<div class="section level3">
<h3 id="new-implementations-0-1-3">New implementations<a class="anchor" aria-label="anchor" href="#new-implementations-0-1-3"></a></h3>
<ul><li>A chord plot for interaction frequncies was included.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.2" id="liana-012-030522">LIANA 0.1.2 (03.05.22)<a class="anchor" aria-label="anchor" href="#liana-012-030522"></a></h2>
<div class="section level3">
<h3 id="new-implementations-0-1-2">New Implementations<a class="anchor" aria-label="anchor" href="#new-implementations-0-1-2"></a></h3>
<ul><li>Frequency Heatmap available via the <code>heat_freq</code> functions, added due to being common requests. This heatmap was inspired by CellPhoneDB and CellChat.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-2">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-2"></a></h3>
<ul><li>Extended basic tutorial to accommodate new heatmap plots.</li></ul></div>
<div class="section level3">
<h3 id="minor-changes-0-1-2">Minor changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-2"></a></h3>
<ul><li>Allow labels to be passed to <code>liana_dotplot</code>
</li>
<li>Cleaned up docs, dependencies, examples, and warnings</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.1" id="liana-011-260422">LIANA 0.1.1 (26.04.22)<a class="anchor" aria-label="anchor" href="#liana-011-260422"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-1">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1"></a></h3>
<ul><li>Change the order of non-expressed genes and empty droplet filtering. I now appropriately filter cells in the <code>sce</code> object <em>after</em> limiting the gene universe to ligands and receptors in the resource.</li></ul></div>
<div class="section level3">
<h3 id="minor-changes-0-1-1">Minor changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-1"></a></h3>
<ul><li><p>Appropriately pass <code>verbose</code> to <code>.filter_sce</code></p></li>
<li><p>Silence expected warning in <code>cellchat_formatDB</code></p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.0" id="liana-010-200422">LIANA 0.1.0 (20.04.22)<a class="anchor" aria-label="anchor" href="#liana-010-200422"></a></h2>
<div class="section level3">
<h3 id="changes-0-1-0">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-0"></a></h3>
<ul><li>For <code>logFC_mean</code>, rather than normalizing the counts by library size, I instead inverse log the counts and use those to calculate log2FC. This is to preserve any prior correction of the counts, i.e. mainly for consistency with the rest of the methods.</li></ul></div>
<div class="section level3">
<h3 id="minor-changes-0-1-0">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-1-0"></a></h3>
<ul><li><p>Flipped <code>x</code> and <code>y</code> axes dotplot according to feedback.</p></li>
<li><p>Added <code>.default_fun</code> parameter to <code>generate_orthologues</code></p></li>
<li><p>Minor code clean up</p></li>
<li><p>Added <code>examples</code> to main exported functions docs</p></li>
<li><p>Removed several low quality interactions from the <code>Consensus</code> resource</p></li>
<li><p>CellChat will now work with the simplified format of intercell (i.e. Consensus resource).</p></li>
<li><p>CellChat and Squidpy should now be called using <code>call_cellchat</code> and <code>call_squidpy</code>, respectively.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.9" id="liana-009-23032022">LIANA 0.0.9 (23.03.2022)<a class="anchor" aria-label="anchor" href="#liana-009-23032022"></a></h2>
<div class="section level3">
<h3 id="new-implementations-0-0-9">New Implementations<a class="anchor" aria-label="anchor" href="#new-implementations-0-0-9"></a></h3>
<ul><li>We now provide a tutorial dedicated to <code>orthology conversion</code> of the resources in LIANA</li></ul></div>
<div class="section level3">
<h3 id="minor-changes-0-0-9">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-0-9"></a></h3>
<p>LIANA will now check if: - There is enough of a gene intesect between the resource and the data (i.e. if they are both for human)</p>
<ul><li><p>There are negative counts</p></li>
<li><p>An assay with normalized counts was been provided (i.e. if the data/logcounts slot is scaled).</p></li>
<li><p>LIANA will now convert non-sparse matrices to sparse.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.8" id="liana-008-05032022">LIANA 0.0.8 (05.03.2022)<a class="anchor" aria-label="anchor" href="#liana-008-05032022"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-8">Changes<a class="anchor" aria-label="anchor" href="#changes-0-0-8"></a></h3>
<ul><li>Reduced dependencies (specifically <code>Seurat</code> and <code>OmniPathR</code>)</li></ul></div>
<div class="section level3">
<h3 id="minor-changes-0-0-8">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-0-8"></a></h3>
<ul><li>testthat tests external methods only if requested explicitly</li>
<li>Readme updated - clarified and accordingly describes the <code>Consensus</code> resource as default</li>
</ul></div>
<div class="section level3">
<h3 id="new-features-0-0-8">New Features<a class="anchor" aria-label="anchor" href="#new-features-0-0-8"></a></h3>
<ul><li>
<code>idents_col</code> is can now be explicitly passed to <code>liana_wrap</code>, if not provided defaults to the active idents/colLabels for SCE and Seurat, respectively.</li>
<li>
<code>verbose</code> param allows to omit any messages and warnings from LIANA</li>
<li>
<code>assay</code> can now be passed explicitly when working with a Seurat object, defaults to the active one otherwise</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.7" id="liana-007">LIANA 0.0.7<a class="anchor" aria-label="anchor" href="#liana-007"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-7">Changes<a class="anchor" aria-label="anchor" href="#changes-0-0-7"></a></h3>
<ul><li><p>LIANA will now use the <code>Consensus</code> resource by default. This is a highly-literature supported resource, generated using similar filtering steps as the ‘OmniPath’ (old default) resource. This resource is similar in size (~4,700 interactions), but contains a higher complex and curation content.</p></li>
<li><p>All resources might show some very minor changes related to an update of UniProt IDs and homology-conversion improvements.</p></li>
<li><p>LIANA now uses <code>mean0</code> to account for heteromeric complexes, i.e. the mean is computed, unless there is a value of 0, then 0 is returned. This means that any complex, the subunit of which is not expressed is filtered. LIANA now also appropriately accepts any custom function to account for complexes.</p></li>
<li><p><code>liana_aggregate</code> now groups by ligand.complex and receptor.complex as well as the subunits, and hence returns a all of those columns</p></li>
</ul></div>
<div class="section level3">
<h3 id="minor-changes-0-0-7">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-0-7"></a></h3>
<ul><li><p>Added option to show complexes on dotplot and is now the default option</p></li>
<li><p>Documentation improvements</p></li>
<li><p><code>decomplexify</code> function is now exported</p></li>
<li><p><code>liana_aggregate</code> will no longer return a median_rank, it’s largely redundant.</p></li>
<li><p>re-arranged the column order of <code>liana_aggregate</code> due to the addition of .complex columns</p></li>
<li><p>Replaced min0 (used to obtain closest to 0 value) to min -&gt; relevant for z-scores used in Connectome.</p></li>
</ul></div>
<div class="section level3">
<h3 id="bugs-0-0-7">Bugs<a class="anchor" aria-label="anchor" href="#bugs-0-0-7"></a></h3>
<ul><li><p>Complexes with missing subunits are not correctly assigned as ‘missing’ and hence filtered/treated as non-expressed.</p></li>
<li><p>Fixed a bug where LIANA will return the minimum subunit expression, instead of the mean for some methods. This stemmed from not properly passing the incorrect <code>complex_policy</code> to certain methods, i.e. they were getting a hard-coded value instead.</p></li>
<li><p>Remove <code>decomplexify</code> logical from <code>liana_call</code> and <code>liana_pipe</code> -&gt; redundant.</p></li>
<li><p>edge case fix: liana_aggregate should now rank interactions with the same subunits, but coming from different complexes seperately</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.6" id="liana-006">LIANA 0.0.6<a class="anchor" aria-label="anchor" href="#liana-006"></a></h2>
<div class="section level3">
<h3 id="new-implementations-0-0-6">New implementations<a class="anchor" aria-label="anchor" href="#new-implementations-0-0-6"></a></h3>
<ul><li><p>LIANA has now been optimized in terms of RAM, by swapping all internal function to rely solely on the BioConductor single-cell framework (for all internal methods).</p></li>
<li><p>LIANA now accepts both <code>SingleCellExperiment</code> and <code>Seurat</code> objects as input.</p></li>
<li><p>added <code>liana_dotplot</code> as a basic, but flexible, dotplot function for LIANA output. (+ tests)</p></li>
</ul></div>
<div class="section level3">
<h3 id="changes-0-0-6">Changes<a class="anchor" aria-label="anchor" href="#changes-0-0-6"></a></h3>
<ul><li><p>LIANA will now perform a basic filtering step, where all genes and cells with 0 summed counts are removed.</p></li>
<li><p><code>global_mean</code> is now calculated in a more efficient manner.</p></li>
<li><p><code>assay.type</code> in <code>liana_pipe</code> was passed to <code>get_logFC</code> would result in using the logcounts, rather than the library-normalized counts.</p></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-0-6">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-6"></a></h3>
<ul><li>Fixed a bug where incorrectly passing method names in different cases results in an error.</li></ul></div>
<div class="section level3">
<h3 id="deprecated-0-0-6">Deprecated<a class="anchor" aria-label="anchor" href="#deprecated-0-0-6"></a></h3>
<ul><li>External LIANA methods (i.e. <code>call_</code>) are now deprecated. The pipelines will be maintained solely for power users, who intend to benchmark the original implementations, but will not be the focus of any downstream analyses. These will be solely developed for the internal (or re-implemented methods). These still rely on a <code>SeuratObject</code> as interface, but will now accept both sce and seurat as input.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.5" id="liana-005">LIANA 0.0.5<a class="anchor" aria-label="anchor" href="#liana-005"></a></h2>
<div class="section level3">
<h3 id="changes-0-0-5">Changes<a class="anchor" aria-label="anchor" href="#changes-0-0-5"></a></h3>
<ul><li><p>I now filter the Crosstalk scores to include only those &gt; 0. Otherwise, LIANA would return all possible combinations of clusters and interactions, which would be simply NAs and 0s for Crosstalk scores. Should do the same for Connectome (&gt;0).</p></li>
<li><p><code>CellChat</code> and Crosstalk scores/<code>cytotalk</code> will no longer by called by default by liana_wrap. However, it both are available as an option to be passed to the <code>method</code> parameter.</p></li>
<li><p>I now filter all methods by <code>expr_prop</code>. This is done in a slightly different manner for Connectome’s scaled weights and crosstalk scores, since they require all pairs/clusters to be present to appropriately calculate their scores. Thus, for them we filter after we calculate the scores, while for the others methods we pre-filter.</p></li>
<li><p>We now provide a tutorial how to make use of <a href="https://saezlab.github.io/liana/articles/liana_intracell.html" class="external-link">intracellular OmniPath</a> as well as how to <a href="https://saezlab.github.io/liana/articles/liana_nichenet.html" class="external-link">combine LIANA with NicheNet</a></p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.4" id="liana-004">LIANA 0.0.4<a class="anchor" aria-label="anchor" href="#liana-004"></a></h2>
<ul><li><p>The OmniPath resource had a major update.</p></li>
<li><p><code>CellCall</code> and <code>Cellinker</code> resources were added, while talklr was removed. The OmniPath resources itself was filtered further and 1,000 lower quality interactions were excluded. Further improvements were made to all resources, most of which were minor. Changes worth mentioning were made to ICELLNET (updated to latest resource version), CellPhoneDB (was filtered for ambigous interactions), and CellChatDB was filtered for mislaballed interactions.</p></li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.3" id="liana-003">LIANA 0.0.3<a class="anchor" aria-label="anchor" href="#liana-003"></a></h2>
<div class="section level3">
<h3 id="improvements-0-0-3">Improvements<a class="anchor" aria-label="anchor" href="#improvements-0-0-3"></a></h3>
<ul><li><p>The R re-implementation of CellPhoneDBv2’s permutation algorithm was optimized to work with sparse matrices (and is now uqicker), and set as the default option in LIANA (replacing the re-implementation of the same algorithm from squidpy)</p></li>
<li><p>Custom proportion filtering - Connectome and CytoTalk are now not filtered by expr_prop as this affects the way that their scores are calculated, since they require all clusters/cluster pairs to be present to appropriately scale or normalize their scores.</p></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-0-3">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-3"></a></h3>
<ul><li>Fixed an issue where logFC was assigned only the value of the ligand</li></ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.2" id="liana-002">LIANA 0.0.2<a class="anchor" aria-label="anchor" href="#liana-002"></a></h2>
<div class="section level3">
<h3 id="new-features-0-0-2">New Features<a class="anchor" aria-label="anchor" href="#new-features-0-0-2"></a></h3>
<ul><li>
<code>Crosstalk scores</code> inspired by <a href="(https://advances.sciencemag.org/content/7/16/eabf1356)" class="external-link">Cytotalk</a> were added. In contrast to the CytoTalk, in our calculation CTS with ligand or receptor with PEM of 0 are assigned 0 CTS. Furthermore, we use the inverse of the non-self-talk scores calculated in CytoTalk to also allow for autocrine signalling interactions, and thus make Crosstalk scores comparable to the rest of the methods in LIANA. Finally, as part of LIANA, CytoTalk’s re-implemented scores would not take account of complexes and we also apply liana-specifc filtering such as according to <code>expr_prop</code>. Worth noting, we only re-implement the cross-talk scores, but we don’t include the intracellular part of Cytotalk.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-0-2">Changes<a class="anchor" aria-label="anchor" href="#changes-0-0-2"></a></h3>
<ul><li>Changed <code>expr_thresh</code> to 0.1, based on lack of improvement in performance when using 0.2, hence opted out for the less conservative threshold as default<br></li>
<li>Changed the way that default parameters are passed to each method<br></li>
<li>Enabled housekeeping score aggregation for external methods (needed for revisions) via <code>.score_housekeep</code>
</li>
<li>Fixed Bug where external methods could not be called with their default DB. The resource is now always decomplexified</li>
<li>Seurat Testdata is now properly normalized</li>
<li>liana_aggragate will now by defaul dissociate complexe for CellChat Complexes</li>
<li>Added tests for changes</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.1" id="liana-001">LIANA 0.0.1<a class="anchor" aria-label="anchor" href="#liana-001"></a></h2>
<div class="section level3">
<h3 id="new-features-0-0-1">New Features<a class="anchor" aria-label="anchor" href="#new-features-0-0-1"></a></h3>
<p><code>liana_wrap</code> and <code>liana_aggragate</code> as the two highest level functions to run all the methods in liana and aggragate them, respectively.</p>
<div class="section level4">
<h4 id="re-implemented-the-following-scores-in-liana-0-0-1">Re-implemented the following scores in LIANA:<a class="anchor" aria-label="anchor" href="#re-implemented-the-following-scores-in-liana-0-0-1"></a></h4>
<ul><li>logFC<br></li>
<li>NATMI specificity edges<br></li>
<li>Connectome scaled_weights<br></li>
<li>CellPhoneDB algorithm<br></li>
<li>SingleCellSignalR LRScore</li>
</ul><p>each called via <code>liana_call</code>, which leverages the statistics provided by <code>liana_pipe</code>,</p>
</div>
<div class="section level4">
<h4 id="others-0-0-1">Others<a class="anchor" aria-label="anchor" href="#others-0-0-1"></a></h4>
<ul><li><p>Not re-implemented method score names now start with <code>call_*</code></p></li>
<li><p><code>decomplexify</code> and <code>recomplexify</code> as functions used to dissociate complexes in resources and account for complexes of the re-implemented methods above</p></li>
<li><p><code>liana_aggragate</code> - a handy wrapper to aggregate results</p></li>
<li><p><code>LIANA</code> and <code>LIANA++</code> are now the user-friendly and benchmark version of LIANA, respectively</p></li>
<li><p>A webpage with vignettes showing the validity of the re-implemented methods, a developer/benchmark-focused vignette, and a vignette to customize OmniPath</p></li>
</ul></div>
</div>
<div class="section level3">
<h3 id="bug-fixes-0-0-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-1"></a></h3>
<p>A number of fixes were implemented thanks to the early stage users. Thanks you.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

