<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run iTALK with OmniPath data — call_italk • liana</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run iTALK with OmniPath data — call_italk" />
<meta property="og:description" content="Run iTALK with OmniPath data" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/liana_custom_op.html">LIANA Customize CCC OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_remethod.html">LIANA Re-implemented Methods</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run iTALK with OmniPath data</h1>
    <small class="dont-index">Source: <a href='https://github.com/saezlab/liana/blob/master/R/italk_pipe.R'><code>R/italk_pipe.R</code></a></small>
    <div class="hidden name"><code>call_italk.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run iTALK with OmniPath data</p>
    </div>

    <pre class="usage"><span class='fu'>call_italk</span><span class='op'>(</span>
  <span class='va'>op_resource</span>,
  <span class='va'>seurat_object</span>,
  assay <span class='op'>=</span> <span class='st'>"RNA"</span>,
  .format <span class='op'>=</span> <span class='cn'>TRUE</span>,
  .DE <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>op_resource</th>
      <td><p>OmniPath Intercell Resource DN</p></td>
    </tr>
    <tr>
      <th>seurat_object</th>
      <td><p>Seurat object as input</p></td>
    </tr>
    <tr>
      <th>assay</th>
      <td><p>assay to use from Seurat object</p></td>
    </tr>
    <tr>
      <th>.format</th>
      <td><p>bool: whether to format output</p></td>
    </tr>
    <tr>
      <th>.DE</th>
      <td><p>bool: whether to use DE (TRUE) or highlyVarGenes (FALSE)</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed on to <code><a href='https://satijalab.org/seurat/reference/FindAllMarkers.html'>Seurat::FindAllMarkers</a></code></p><dl>
    <dt><code>object</code></dt><dd><p>An object</p></dd>
    <dt><code>features</code></dt><dd><p>Genes to test. Default is to use all genes</p></dd>
    <dt><code>logfc.threshold</code></dt><dd><p>Limit testing to genes which show, on average, at least
X-fold difference (log-scale) between the two groups of cells. Default is 0.25
Increasing logfc.threshold speeds up the function, but can miss weaker signals.</p></dd>
    <dt><code>test.use</code></dt><dd><p>Denotes which test to use. Available options are:
<ul>
<li><p>"wilcox" : Identifies differentially expressed genes between two
 groups of cells using a Wilcoxon Rank Sum test (default)</p></li>
<li><p>"bimod" : Likelihood-ratio test for single cell gene expression,
 (McDavid et al., Bioinformatics, 2013)</p></li>
<li><p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
 For each gene, evaluates (using AUC) a classifier built on that gene alone,
 to classify between two groups of cells. An AUC value of 1 means that
 expression values for this gene alone can perfectly classify the two
 groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
 each of the cells in cells.2). An AUC value of 0 also means there is perfect
 classification, but in the other direction. A value of 0.5 implies that
 the gene has no predictive power to classify the two groups. Returns a
 'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
 expressed genes.</p></li>
<li><p>"t" : Identify differentially expressed genes between two groups of
 cells using the Student's t-test.</p></li>
<li><p>"negbinom" : Identifies differentially expressed genes between two
  groups of cells using a negative binomial generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"poisson" : Identifies differentially expressed genes between two
  groups of cells using a poisson generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"LR" : Uses a logistic regression framework to determine differentially
 expressed genes. Constructs a logistic regression model predicting group
 membership based on each feature individually and compares this to a null
 model with a likelihood ratio test.</p></li>
<li><p>"MAST" : Identifies differentially expressed genes between two groups
 of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
 package to run the DE testing.</p></li>
<li><p>"DESeq2" : Identifies differentially expressed genes between two groups
 of cells based on a model using DESeq2 which uses a negative binomial
 distribution (Love et al, Genome Biology, 2014).This test does not support
 pre-filtering of genes based on average difference (or percent detection rate)
 between cell groups. However, genes may be pre-filtered based on their
 minimum detection rate (min.pct) across both cell groups. To use this method,
 please install DESeq2, using the instructions at
 https://bioconductor.org/packages/release/bioc/html/DESeq2.html</p></li>
</ul></p></dd>
    <dt><code>slot</code></dt><dd><p>Slot to pull data from; note that if <code>test.use</code> is "negbinom", "poisson", or "DESeq2",
<code>slot</code> will be set to "counts"</p></dd>
    <dt><code>min.pct</code></dt><dd><p>only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations. Meant to speed up the function
by not testing genes that are very infrequently expressed. Default is 0.1</p></dd>
    <dt><code>min.diff.pct</code></dt><dd><p>only test genes that show a minimum difference in the
fraction of detection between the two groups. Set to -Inf by default</p></dd>
    <dt><code>node</code></dt><dd><p>A node to find markers for and all its children; requires
<code><a href='https://satijalab.org/seurat/reference/BuildClusterTree.html'>BuildClusterTree</a></code> to have been run previously; replaces <code>FindAllMarkersNode</code></p></dd>
    <dt><code>verbose</code></dt><dd><p>Print a progress bar once expression testing begins</p></dd>
    <dt><code>only.pos</code></dt><dd><p>Only return positive markers (FALSE by default)</p></dd>
    <dt><code>max.cells.per.ident</code></dt><dd><p>Down sample each identity class to a max number.
Default is no downsampling. Not activated by default (set to Inf)</p></dd>
    <dt><code>random.seed</code></dt><dd><p>Random seed for downsampling</p></dd>
    <dt><code>latent.vars</code></dt><dd><p>Variables to test, used only when <code>test.use</code> is one of
'LR', 'negbinom', 'poisson', or 'MAST'</p></dd>
    <dt><code>min.cells.feature</code></dt><dd><p>Minimum number of cells expressing the feature in at least one
of the two groups, currently only used for poisson and negative binomial tests</p></dd>
    <dt><code>min.cells.group</code></dt><dd><p>Minimum number of cells in one of the groups</p></dd>
    <dt><code>pseudocount.use</code></dt><dd><p>Pseudocount to add to averaged expression values when
calculating logFC. 1 by default.</p></dd>
    <dt><code>mean.fxn</code></dt><dd><p>Function to use for fold change or average difference calculation.
If NULL, the appropriate function will be chose according to the slot used</p></dd>
    <dt><code>fc.name</code></dt><dd><p>Name of the fold change, average difference, or custom function column
in the output data.frame. If NULL, the fold change column will be named
according to the logarithm base (eg, "avg_log2FC"), or if using the scale.data
slot "avg_diff".</p></dd>
    <dt><code>base</code></dt><dd><p>The base with respect to which logarithms are computed.</p></dd>
    <dt><code>return.thresh</code></dt><dd><p>Only return markers that have a p-value &lt; return.thresh, or a power &gt; return.thresh (if the test is ROC)</p></dd>
  
</dl></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An unfiltered iTALK df sorted by relevance</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Stats:
Ligand and Receptor Expressions (and P-values if ran with .DE==TRUE)
Dot params are inherited from Seurat::FindAllMarkers, if .deg = TRUE</p>
<p>In this case, we use the product of the logFC rather than thresholding, as in
the original implementation.</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


