<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]"><title>Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]] — call_natmi • liana</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]] — call_natmi"><meta property="og:description" content="Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">liana</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.10</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/liana_cc2tensor.html">Context Factorisation with tensor-cell2cell</a>
    <a class="dropdown-item" href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    <a class="dropdown-item" href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    <a class="dropdown-item" href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    <a class="dropdown-item" href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    <a class="dropdown-item" href="../articles/liana_tutorial.html">LIANA Tutorial</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/liana/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]</h1>
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/HEAD/R/natmi_pipe.R" class="external-link"><code>R/natmi_pipe.R</code></a></small>
      <div class="d-none name"><code>call_natmi.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">call_natmi</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  <span class="va">op_resource</span>,</span>
<span>  expr_file <span class="op">=</span> <span class="st">"em.csv"</span>,</span>
<span>  meta_file <span class="op">=</span> <span class="st">"metadata.csv"</span>,</span>
<span>  output_dir <span class="op">=</span> <span class="st">"NATMI_test"</span>,</span>
<span>  reso_name <span class="op">=</span> <span class="st">"placeholder"</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  num_cor <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  conda_env <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>  .format <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  .overwrite_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  .seed <span class="op">=</span> <span class="fl">1004</span>,</span>
<span>  .natmi_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  .delete_input_output <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>sce</dt>
<dd><p>Seurat or SingleCellExperiment object</p></dd>


<dt>op_resource</dt>
<dd><p>List of OmniPath resources</p></dd>


<dt>expr_file</dt>
<dd><p>expression matrix file name</p></dd>


<dt>meta_file</dt>
<dd><p>annotations (i.e. clusters) file name</p></dd>


<dt>output_dir</dt>
<dd><p>NATMI output directory</p></dd>


<dt>reso_name</dt>
<dd><p>name of the resource usually in the format list(name = op_resource)</p></dd>


<dt>assay</dt>
<dd><p>Seurat assay to be used</p></dd>


<dt>num_cor</dt>
<dd><p>number of cores to be used</p></dd>


<dt>conda_env</dt>
<dd><p>name of the conda environment via which NATMI is called</p></dd>


<dt>assay.type</dt>
<dd><p>logcounts by default, but it's converted back into counts
as suggested by the authors</p></dd>


<dt>.format</dt>
<dd><p>bool whether to format output</p></dd>


<dt>.overwrite_data</dt>
<dd><p>bool whether Extract and overwrite csv with
data from Seurat Object</p></dd>


<dt>.seed</dt>
<dd><p>random seed</p></dd>


<dt>.natmi_path</dt>
<dd><p>path of NATMI code and dbs (by default set to liana path)</p></dd>


<dt>.delete_input_output</dt>
<dd><p>logical whether to delete input and
output after run.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>DF with NATMI results</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function will save NATMI dbs folder, then it will call the
    NATMI Python from the NATMI dir and save the output into a specified
    directory in NATMI's path.
    It will then load the csvs and format the output to a list of lists.</p>
<p>By default, NATMI's path is set to that of LIANA, but any alternative
    path can be passed</p>
<p>==============================================================================
NATMI Arguments:
  --interDB INTERDB
                        lrc2p (default) has literature supported ligand-receptor pairs | lrc2a has putative and literature supported ligand-receptor pairs | the user-supplied interaction database can also be used by calling the name of database file without extension
  --interSpecies INTERSPECIES
                        human (default) | mouse | expandp | expanda
  --emFile EMFILE       the path to the normalised expression matrix file with row names (gene identifiers) and column names (cell-type/single-cell identifiers)
  --annFile ANNFILE     the path to the metafile in which column one has single-cell identifiers and column two has corresponding cluster IDs (see file 'toy.sc.ann.txt' as an example)
  --species SPECIES     human (default) | mouse | rat | zebrafish | fruitfly | chimpanzee | dog | monkey | cattle | chicken | frog | mosquito | nematode | thalecress | rice | riceblastfungus | bakeryeast | neurosporacrassa | fissionyeast | eremotheciumgossypii | kluyveromyceslactis, 21 species are supported
  --idType IDTYPE       symbol (default) | entrez(https://www.ncbi.nlm.nih.gov/gene) | ensembl(https://www.ensembl.org/) | uniprot(https://www.uniprot.org/) | hgnc(https://www.genenames.org/) | mgi(http://www.informatics.jax.org/mgihome/nomen/index.shtml) | custom(gene identifier used in the expression matrix)
  --coreNum CORENUM     the number of CPU cores used, default is one
  --out OUT             the path to save the analysis results
(Taken From NATMI's GitHub Page)</p>
<p>Stats:
1) The mean-expression edge weights
2) The specificity-based edge weights
* a weight of 1 means both the ligand and receptor are only expressed
 in one cell type</p>
<p>Note that `call_natmi` will write the expression matrix to CSV each time
its called, unless .overwrite_data is set to FALSE!
This can be an extremely time consuming step when working with large datasets</p>
<p>Also, NATMI will sometimes create duplicate files, so please consider
 saving each run in a new folder. An easy fix would be to simply delete the
 output, but I am reluctant to automatically delete files via an R script.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

