<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]] — call_natmi • liana</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]] — call_natmi"><meta property="og:description" content="Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_intracell.html">LIANA with Intracellular Omnipath</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/saezlab/liana/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org" class="external-link">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]</h1>
    <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/HEAD/R/natmi_pipe.R" class="external-link"><code>R/natmi_pipe.R</code></a></small>
    <div class="hidden name"><code>call_natmi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Call NATMI Pipeline from R with Resources Querried from OmniPath [[DEPRECATED]]</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">call_natmi</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  <span class="va">op_resource</span>,</span>
<span>  expr_file <span class="op">=</span> <span class="st">"em.csv"</span>,</span>
<span>  meta_file <span class="op">=</span> <span class="st">"metadata.csv"</span>,</span>
<span>  output_dir <span class="op">=</span> <span class="st">"NATMI_test"</span>,</span>
<span>  reso_name <span class="op">=</span> <span class="st">"placeholder"</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  num_cor <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  conda_env <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>  .format <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  .overwrite_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  .seed <span class="op">=</span> <span class="fl">1004</span>,</span>
<span>  .natmi_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  .delete_input_output <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>sce</dt>
<dd><p>Seurat or SingleCellExperiment object</p></dd>


<dt>op_resource</dt>
<dd><p>List of OmniPath resources</p></dd>


<dt>expr_file</dt>
<dd><p>expression matrix file name</p></dd>


<dt>meta_file</dt>
<dd><p>annotations (i.e. clusters) file name</p></dd>


<dt>output_dir</dt>
<dd><p>NATMI output directory</p></dd>


<dt>reso_name</dt>
<dd><p>name of the resource usually in the format list(name = op_resource)</p></dd>


<dt>assay</dt>
<dd><p>Seurat assay to be used</p></dd>


<dt>num_cor</dt>
<dd><p>number of cores to be used</p></dd>


<dt>conda_env</dt>
<dd><p>name of the conda environment via which NATMI is called</p></dd>


<dt>assay.type</dt>
<dd><p>logcounts by default, but it's converted back into counts
as suggested by the authors</p></dd>


<dt>.format</dt>
<dd><p>bool whether to format output</p></dd>


<dt>.overwrite_data</dt>
<dd><p>bool whether Extract and overwrite csv with
data from Seurat Object</p></dd>


<dt>.seed</dt>
<dd><p>random seed</p></dd>


<dt>.natmi_path</dt>
<dd><p>path of NATMI code and dbs (by default set to liana path)</p></dd>


<dt>.delete_input_output</dt>
<dd><p>logical whether to delete input and
output after run.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>DF with NATMI results</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function will save NATMI dbs folder, then it will call the
    NATMI Python from the NATMI dir and save the output into a specified
    directory in NATMI's path.
    It will then load the csvs and format the output to a list of lists.</p>
<p>By default, NATMI's path is set to that of LIANA, but any alternative
    path can be passed</p>
<p>==============================================================================
NATMI Arguments:
  --interDB INTERDB
                        lrc2p (default) has literature supported ligand-receptor pairs | lrc2a has putative and literature supported ligand-receptor pairs | the user-supplied interaction database can also be used by calling the name of database file without extension
  --interSpecies INTERSPECIES
                        human (default) | mouse | expandp | expanda
  --emFile EMFILE       the path to the normalised expression matrix file with row names (gene identifiers) and column names (cell-type/single-cell identifiers)
  --annFile ANNFILE     the path to the metafile in which column one has single-cell identifiers and column two has corresponding cluster IDs (see file 'toy.sc.ann.txt' as an example)
  --species SPECIES     human (default) | mouse | rat | zebrafish | fruitfly | chimpanzee | dog | monkey | cattle | chicken | frog | mosquito | nematode | thalecress | rice | riceblastfungus | bakeryeast | neurosporacrassa | fissionyeast | eremotheciumgossypii | kluyveromyceslactis, 21 species are supported
  --idType IDTYPE       symbol (default) | entrez(https://www.ncbi.nlm.nih.gov/gene) | ensembl(https://www.ensembl.org/) | uniprot(https://www.uniprot.org/) | hgnc(https://www.genenames.org/) | mgi(http://www.informatics.jax.org/mgihome/nomen/index.shtml) | custom(gene identifier used in the expression matrix)
  --coreNum CORENUM     the number of CPU cores used, default is one
  --out OUT             the path to save the analysis results
(Taken From NATMI's GitHub Page)</p>
<p>Stats:
1) The mean-expression edge weights
2) The specificity-based edge weights
* a weight of 1 means both the ligand and receptor are only expressed
 in one cell type</p>
<p>Note that `call_natmi` will write the expression matrix to CSV each time
its called, unless .overwrite_data is set to FALSE!
This can be an extremely time consuming step when working with large datasets</p>
<p>Also, NATMI will sometimes create duplicate files, so please consider
 saving each run in a new folder. An easy fix would be to simply delete the
 output, but I am reluctant to automatically delete files via an R script.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

