<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>LIANA wrapper function — liana_wrap • liana</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="LIANA wrapper function — liana_wrap"><meta property="og:description" content="LIANA wrapper function"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/liana_cc2tensor.html">Context Factorisation with tensor-cell2cell</a>
    </li>
    <li>
      <a href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/saezlab/liana/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org" class="external-link">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>LIANA wrapper function</h1>
    <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/HEAD/R/liana_wrapper.R" class="external-link"><code>R/liana_wrapper.R</code></a></small>
    <div class="hidden name"><code>liana_wrap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>LIANA wrapper function</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">liana_wrap</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"natmi"</span>, <span class="st">"connectome"</span>, <span class="st">"logfc"</span>, <span class="st">"sca"</span>, <span class="st">"cellphonedb"</span><span class="op">)</span>,</span>
<span>  resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Consensus"</span><span class="op">)</span>,</span>
<span>  idents_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">external_resource</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  .simplify <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cell.adj <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  base <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>sce</dt>
<dd><p>`SingleCellExperiment` object or `SeuratObject`</p></dd>


<dt>method</dt>
<dd><p>method(s) to be run via liana</p></dd>


<dt>resource</dt>
<dd><p>resource(s) to be used by the methods (`Consensus` by default), Use `all` to run all resources in one go),
or `custom` to run liana_wrap with an appropriately formatted custom resource, passed via `exernal_resource`</p></dd>


<dt>idents_col</dt>
<dd><p>the cell identities/labels to be used. By default set to NULL, and will used the active
Idents or colLabels from the seurat_object or SCE, respectively.</p></dd>


<dt>external_resource</dt>
<dd><p>external resource in OmniPath tibble format</p></dd>


<dt>verbose</dt>
<dd><p>logical whether to output messages and warnings (`TRUE` by default)</p></dd>


<dt>assay</dt>
<dd><p>assay to be used by Seurat, by default set to `NULL` and will use the DefaultAssay.</p></dd>


<dt>.simplify</dt>
<dd><p>if methods are run with only 1 resource, return a list
of tibbles for each method (default), rather than a list of lists with
method-resource combinations</p></dd>


<dt>cell.adj</dt>
<dd><p>cell adjacency tibble/dataframe /w weights by which we will
`multiply` the relevant columns. Any cell pairs with a weights of 0 will be
filtered out.
Note that if working with LIANA's default methods, we suggest weights &gt;= 0 &amp; =&lt; 1.
This ensure that all methods' score will be meaningfully weighed without
changing the interpretation of their scores, thus allow one to filter SCA,
rank NATMI, etc.</p></dd>


<dt>base</dt>
<dd><p>Default to NULL (i.e. log2-transformation is assumed
for SCE, and log-tranformation for Seurat). This is a requred step for the
calculation of the logFC method - ensures that any other preprocessing of
the counts is preserved. One could also pass `NaN` if they wish to use the
counts stored in the counts assay/slot, or any other number according to the
base that was used for log-tranformation.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="liana_defaults.html">liana_defaults</a></code></p><dl><dt><code>liana_pipe.params</code></dt>
<dd><p>list of Parameters passed to NATMI <code><a href="liana_pipe.html">liana_pipe</a></code></p></dd>

    <dt><code>liana_call.params</code></dt>
<dd><p>list of Parameters passed to NATMI <code><a href="liana_call.html">liana_call</a></code></p></dd>

    <dt><code>cellchat.params</code></dt>
<dd><p>list of Parameters passed to CellChat <code><a href="call_cellchat.html">call_cellchat</a></code></p></dd>

    <dt><code>squidpy.params</code></dt>
<dd><p>list of Parameters passed to Squidpy <code><a href="call_squidpy.html">call_squidpy</a></code></p></dd>

    <dt><code>call_connectome.params</code></dt>
<dd><p>list of Parameters passed to Connectome <code><a href="call_connectome.html">call_connectome</a></code></p></dd>

    <dt><code>call_italk.params</code></dt>
<dd><p>list of Parameters passed to iTALK <code><a href="call_italk.html">call_italk</a></code></p></dd>

    <dt><code>call_natmi.params</code></dt>
<dd><p>list of Parameters passed to NATMI <code><a href="call_natmi.html">call_natmi</a></code></p></dd>

    <dt><code>call_sca.params</code></dt>
<dd><p>list of Parameters passed to SingleCellSignalR <code><a href="call_sca.html">call_sca</a></code></p></dd>

    <dt><code>cellphonedb.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
cellphonedb implementation <code><a href="cellphonedb_score.html">cellphonedb_score</a></code></p></dd>

    <dt><code>natmi.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
edge_specificity implementation <code>natmi_score</code></p></dd>

    <dt><code>sca.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
LRScore implementation <code>sca_score</code></p></dd>

    <dt><code>connectome.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
connectome's weight_sc implementation <code>connectome_score</code></p></dd>

    <dt><code>cytotalk.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
crosstalk scores implementation <code><a href="cytotalk_score.html">cytotalk_score</a></code></p></dd>

    <dt><code>logfc.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
logFC implementation <code>logfc_score</code></p></dd>

    <dt><code>permutation.params</code></dt>
<dd><p>list of parameters passed to permutation methods</p></dd>

    <dt><code>expr_prop</code></dt>
<dd><p>minimum proportion of gene expression per cell type (0.1 by default).
This is further applied at a method-specific level - it is not applied to
 Connectome and Cytotalk, as these methods calculate/normalize the score (or
 statistics used in their scores) across all cell types or cell type pairs.</p></dd>

    <dt><code>complex_policy</code></dt>
<dd><p>policy how to account for the presence of complexes.</p></dd>

    <dt><code>seed</code></dt>
<dd><p>random seed integer</p></dd>

    <dt><code>parallelize</code></dt>
<dd><p>whether to parallelize cellphonedb-like</p></dd>

    <dt><code>workers</code></dt>
<dd><p>number of workers to be called</p></dd>

    <dt><code>assay.type</code></dt>
<dd><p>- the type of data to be used to calculate the means
(logcounts by default), available options are: "counts" and "logcounts"</p></dd>

  
</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list of method-resource results - i.e. provided resources are run
with each method
If only one resource is selected, a single tibble (with results for that
 resource) will be returned for each of the selected methods</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>LIANA wrapper method that can be used to call each method with
 a given set of intercellular resources from the OmniPath universe.
 Please see `liana_defaults()` for more information about the
 default parameters passed used by `liana_wrap`, if you wish to modify them.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">liana_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"liana"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># load testdata</span></span></span>
<span class="r-in"><span><span class="va">testdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">liana_path</span> , <span class="st">"testdata"</span>, <span class="st">"input"</span>, <span class="st">"testdata.rds"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># run only 2 methods from liana</span></span></span>
<span class="r-in"><span><span class="va">liana_res</span> <span class="op">&lt;-</span> <span class="fu">liana_wrap</span><span class="op">(</span><span class="va">testdata</span>, method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cellphonedb'</span>, <span class="st">'sca'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                        resource <span class="op">=</span> <span class="st">'Consensus'</span>, <span class="co"># default resource</span></span></span>
<span class="r-in"><span>                        <span class="co"># example run with *only* 2 permutations for cpdb</span></span></span>
<span class="r-in"><span>                        permutation.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>nperms <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Expression from the `RNA` assay will be used</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running LIANA with `seurat_annotations` as labels!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>3465 genes and/or 0 cells were removed as they had no counts!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LIANA: LR summary stats calculated!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now Running: Cellphonedb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now Running: Sca</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

