<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="LIANA wrapper function"><title>LIANA wrapper function — liana_wrap • liana</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="LIANA wrapper function — liana_wrap"><meta property="og:description" content="LIANA wrapper function"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">liana</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/liana_cc2tensor.html">Context Factorisation with tensor-cell2cell</a>
    <a class="dropdown-item" href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    <a class="dropdown-item" href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    <a class="dropdown-item" href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    <a class="dropdown-item" href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    <a class="dropdown-item" href="../articles/liana_tutorial.html">LIANA Tutorial</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/liana/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>LIANA wrapper function</h1>
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/HEAD/R/liana_wrapper.R" class="external-link"><code>R/liana_wrapper.R</code></a></small>
      <div class="d-none name"><code>liana_wrap.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>LIANA wrapper function</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">liana_wrap</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"natmi"</span>, <span class="st">"connectome"</span>, <span class="st">"logfc"</span>, <span class="st">"sca"</span>, <span class="st">"cellphonedb"</span><span class="op">)</span>,</span>
<span>  resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Consensus"</span><span class="op">)</span>,</span>
<span>  idents_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">external_resource</span>,</span>
<span>  min_cells <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  return_all <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  supp_columns <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  .simplify <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cell.adj <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  base <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>sce</dt>
<dd><p>`SingleCellExperiment` object or `SeuratObject`</p></dd>


<dt>method</dt>
<dd><p>method(s) to be run via liana</p></dd>


<dt>resource</dt>
<dd><p>resource(s) to be used by the methods (`Consensus` by default), Use `all` to run all resources in one go),
or `custom` to run liana_wrap with an appropriately formatted custom resource, passed via `exernal_resource`</p></dd>


<dt>idents_col</dt>
<dd><p>the cell identities/labels to be used. By default set to NULL, and will used the active
Idents or colLabels from the seurat_object or SCE, respectively.</p></dd>


<dt>external_resource</dt>
<dd><p>external resource in OmniPath tibble format</p></dd>


<dt>min_cells</dt>
<dd><p>minimum cell per cell identity to be considered for analysis</p></dd>


<dt>return_all</dt>
<dd><p>whether to return all possible interactions. Any
interaction with `expr_prop` below the specific threshold will be
assigned to the *worst* possible score in those that pass the threshold.
 For example, p-values from CellPhoneDB will be assigned to max(pvalue)
  - likely 1, and lr_means will be assigned to min(lr_means).
 Note that this applies only to the internal methods of liana.</p></dd>


<dt>supp_columns</dt>
<dd><p>any supplementary/additional columns which are to be
returned by liana. Possibilities include: c("ligand.expr", "receptor.expr"
"ligand.stat", "receptor.stat", "ligand.pval", "receptor.pval",
"ligand.FDR", "receptor.FDR", etc)</p></dd>


<dt>verbose</dt>
<dd><p>logical whether to output messages and warnings (`TRUE` by default)</p></dd>


<dt>assay</dt>
<dd><p>assay to be used by Seurat, by default set to `NULL` and will use the DefaultAssay.</p></dd>


<dt>.simplify</dt>
<dd><p>if methods are run with only 1 resource, return a list
of tibbles for each method (default), rather than a list of lists with
method-resource combinations</p></dd>


<dt>cell.adj</dt>
<dd><p>cell adjacency tibble/dataframe /w weights by which we will
`multiply` the relevant columns. Any cell pairs with a weights of 0 will be
filtered out.
Note that if working with LIANA's default methods, we suggest weights &gt;= 0 &amp; =&lt; 1.
This ensure that all methods' score will be meaningfully weighed without
changing the interpretation of their scores, thus allow one to filter SCA,
rank NATMI, etc.</p></dd>


<dt>base</dt>
<dd><p>Default to NULL (i.e. log2-transformation is assumed
for SCE, and log-tranformation for Seurat). This is a requred step for the
calculation of the logFC method - ensures that any other preprocessing of
the counts is preserved. One could also pass `NaN` if they wish to use the
counts stored in the counts assay/slot, or any other number according to the
base that was used for log-tranformation.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="liana_defaults.html">liana_defaults</a></code></p><dl><dt><code>expr_prop</code></dt>
<dd><p>minimum proportion of gene expression per cell type (0.1 by default).
Note that when working with complexes, the minimum subunit proportion will
be used for filtering.</p></dd>

    <dt><code>complex_policy</code></dt>
<dd><p>policy how to account for the presence of complexes.</p></dd>

    <dt><code>seed</code></dt>
<dd><p>random seed integer</p></dd>

    <dt><code>parallelize</code></dt>
<dd><p>whether to parallelize cellphonedb-like</p></dd>

    <dt><code>workers</code></dt>
<dd><p>number of workers to be called</p></dd>

    <dt><code>liana_pipe.params</code></dt>
<dd><p>list of Parameters passed to NATMI <code><a href="liana_pipe.html">liana_pipe</a></code></p></dd>

    <dt><code>liana_call.params</code></dt>
<dd><p>list of Parameters passed to NATMI <code><a href="liana_call.html">liana_call</a></code></p></dd>

    <dt><code>cellchat.params</code></dt>
<dd><p>list of Parameters passed to CellChat <code><a href="call_cellchat.html">call_cellchat</a></code></p></dd>

    <dt><code>squidpy.params</code></dt>
<dd><p>list of Parameters passed to Squidpy <code><a href="call_squidpy.html">call_squidpy</a></code></p></dd>

    <dt><code>call_connectome.params</code></dt>
<dd><p>list of Parameters passed to Connectome <code><a href="call_connectome.html">call_connectome</a></code></p></dd>

    <dt><code>call_italk.params</code></dt>
<dd><p>list of Parameters passed to iTALK <code><a href="call_italk.html">call_italk</a></code></p></dd>

    <dt><code>call_natmi.params</code></dt>
<dd><p>list of Parameters passed to NATMI <code><a href="call_natmi.html">call_natmi</a></code></p></dd>

    <dt><code>call_sca.params</code></dt>
<dd><p>list of Parameters passed to SingleCellSignalR <code><a href="call_sca.html">call_sca</a></code></p></dd>

    <dt><code>cellphonedb.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
cellphonedb implementation <code><a href="cellphonedb_score.html">cellphonedb_score</a></code></p></dd>

    <dt><code>natmi.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
edge_specificity implementation <code>natmi_score</code></p></dd>

    <dt><code>sca.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
LRScore implementation <code>sca_score</code></p></dd>

    <dt><code>connectome.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
connectome's weight_sc implementation <code>connectome_score</code></p></dd>

    <dt><code>cytotalk.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
crosstalk scores implementation <code><a href="cytotalk_score.html">cytotalk_score</a></code></p></dd>

    <dt><code>logfc.params</code></dt>
<dd><p>list of Parameters passed to liana's internal
logFC implementation <code>logfc_score</code></p></dd>

    <dt><code>permutation.params</code></dt>
<dd><p>list of parameters passed to permutation methods</p></dd>

    <dt><code>assay.type</code></dt>
<dd><p>- the type of data to be used to calculate the means
(logcounts by default), available options are: "counts" and "logcounts"</p></dd>

  
</dl></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A list of method-resource results - i.e. provided resources are run
with each method
If only one resource is selected, a single tibble (with results for that
 resource) will be returned for each of the selected methods</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>LIANA wrapper method that can be used to call each method with
 a given set of intercellular resources from the OmniPath universe.
 Please see `liana_defaults()` for more information about the
 default parameters passed used by `liana_wrap`, if you wish to modify them.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">liana_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"liana"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># load testdata</span></span></span>
<span class="r-in"><span><span class="va">testdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">liana_path</span> , <span class="st">"testdata"</span>, <span class="st">"input"</span>, <span class="st">"testdata.rds"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># run only 2 methods from liana</span></span></span>
<span class="r-in"><span><span class="va">liana_res</span> <span class="op">&lt;-</span> <span class="fu">liana_wrap</span><span class="op">(</span><span class="va">testdata</span>, method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cellphonedb'</span>, <span class="st">'sca'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                        resource <span class="op">=</span> <span class="st">'Consensus'</span>, <span class="co"># default resource</span></span></span>
<span class="r-in"><span>                        <span class="co"># example run with *only* 2 permutations for cpdb</span></span></span>
<span class="r-in"><span>                        permutation.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>nperms <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Expression from the `RNA` assay will be used</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running LIANA with `seurat_annotations` as labels!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>3465 genes and/or 0 cells were removed as they had no counts!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LIANA: LR summary stats calculated!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now Running: Cellphonedb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Now Running: Sca</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

