<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>LIANA Tutorial • liana</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="LIANA Tutorial">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="liana_tutorial_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>LIANA Tutorial</h1>
                        <h4 class="author">Daniel Dimitrov</h4>
            <address class="author_afil">
      Saezlab, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:daniel.dimitrov@uni-heidelberg.de" class="email">daniel.dimitrov@uni-heidelberg.de</a>
      </address>
                  
            <h4 class="date">17/06/2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/master/vignettes/liana_tutorial.Rmd"><code>vignettes/liana_tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>liana_tutorial.Rmd</code></div>

    </div>

    
    
<div id="liana-intro" class="section level1">
<h1 class="hasAnchor">
<a href="#liana-intro" class="anchor"></a><code>liana</code>: Intro</h1>
<p>The continuous developments of single-cell RNA-Seq (scRNA-Seq) have sparked an immense interest in understanding intercellular crosstalk. Multiple tools and resources that aid the investigation of cell-cell communication (CCC) were published recently. However, these methods and resources are usually in a fixed combination of a tool and its corresponding resource, but in principle any resource could be combined with any method.</p>
<p>To this end, we built <code>liana</code> - a framework to decouple the tools from their corresponding resources.</p>
<p><img src="../ligrec_pipe.png"></p>
<div id="load-required-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-required-packages" class="anchor"></a>Load required packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://omnipathdb.org/">liana</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/">tibble</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org">purrr</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="ccc-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#ccc-resources" class="anchor"></a>CCC Resources</h2>
<p><code>liana</code> provides CCC resources obtained and formatted via <a href="https://github.com/saezlab/OmnipathR"><code>OmnipathR</code></a> which are then converted to the appropriate format to each method.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Resource currently included in OmniPathR (and hence `liana`) include:</span>
<span class="fu"><a href="../reference/get_resources.html">get_resources</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Reshuffled"       "Default"          "CellChatDB"       "CellPhoneDB"     </span>
<span class="co">#&gt;  [5] "Ramilowski2015"   "Baccin2019"       "LRdb"             "Kirouac2010"     </span>
<span class="co">#&gt;  [9] "ICELLNET"         "iTALK"            "EMBRACE"          "HPMR"            </span>
<span class="co">#&gt; [13] "Guide2Pharma"     "connectomeDB2020" "talklr"           "CellTalkDB"      </span>
<span class="co">#&gt; [17] "OmniPath"</span>

<span class="co"># A list of resources can be obtained using the `select_resource()` function:</span>
<span class="co"># See `?select_resource()` documentation for further information.</span>
<span class="co"># select_resource(c('OmniPath')) %&gt;% glimpse() </span></code></pre></div>
</div>
<div id="ccc-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#ccc-methods" class="anchor"></a>CCC Methods</h2>
<p>Each of the resources, can then be run with any of the following methods:</p>
<pre><code>- CellPhoneDB algorithm (via Squidpy)
- CellChat
- NATMI
- Connectome
- SingleCellSignalR (SCA)
- iTALK</code></pre>
</div>
<div id="calling-the-liana-wrapper-function" class="section level2">
<h2 class="hasAnchor">
<a href="#calling-the-liana-wrapper-function" class="anchor"></a>Calling the <code>liana</code> wrapper function</h2>
<p>To run <code>liana</code>, we will use a toy PBMCs scRNA-Seq data set <code>liana</code> takes <code>Seurat</code> objects as input, containing processed counts and clustered cells</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">liana_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"liana"</span><span class="op">)</span>
<span class="va">testdata</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">liana_path</span> , <span class="st">"testdata"</span>, <span class="st">"input"</span>, <span class="st">"testdata.rds"</span><span class="op">)</span><span class="op">)</span>

<span class="va">testdata</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Formal class 'Seurat' [package "Seurat"] with 13 slots</span>
<span class="co">#&gt;   ..@ assays      :List of 1</span>
<span class="co">#&gt;   .. ..$ RNA:Formal class 'Assay' [package "Seurat"] with 8 slots</span>
<span class="co">#&gt;   ..@ meta.data   :'data.frame': 90 obs. of  4 variables:</span>
<span class="co">#&gt;   .. ..$ orig.ident        : Factor w/ 1 level "pbmc3k": 1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="co">#&gt;   .. ..$ nCount_RNA        : num [1:90] 4903 3914 4973 3281 2641 ...</span>
<span class="co">#&gt;   .. ..$ nFeature_RNA      : int [1:90] 1352 1112 1445 1015 928 937 899 1713 960 888 ...</span>
<span class="co">#&gt;   .. ..$ seurat_annotations: Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 1 3 3 1 1 1 1 ...</span>
<span class="co">#&gt;   ..@ active.assay: chr "RNA"</span>
<span class="co">#&gt;   ..@ active.ident: Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 1 3 3 1 1 1 1 ...</span>
<span class="co">#&gt;   .. ..- attr(*, "names")= chr [1:90] "AAACATTGAGCTAC" "AAACTTGAAAAACG" "AAAGGCCTGTCTAG" "AAAGTTTGGGGTGA" ...</span>
<span class="co">#&gt;   ..@ graphs      : list()</span>
<span class="co">#&gt;   ..@ neighbors   : list()</span>
<span class="co">#&gt;   ..@ reductions  : list()</span>
<span class="co">#&gt;   ..@ images      : list()</span>
<span class="co">#&gt;   ..@ project.name: chr "SeuratProject"</span>
<span class="co">#&gt;   ..@ misc        : list()</span>
<span class="co">#&gt;   ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1</span>
<span class="co">#&gt;   .. ..$ : int [1:3] 3 2 3</span>
<span class="co">#&gt;   ..@ commands    :List of 1</span>
<span class="co">#&gt;   .. ..$ FindVariableFeatures.RNA:Formal class 'SeuratCommand' [package "Seurat"] with 5 slots</span>
<span class="co">#&gt;   ..@ tools       : list()</span></code></pre></div>
<p><code>liana_wrap</code> calls a number of methods and and each method is run with the provided resources.</p>
<p>Here, we call Connectome and Squidpy for the sake of computational speed, but any of c(“sca”, “natmi”, “italk”, “cellchat”) can also be run by simply adding their name to the <code>method</code> parameter.</p>
<p>Similarly, here we use only the “OmniPath” CCC resource, but any of the aforementioned ones (available via <code><a href="../reference/get_resources.html">get_resources()</a></code>) can be added to the <code>resource</code> parameter</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run liana</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">testdata</span>,
                         method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'squidpy'</span>, <span class="st">'connectome'</span><span class="op">)</span>,
                         resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'OmniPath'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span>

<span class="co">#&gt; Warning in data.frame(source = sources[i], target = targets[j], ligand =</span>
<span class="co">#&gt; ligands, : row names were found from a short variable and have been discarded</span></code></pre></div>
</div>
<div id="obtain-consensus-ranks" class="section level2">
<h2 class="hasAnchor">
<a href="#obtain-consensus-ranks" class="anchor"></a>Obtain Consensus Ranks</h2>
<p><code>liana</code> also provides consensus ranks for the results obtained using different methods. By default, liana will provide mean, median, and a aggregate* consensus ranks</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Liana returns a list of results, each element of which corresponds to a method</span>
<span class="va">liana_test</span>
<span class="co">#&gt; $squidpy</span>
<span class="co">#&gt; # A tibble: 9,396 x 8</span>
<span class="co">#&gt;    ligand receptor source target means pvalue uniprot_source unprot_target</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;          &lt;chr&gt;        </span>
<span class="co">#&gt;  1 NRP2   SEMA4F   B      B          0    NaN O60462         O95754       </span>
<span class="co">#&gt;  2 NRP2   SEMA4F   B      CD8 T      0    NaN O60462         O95754       </span>
<span class="co">#&gt;  3 NRP2   SEMA4F   B      NK         0    NaN O60462         O95754       </span>
<span class="co">#&gt;  4 NRP2   SEMA4F   CD8 T  B          0    NaN O60462         O95754       </span>
<span class="co">#&gt;  5 NRP2   SEMA4F   CD8 T  CD8 T      0    NaN O60462         O95754       </span>
<span class="co">#&gt;  6 NRP2   SEMA4F   CD8 T  NK         0    NaN O60462         O95754       </span>
<span class="co">#&gt;  7 NRP2   SEMA4F   NK     B          0    NaN O60462         O95754       </span>
<span class="co">#&gt;  8 NRP2   SEMA4F   NK     CD8 T      0    NaN O60462         O95754       </span>
<span class="co">#&gt;  9 NRP2   SEMA4F   NK     NK         0    NaN O60462         O95754       </span>
<span class="co">#&gt; 10 BAMBI  BMPR1A   B      B          0    NaN Q13145         P36894       </span>
<span class="co">#&gt; # … with 9,386 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $connectome</span>
<span class="co">#&gt; # A tibble: 9,396 x 8</span>
<span class="co">#&gt;    source target ligand receptor weight_norm weight_sc p_val_adj.lig</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt;</span>
<span class="co">#&gt;  1 CD8 T  B      BMPR2  ACVR1              0    0.0375             1</span>
<span class="co">#&gt;  2 B      B      BMPR2  ACVR1              0    0.0375             1</span>
<span class="co">#&gt;  3 NK     B      BMPR2  ACVR1              0    0.0375             1</span>
<span class="co">#&gt;  4 CD8 T  CD8 T  BMPR2  ACVR1              0   -0.0750             1</span>
<span class="co">#&gt;  5 B      CD8 T  BMPR2  ACVR1              0   -0.0750             1</span>
<span class="co">#&gt;  6 NK     CD8 T  BMPR2  ACVR1              0   -0.0750             1</span>
<span class="co">#&gt;  7 CD8 T  NK     BMPR2  ACVR1              0    0.0375             1</span>
<span class="co">#&gt;  8 NK     NK     BMPR2  ACVR1              0    0.0375             1</span>
<span class="co">#&gt;  9 B      NK     BMPR2  ACVR1              0    0.0375             1</span>
<span class="co">#&gt; 10 B      B      GDF11  ACVR1B             0    0.105              1</span>
<span class="co">#&gt; # … with 9,386 more rows, and 1 more variable: p_val_adj.rec &lt;dbl&gt;</span>

<span class="co"># We can aggregate these results into a dataframe with consensus ranks</span>
<span class="va">liana_test</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 9,396 x 11</span>
<span class="co">#&gt;    source ligand target receptor aggregate_rank mean_rank median_rank</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;             &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 NK     HLA-B  CD8 T  CD8A         0.00000999      11          11  </span>
<span class="co">#&gt;  2 NK     B2M    NK     CD247        0.00000999       4           4  </span>
<span class="co">#&gt;  3 NK     B2M    NK     KLRD1        0.0000191        5           5  </span>
<span class="co">#&gt;  4 NK     B2M    NK     KLRC1        0.0000522       24.5        24.5</span>
<span class="co">#&gt;  5 NK     B2M    NK     KIR2DL3      0.000102        34          34  </span>
<span class="co">#&gt;  6 NK     HLA-B  NK     KIR2DL3      0.000108        58.5        58.5</span>
<span class="co">#&gt;  7 B      LGALS9 NK     HAVCR2       0.000179        45          45  </span>
<span class="co">#&gt;  8 NK     HLA-B  NK     KIR3DL2      0.000192        70          70  </span>
<span class="co">#&gt;  9 NK     HLA-G  NK     KLRC1        0.000196        70.5        70.5</span>
<span class="co">#&gt; 10 CD8 T  PTPRC  B      CD22         0.000236        75          75  </span>
<span class="co">#&gt; # … with 9,386 more rows, and 4 more variables: squidpy.pvalue &lt;dbl&gt;,</span>
<span class="co">#&gt; #   squidpy.rank &lt;dbl&gt;, connectome.weight_sc &lt;dbl&gt;, connectome.rank &lt;dbl&gt;</span></code></pre></div>
<p>*The aggregate consnesus rank is obtained using the RobustRankAggreg package. The RobustRankAggreg package provides different functions to obtain aggregate ranks, by default the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3278763/"><code>RRA</code></a> method is used. RRA provides rank scores which can be interpreted as p-values. Interactions which ranked consistently higher than random are assigned low scores/p-values.</p>
</div>
<div id="call-liana-with-overwritten-default-settings" class="section level2">
<h2 class="hasAnchor">
<a href="#call-liana-with-overwritten-default-settings" class="anchor"></a>Call <code>liana</code> with overwritten default settings</h2>
<p>By default <code>liana</code> is run with the default for each method which can be obtained via <code>liana_default()</code></p>
<p>Alternatively, one can also overwrite the default settings by simply passing them to the liana wrapper function</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Overwrite CellChat default parameters by providing a list of parameters</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">testdata</span>,
                         method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'cellchat'</span>, <span class="st">'squidpy'</span><span class="op">)</span>,
                         resource <span class="op">=</span> <span class="st">'Default'</span>, <span class="co"># Run with Default Resources</span>
                         cellchat.params <span class="op">=</span> 
                           <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
                             nboot <span class="op">=</span> <span class="fl">10</span>,
                             exclude_anns <span class="op">=</span> <span class="cn">NULL</span>,
                             thresh <span class="op">=</span> <span class="fl">1</span>,
                             assay <span class="op">=</span> <span class="st">"RNA"</span>,
                             .normalize <span class="op">=</span> <span class="cn">TRUE</span>,
                             .do_parallel <span class="op">=</span> <span class="cn">FALSE</span>,
                             .raw_use <span class="op">=</span> <span class="cn">TRUE</span>
                             <span class="op">)</span>
                         <span class="op">)</span>
<span class="co">#&gt; The cell groups used for CellChat analysis are  B CD8 T NK</span>

<span class="co"># This returns a list of results for each method</span>
<span class="va">liana_test</span> <span class="op">%&gt;%</span> <span class="va">glimpse</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ cellchat: tibble [56 × 6] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;   ..$ source  : Factor w/ 3 levels "B","CD8 T","NK": 1 2 3 1 2 3 1 2 3 1 ...</span>
<span class="co">#&gt;   ..$ target  : Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 2 2 2 3 3 3 1 ...</span>
<span class="co">#&gt;   ..$ ligand  : chr [1:56] "MIF" "MIF" "MIF" "MIF" ...</span>
<span class="co">#&gt;   ..$ receptor: chr [1:56] "CD74_CXCR4" "CD74_CXCR4" "CD74_CXCR4" "CD74_CXCR4" ...</span>
<span class="co">#&gt;   ..$ prob    : num [1:56] 0.123 0.151 0.166 0.111 0.137 ...</span>
<span class="co">#&gt;   ..$ pval    : num [1:56] 0.7 0.1 0 0.8 0.2 0 0.9 0.9 0.8 0.8 ...</span>
<span class="co">#&gt;  $ squidpy : Named list()</span></code></pre></div>
<div id="information-about-the-parameters-passed-to-each-method-can-be-obtained-via" class="section level6">
<h6>Information about the parameters passed to each method can be obtained via</h6>
<p><code>call_*</code> functions - e.g. ?call_cellchat or ?call_squidpy</p>
</div>
</div>
<div id="call-a-method-alone" class="section level2">
<h2 class="hasAnchor">
<a href="#call-a-method-alone" class="anchor"></a>Call a method alone</h2>
<p>To call a any single method, you can use any of the <code>call_*</code> methods:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RUN cellchat alone with OmniPath</span>
<span class="va">cc_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/call_cellchat.html">call_cellchat</a></span><span class="op">(</span>
  seurat_object <span class="op">=</span> <span class="va">testdata</span>,
  op_resource <span class="op">=</span> <span class="fu"><a href="../reference/select_resource.html">select_resource</a></span><span class="op">(</span><span class="st">'OmniPath'</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  .normalize <span class="op">=</span> <span class="cn">TRUE</span>,
  nboot <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Create a CellChat object from a data matrix</span>
<span class="co">#&gt; Set cell identities for the new CellChat object</span>
<span class="co">#&gt; The cell groups used for CellChat analysis are  B CD8 T NK </span>
<span class="co">#&gt; Issue identified!! Please check the official Gene Symbol of the following genes:  </span>
<span class="co">#&gt;  CYR61 WISP1 CTGF FAM19A4 CD97</span>

<span class="co"># Show CellChat Results</span>
<span class="va">cc_res</span>
<span class="co">#&gt; # A tibble: 70 x 6</span>
<span class="co">#&gt;    source target ligand receptor  prob  pval</span>
<span class="co">#&gt;    &lt;fct&gt;  &lt;fct&gt;  &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1 NK     NK     TYROBP KLRD1    0.144     0</span>
<span class="co">#&gt;  2 B      CD8 T  HLA-B  CD8A     0.154     0</span>
<span class="co">#&gt;  3 CD8 T  CD8 T  HLA-B  CD8A     0.169     0</span>
<span class="co">#&gt;  4 NK     CD8 T  HLA-B  CD8A     0.183     0</span>
<span class="co">#&gt;  5 B      CD8 T  HLA-B  CD3D     0.266     0</span>
<span class="co">#&gt;  6 CD8 T  CD8 T  HLA-B  CD3D     0.288     0</span>
<span class="co">#&gt;  7 NK     CD8 T  HLA-B  CD3D     0.307     0</span>
<span class="co">#&gt;  8 B      NK     HLA-B  KLRD1    0.171     0</span>
<span class="co">#&gt;  9 CD8 T  NK     HLA-B  KLRD1    0.187     0</span>
<span class="co">#&gt; 10 NK     NK     HLA-B  KLRD1    0.201     0</span>
<span class="co">#&gt; # … with 60 more rows</span></code></pre></div>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h2>
<p>We appreciate any feedback, so please do not hesitate to open an issue to the <a href="https://github.com/saezlab/liana">liana github page</a>!</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"liana"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To cite liana in publications use:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Dimitrov, D., Türei, D., Boys, C., Nagai, J.S., Flores, R.O.R., Kim,</span>
<span class="co">#&gt;   H., Szalai, B., Costa, I.G., Dugourd, A., Valdeolivas, A. and</span>
<span class="co">#&gt;   Saez-Rodriguez, J., 2021.  Comparison of Resources and Methods to</span>
<span class="co">#&gt;   infer Cell-Cell Communication from Single-cell RNA Data. bioRxiv.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{,</span>
<span class="co">#&gt;     author = {Daniel Dimitrov and Charlotte Boys and Denes Turei and James Nagai S. and Ricardo Ramirez Flores O. and Hyojin Kim and Bence Szalai and Ivan Costa G. and Aurélien Dugourd and Alberto Valdeolivas and Julio Saez-Rodriguez},</span>
<span class="co">#&gt;     title = {Comparison of Resources and Methods to infer Cell-Cell Communication from Single-cell RNA Data},</span>
<span class="co">#&gt;     journal = {BioRxiv},</span>
<span class="co">#&gt;     year = {2021},</span>
<span class="co">#&gt;     doi = {10.1101/2021.05.21.445160},</span>
<span class="co">#&gt;     encoding = {UTF-8},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">120</span><span class="op">)</span>
<span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt;  setting  value                       </span>
<span class="co">#&gt;  version  R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt;  os       Ubuntu 20.04.2 LTS          </span>
<span class="co">#&gt;  system   x86_64, linux-gnu           </span>
<span class="co">#&gt;  ui       X11                         </span>
<span class="co">#&gt;  language (EN)                        </span>
<span class="co">#&gt;  collate  en_US.UTF-8                 </span>
<span class="co">#&gt;  ctype    en_US.UTF-8                 </span>
<span class="co">#&gt;  tz       Europe/Berlin               </span>
<span class="co">#&gt;  date     2021-07-05                  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt;  package              * version    date       lib source                               </span>
<span class="co">#&gt;  abind                  1.4-5      2016-07-21 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  annotate               1.68.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  AnnotationDbi          1.52.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  assertthat             0.2.1      2019-03-21 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  backports              1.2.1      2020-12-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  bbmle                  1.0.23.1   2020-02-03 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  bdsmatrix              1.3-4      2020-01-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  Biobase              * 2.50.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  BiocGenerics         * 0.36.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  BiocManager            1.30.10    2019-11-16 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  BiocParallel           1.24.1     2020-11-06 [2] Bioconductor                         </span>
<span class="co">#&gt;  BiocStyle            * 2.18.1     2020-11-24 [2] Bioconductor                         </span>
<span class="co">#&gt;  bit                    4.0.4      2020-08-04 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  bit64                  4.0.5      2020-08-30 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  bitops                 1.0-6      2013-08-17 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  blob                   1.2.1      2020-01-20 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  bookdown               0.21       2020-10-13 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  brew                   1.0-6      2011-04-13 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  bslib                  0.2.4      2021-01-25 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  cachem                 1.0.4      2021-02-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  Cairo                  1.5-12.2   2020-07-07 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  caTools                1.18.1     2021-01-11 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  CellChat               1.1.0      2021-05-02 [2] Github (sqjin/CellChat@b3ccf96)      </span>
<span class="co">#&gt;  cellranger             1.1.0      2016-07-27 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  checkmate              2.0.0      2020-02-06 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  circlize               0.4.12     2021-01-08 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  cli                    2.5.0      2021-04-26 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  clue                   0.3-58     2020-12-03 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  cluster                2.1.0      2019-06-19 [5] CRAN (R 4.0.0)                       </span>
<span class="co">#&gt;  coda                   0.19-4     2020-09-30 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  codetools              0.2-16     2018-12-24 [5] CRAN (R 4.0.0)                       </span>
<span class="co">#&gt;  colorspace             2.0-0      2020-11-11 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  combinat               0.0-8      2012-10-29 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  ComplexHeatmap         2.6.2      2020-11-12 [2] Bioconductor                         </span>
<span class="co">#&gt;  Connectome             1.0.1      2021-02-21 [2] Github (msraredon/Connectome@4531dd9)</span>
<span class="co">#&gt;  conquer                1.0.2      2020-08-27 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  cowplot                1.1.1      2020-12-30 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  crayon                 1.4.1      2021-02-08 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  curl                   4.3        2019-12-02 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  data.table             1.14.0     2021-02-21 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  DBI                    1.1.1      2021-01-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  DDRTree                0.1.5      2017-04-30 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  DelayedArray           0.16.1     2021-01-22 [2] Bioconductor                         </span>
<span class="co">#&gt;  deldir                 0.2-10     2021-02-16 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  densityClust           0.3        2017-10-24 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  desc                   1.2.0      2018-05-01 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  DESeq2                 1.30.1     2021-02-19 [2] Bioconductor                         </span>
<span class="co">#&gt;  DEsingle               1.10.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  digest                 0.6.27     2020-10-24 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  distillery             1.2        2020-11-21 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  docopt                 0.7.1      2020-06-24 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  doParallel             1.0.16     2020-10-16 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  dplyr                * 1.0.7      2021-06-18 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  edgeR                  3.32.1     2021-01-14 [2] Bioconductor                         </span>
<span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  evaluate               0.14       2019-05-28 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  extRemes               2.1        2020-11-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  fansi                  0.4.2      2021-01-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  fastICA                1.2-2      2019-07-08 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  fastmap                1.1.0      2021-01-25 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  fitdistrplus           1.1-3      2020-12-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  flexmix                2.3-17     2020-10-12 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  FNN                    1.1.3      2019-02-15 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  foreach                1.5.1      2020-10-15 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  fs                     1.5.0      2020-07-31 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  future                 1.21.0     2020-12-10 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  future.apply           1.7.0      2021-01-04 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  gamlss                 5.2-0      2020-09-12 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  gamlss.data            5.1-4      2019-05-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  gamlss.dist            5.1-7      2020-07-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  genefilter             1.72.1     2021-01-21 [2] Bioconductor                         </span>
<span class="co">#&gt;  geneplotter            1.68.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  generics               0.1.0      2020-10-31 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  GenomeInfoDb           1.26.2     2020-12-08 [2] Bioconductor                         </span>
<span class="co">#&gt;  GenomeInfoDbData       1.2.4      2020-12-14 [2] Bioconductor                         </span>
<span class="co">#&gt;  GenomicRanges          1.42.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  GetoptLong             1.0.5      2020-12-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  gg.gap                 1.3        2019-09-30 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  ggalluvial             0.12.3     2020-12-05 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  ggplot2                3.3.3      2020-12-30 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  ggrepel                0.9.1      2021-01-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  ggridges               0.5.3      2021-01-08 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  GlobalOptions          0.1.2      2020-06-10 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  globals                0.14.0     2020-11-22 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  glue                   1.4.2      2020-08-27 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  goftest                1.2-2      2019-12-02 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  gplots                 3.1.1      2020-11-28 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  gridBase               0.4-7      2014-02-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  gtable                 0.3.0      2019-03-25 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  gtools                 3.8.2      2020-03-31 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  hms                    1.0.0      2021-01-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  HSMMSingleCell         1.10.0     2020-10-29 [2] Bioconductor                         </span>
<span class="co">#&gt;  htmltools              0.5.1.1    2021-01-22 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  htmlwidgets            1.5.3      2020-12-10 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  httpuv                 1.5.5      2021-01-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  httr                   1.4.2      2020-07-20 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  ica                    1.0-2      2018-05-24 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  igraph                 1.2.6      2020-10-06 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  IRanges                2.24.1     2020-12-12 [2] Bioconductor                         </span>
<span class="co">#&gt;  irlba                  2.3.3      2019-02-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  iTALK                  0.1.0      2021-02-21 [2] Github (Coolgenome/iTALK@6d9b390)    </span>
<span class="co">#&gt;  iterators              1.0.13     2020-10-15 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  jquerylib              0.1.3      2020-12-17 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  jsonlite               1.7.2      2020-12-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  KernSmooth             2.23-17    2020-04-26 [5] CRAN (R 4.0.0)                       </span>
<span class="co">#&gt;  knitr                  1.31       2021-01-27 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  later                  1.1.0.1    2020-06-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  lattice                0.20-41    2020-04-02 [5] CRAN (R 4.0.0)                       </span>
<span class="co">#&gt;  lazyeval               0.2.2      2019-03-15 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  leiden                 0.3.7      2021-01-26 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  liana                * 0.0.1      2021-07-05 [1] local                                </span>
<span class="co">#&gt;  lifecycle              1.0.0      2021-02-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  limma                  3.46.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  listenv                0.8.0      2019-12-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  Lmoments               1.3-1      2019-03-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  lmtest                 0.9-38     2020-09-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  locfit                 1.5-9.4    2020-03-25 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  logger                 0.2.0      2021-03-04 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  magrittr               2.0.1      2020-11-17 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  MASS                   7.3-53     2020-09-09 [5] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  MAST                   1.16.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  Matrix                 1.3-4      2021-06-01 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  MatrixGenerics         1.2.1      2021-01-30 [2] Bioconductor                         </span>
<span class="co">#&gt;  MatrixModels           0.4-1      2015-08-22 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  matrixStats            0.58.0     2021-01-29 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  maxLik                 1.4-6      2020-11-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  memoise                2.0.0      2021-01-26 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  mgcv                   1.8-33     2020-08-27 [5] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  mime                   0.10       2021-02-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  miniUI                 0.1.1.1    2018-05-18 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  miscTools              0.6-26     2019-12-08 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  modeltools             0.2-23     2020-03-05 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  monocle                2.18.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  multtest               2.46.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  munsell                0.5.0      2018-06-12 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  mvtnorm                1.1-1      2020-06-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  network                1.16.1     2020-10-07 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  nlme                   3.1-149    2020-08-23 [5] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  NMF                    0.23.0     2020-08-01 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  nnet                   7.3-14     2020-04-26 [5] CRAN (R 4.0.0)                       </span>
<span class="co">#&gt;  numDeriv               2016.8-1.1 2019-06-06 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  OmnipathR              2.99.18    2021-06-09 [2] Github (saezlab/OmnipathR@ff3ad88)   </span>
<span class="co">#&gt;  parallelly             1.24.0     2021-03-14 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  patchwork              1.1.1      2020-12-17 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  pbapply                1.4-3      2020-08-18 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  pcaMethods             1.82.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  pheatmap               1.0.12     2019-01-04 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  pillar                 1.6.1      2021-05-16 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  pkgdown                1.6.1      2020-09-12 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  pkgmaker               0.32.2     2020-10-20 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  plotly                 4.9.3      2021-01-10 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  plotrix              * 3.8-1      2021-01-21 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  plyr                   1.8.6      2020-03-03 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  png                    0.1-7      2013-12-03 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  polyclip               1.10-0     2019-03-14 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  pracma                 2.3.3      2021-01-23 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  prettyunits            1.1.1      2020-01-24 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  progress               1.2.2      2019-05-16 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  promises               1.2.0.1    2021-02-11 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  pscl                   1.5.5      2020-03-07 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  purrr                * 0.3.4      2020-04-17 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  qlcMatrix              0.9.7      2018-04-20 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  quantreg               5.83       2021-01-22 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  R6                     2.5.0      2020-10-28 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  ragg                   1.1.0      2021-02-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  randomcoloR            1.1.0.1    2019-11-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  RANN                   2.6.1      2019-01-08 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  rappdirs               0.3.3      2021-01-31 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  RColorBrewer           1.1-2      2014-12-07 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  Rcpp                   1.0.6      2021-01-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  RcppAnnoy              0.0.18     2020-12-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  RcppArmadillo          0.10.2.1.0 2021-02-09 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  RCurl                  1.98-1.3   2021-03-16 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  readr                  1.4.0      2020-10-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  readxl                 1.3.1      2019-03-13 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  registry               0.5-1      2019-03-05 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  reshape2               1.4.4      2020-04-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  reticulate             1.18       2020-10-25 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  rjson                  0.2.20     2018-06-08 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  rlang                  0.4.10     2020-12-30 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  rle                    0.9.2      2020-09-25 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  rmarkdown              2.7        2021-02-19 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  RMTstat                0.3        2014-11-01 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  rngtools               1.5        2020-01-23 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  RobustRankAggreg       1.1        2013-06-03 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  ROCR                   1.0-11     2020-05-02 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  Rook                   1.1-1      2014-10-20 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  rpart                  4.1-15     2019-04-12 [5] CRAN (R 4.0.0)                       </span>
<span class="co">#&gt;  rprojroot              2.0.2      2020-11-15 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  RSpectra               0.16-0     2019-12-01 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  RSQLite                2.2.3      2021-01-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  rstudioapi             0.13       2020-11-12 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  Rtsne                  0.15       2018-11-10 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  S4Vectors              0.28.1     2020-12-09 [2] Bioconductor                         </span>
<span class="co">#&gt;  sandwich               3.0-0      2020-10-02 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  sass                   0.3.1      2021-01-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  scales                 1.1.1      2020-05-11 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  SCAomni                0.0.1.8    2021-02-21 [2] Bioconductor                         </span>
<span class="co">#&gt;  scattermore            0.7        2020-11-24 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  scde                   2.18.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  sctransform            0.3.2      2020-12-16 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  sessioninfo            1.1.1      2018-11-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  Seurat               * 4.0.3      2021-06-10 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  SeuratObject         * 4.0.2      2021-06-09 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  shape                  1.4.5      2020-09-13 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  shiny                  1.6.0      2021-01-25 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  SIMLR                  1.16.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  SingleCellExperiment   1.12.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  slam                   0.1-48     2020-12-03 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  sna                    2.6        2020-10-06 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  SparseM                1.81       2021-02-18 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  sparsesvd              0.2        2019-07-15 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  spatstat.core          2.0-0      2021-03-23 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  spatstat.data          2.1-0      2021-03-21 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  spatstat.geom          2.0-1      2021-03-22 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  spatstat.sparse        2.0-0      2021-03-16 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  spatstat.utils         2.1-0      2021-03-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  statnet.common         4.4.1      2020-10-03 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  stringi                1.5.3      2020-09-09 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  stringr                1.4.0      2019-02-10 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  SummarizedExperiment   1.20.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  survival               3.2-7      2020-09-28 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  svglite                2.0.0      2021-02-20 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  systemfonts            1.0.1      2021-02-09 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  tensor                 1.5        2012-05-05 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  textshaping            0.3.0      2021-02-10 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  tibble               * 3.1.2      2021-05-16 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  tidyr                  1.1.3      2021-03-03 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  tidyselect             1.1.0      2020-05-11 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  utf8                   1.1.4      2018-05-24 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  uwot                   0.1.10     2020-12-15 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  V8                     3.4.0      2020-11-04 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  vctrs                  0.3.8      2021-04-29 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  VGAM                   1.1-5      2021-01-14 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  viridis                0.5.1      2018-03-29 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  viridisLite            0.3.0      2018-02-01 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  withr                  2.4.1      2021-01-26 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  xfun                   0.21       2021-02-10 [2] CRAN (R 4.0.3)                       </span>
<span class="co">#&gt;  XML                    3.99-0.5   2020-07-23 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  xml2                   1.3.2      2020-04-23 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  XVector                0.30.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  yaml                   2.2.1      2020-02-01 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt;  zlibbioc               1.36.0     2020-10-27 [2] Bioconductor                         </span>
<span class="co">#&gt;  zoo                    1.8-8      2020-05-02 [2] CRAN (R 4.0.2)                       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [1] /tmp/RtmpMLkNvD/temp_libpath188f24a0ff353</span>
<span class="co">#&gt; [2] /home/dbdimitrov/R/x86_64-pc-linux-gnu-library/4.0</span>
<span class="co">#&gt; [3] /usr/local/lib/R/site-library</span>
<span class="co">#&gt; [4] /usr/lib/R/site-library</span>
<span class="co">#&gt; [5] /usr/lib/R/library</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
