<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>LIANA Tutorial • liana</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="LIANA Tutorial">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="liana_tutorial_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>LIANA Tutorial</h1>
                        <h4 class="author">Daniel Dimitrov</h4>
            <address class="author_afil">
      Saezlab, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:daniel.dimitrov@uni-heidelberg.de" class="email">daniel.dimitrov@uni-heidelberg.de</a>
      </address>
                  
            <h4 class="date">15/06/2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/master/vignettes/liana_tutorial.Rmd"><code>vignettes/liana_tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>liana_tutorial.Rmd</code></div>

    </div>

    
    
<div id="liana-intro-and-tutorial" class="section level1">
<h1 class="hasAnchor">
<a href="#liana-intro-and-tutorial" class="anchor"></a><code>liana</code>: Intro and Tutorial</h1>
<p>The continuous developments of single-cell RNA-Seq (scRNA-Seq) have sparked an immense interest in understanding intercellular crosstalk. Multiple tools and resources that aid the investigation of cell-cell communication (CCC) were published recently. However, these methods and resources are usually in a fixed combination of a tool and its corresponding resource, but in principle any resource could be combined with any method.</p>
<p>To this end we built <code>liana</code> - a framework to decouple the tools from their corresponding resources.</p>
<p><img src="../ligrec_pipe.png"></p>
<div id="ccc-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#ccc-resources" class="anchor"></a>CCC Resources</h2>
<p><code>liana</code> provides CCC resources obtained and formatted via <a href="https://github.com/saezlab/OmnipathR"><code>OmnipathR</code></a> which are then converted to the appropriate format to each method.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Resource currently included in OmniPathR (and hence `liana`) include:</span>
<span class="fu"><a href="../reference/get_resources.html">get_resources</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Reshuffled"       "Default"          "CellChatDB"       "CellPhoneDB"     </span>
<span class="co">#&gt;  [5] "Ramilowski2015"   "Baccin2019"       "LRdb"             "Kirouac2010"     </span>
<span class="co">#&gt;  [9] "ICELLNET"         "iTALK"            "EMBRACE"          "HPMR"            </span>
<span class="co">#&gt; [13] "Guide2Pharma"     "connectomeDB2020" "talklr"           "CellTalkDB"      </span>
<span class="co">#&gt; [17] "OmniPath"</span>

<span class="co"># A list of resources can be obtained using the `select_resource()` function:</span>
<span class="fu"><a href="../reference/select_resource.html">select_resource</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'OmniPath'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; List of 1</span>
<span class="co">#&gt;  $ OmniPath:</span>
<span class="co">#&gt; Registered S3 method overwritten by 'cli':</span>
<span class="co">#&gt;   method     from    </span>
<span class="co">#&gt;   print.boxx spatstat</span>
<span class="co">#&gt; tibble [6,103 × 48] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;   ..$ category_intercell_source                     : chr [1:6103] "activating_cofactor" "activating_cofactor" "activating_cofactor" "activating_cofactor" ...</span>
<span class="co">#&gt;   ..$ parent_intercell_source                       : chr [1:6103] "receptor_regulator" "receptor_regulator" "receptor_regulator" "receptor_regulator" ...</span>
<span class="co">#&gt;   ..$ source                                        : chr [1:6103] "O14786" "O14786" "O60462" "O60462" ...</span>
<span class="co">#&gt;   ..$ target                                        : chr [1:6103] "P35968" "Q9UIW2" "O95754" "P35968" ...</span>
<span class="co">#&gt;   ..$ category_intercell_target                     : chr [1:6103] "receptor" "receptor" "receptor" "receptor" ...</span>
<span class="co">#&gt;   ..$ parent_intercell_target                       : chr [1:6103] "receptor" "receptor" "receptor" "receptor" ...</span>
<span class="co">#&gt;   ..$ target_genesymbol                             : chr [1:6103] "KDR" "PLXNA1" "SEMA4F" "KDR" ...</span>
<span class="co">#&gt;   ..$ source_genesymbol                             : chr [1:6103] "NRP1" "NRP1" "NRP2" "NRP2" ...</span>
<span class="co">#&gt;   ..$ is_directed                                   : num [1:6103] 1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="co">#&gt;   ..$ is_stimulation                                : num [1:6103] 1 0 0 1 1 0 1 1 1 1 ...</span>
<span class="co">#&gt;   ..$ is_inhibition                                 : num [1:6103] 0 1 0 0 1 0 0 1 0 0 ...</span>
<span class="co">#&gt;   ..$ consensus_direction                           : num [1:6103] 1 1 0 1 1 0 1 1 1 1 ...</span>
<span class="co">#&gt;   ..$ consensus_stimulation                         : num [1:6103] 1 0 0 1 1 0 1 1 1 1 ...</span>
<span class="co">#&gt;   ..$ consensus_inhibition                          : num [1:6103] 0 1 0 0 0 0 0 0 0 0 ...</span>
<span class="co">#&gt;   ..$ dip_url                                       : chr [1:6103] NA NA NA NA ...</span>
<span class="co">#&gt;   ..$ omnipath                                      : logi [1:6103] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;   ..$ kinaseextra                                   : logi [1:6103] FALSE FALSE FALSE FALSE TRUE TRUE ...</span>
<span class="co">#&gt;   ..$ pathwayextra                                  : logi [1:6103] TRUE TRUE FALSE FALSE TRUE FALSE ...</span>
<span class="co">#&gt;   ..$ ligrecextra                                   : logi [1:6103] TRUE FALSE TRUE TRUE TRUE FALSE ...</span>
<span class="co">#&gt;   ..$ sources                                       : chr [1:6103] "CellChatDB;HPRD;Wang" "HPMR;HPRD;Wang" "Baccin2019;HPRD;Ramilowski2015_Baccin2019" "CellChatDB" ...</span>
<span class="co">#&gt;   ..$ references                                    : chr [1:6103] "CellChatDB:16633338;HPRD:11333271;HPRD:16763549;HPRD:17222790" "HPMR:10520994;HPMR:15121181;HPRD:10520994" "HPRD:9856463" "CellChatDB:16633338" ...</span>
<span class="co">#&gt;   ..$ curation_effort                               : num [1:6103] 4 3 1 1 21 4 2 2 1 4 ...</span>
<span class="co">#&gt;   ..$ n_references                                  : int [1:6103] 4 2 1 1 19 3 2 2 1 2 ...</span>
<span class="co">#&gt;   ..$ n_resources                                   : int [1:6103] 3 3 2 1 6 4 2 3 1 6 ...</span>
<span class="co">#&gt;   ..$ database_intercell_source                     : chr [1:6103] "CellChatDB" "CellChatDB" "CellChatDB" "CellChatDB" ...</span>
<span class="co">#&gt;   ..$ scope_intercell_source                        : chr [1:6103] "generic" "generic" "generic" "generic" ...</span>
<span class="co">#&gt;   ..$ aspect_intercell_source                       : chr [1:6103] "functional" "functional" "functional" "functional" ...</span>
<span class="co">#&gt;   ..$ category_source_intercell_source              : chr [1:6103] "resource_specific" "resource_specific" "resource_specific" "resource_specific" ...</span>
<span class="co">#&gt;   ..$ genesymbol_intercell_source                   : chr [1:6103] "NRP1" "NRP1" "NRP2" "NRP2" ...</span>
<span class="co">#&gt;   ..$ entity_type_intercell_source                  : chr [1:6103] "protein" "protein" "protein" "protein" ...</span>
<span class="co">#&gt;   ..$ consensus_score_intercell_source              : num [1:6103] 0 0 0 0 0 0 0 0 0 0 ...</span>
<span class="co">#&gt;   ..$ transmitter_intercell_source                  : logi [1:6103] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;   ..$ receiver_intercell_source                     : logi [1:6103] FALSE FALSE FALSE FALSE FALSE FALSE ...</span>
<span class="co">#&gt;   ..$ database_intercell_target                     : chr [1:6103] "talklr;connectomeDB2020;iTALK;Almen2009;CellCellInteractions;HGNC;CellPhoneDB;HPMR;ICELLNET;CellChatDB;CellTalk"| __truncated__ "talklr;connectomeDB2020;iTALK;Almen2009;CellCellInteractions;EMBRACE;HGNC;HPMR;CellChatDB;CellTalkDB;Surfaceome"| __truncated__ "CellCellInteractions;Baccin2019;OmniPath" "talklr;connectomeDB2020;iTALK;Almen2009;CellCellInteractions;HGNC;CellPhoneDB;HPMR;ICELLNET;CellChatDB;CellTalk"| __truncated__ ...</span>
<span class="co">#&gt;   ..$ scope_intercell_target                        : chr [1:6103] "generic" "generic" "generic" "generic" ...</span>
<span class="co">#&gt;   ..$ aspect_intercell_target                       : chr [1:6103] "functional" "functional" "functional" "functional" ...</span>
<span class="co">#&gt;   ..$ category_source_intercell_target              : chr [1:6103] "resource_specific" "resource_specific" "resource_specific" "resource_specific" ...</span>
<span class="co">#&gt;   ..$ genesymbol_intercell_target                   : chr [1:6103] "KDR" "PLXNA1" "SEMA4F" "KDR" ...</span>
<span class="co">#&gt;   ..$ entity_type_intercell_target                  : chr [1:6103] "protein" "protein" "protein" "protein" ...</span>
<span class="co">#&gt;   ..$ consensus_score_intercell_target              : num [1:6103] 17 14 2 17 18 3 17 15 3 17 ...</span>
<span class="co">#&gt;   ..$ transmitter_intercell_target                  : logi [1:6103] FALSE FALSE FALSE FALSE FALSE FALSE ...</span>
<span class="co">#&gt;   ..$ receiver_intercell_target                     : logi [1:6103] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;   ..$ secreted_intercell_source                     : logi [1:6103] TRUE TRUE TRUE TRUE FALSE FALSE ...</span>
<span class="co">#&gt;   ..$ plasma_membrane_transmembrane_intercell_source: logi [1:6103] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;   ..$ plasma_membrane_peripheral_intercell_source   : logi [1:6103] FALSE FALSE FALSE FALSE FALSE FALSE ...</span>
<span class="co">#&gt;   ..$ secreted_intercell_target                     : logi [1:6103] TRUE FALSE FALSE TRUE FALSE TRUE ...</span>
<span class="co">#&gt;   ..$ plasma_membrane_transmembrane_intercell_target: logi [1:6103] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;   ..$ plasma_membrane_peripheral_intercell_target   : logi [1:6103] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></code></pre></div>
</div>
<div id="ccc-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#ccc-methods" class="anchor"></a>CCC Methods</h2>
<p>Each of the resources, can then be run with any of the following methods: * CellPhoneDB algorithm (via Squidpy) * CellChat * NATMI * Connectome * SingleCellSignalR (SCA) * iTALK</p>
</div>
<div id="calling-the-liana-wrapper-function" class="section level2">
<h2 class="hasAnchor">
<a href="#calling-the-liana-wrapper-function" class="anchor"></a>Calling the <code>liana</code> wrapper function</h2>
<p>To run <code>liana</code>, we will use a toy PBMCs scRNA-Seq data set <code>liana</code> takes <code>Seurat</code> objects as input, containing processed counts and clustered cells</p>
<pre><code>#&gt; Formal class 'Seurat' [package "Seurat"] with 13 slots
#&gt;   ..@ assays      :List of 1
#&gt;   .. ..$ RNA:Formal class 'Assay' [package "Seurat"] with 8 slots
#&gt;   ..@ meta.data   :'data.frame': 90 obs. of  4 variables:
#&gt;   .. ..$ orig.ident        : Factor w/ 1 level "pbmc3k": 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   .. ..$ nCount_RNA        : num [1:90] 4903 3914 4973 3281 2641 ...
#&gt;   .. ..$ nFeature_RNA      : int [1:90] 1352 1112 1445 1015 928 937 899 1713 960 888 ...
#&gt;   .. ..$ seurat_annotations: Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 1 3 3 1 1 1 1 ...
#&gt;   ..@ active.assay: chr "RNA"
#&gt;   ..@ active.ident: Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 1 3 3 1 1 1 1 ...
#&gt;   .. ..- attr(*, "names")= chr [1:90] "AAACATTGAGCTAC" "AAACTTGAAAAACG" "AAAGGCCTGTCTAG" "AAAGTTTGGGGTGA" ...
#&gt;   ..@ graphs      : list()
#&gt;   ..@ neighbors   : list()
#&gt;   ..@ reductions  : list()
#&gt;   ..@ images      : list()
#&gt;   ..@ project.name: chr "SeuratProject"
#&gt;   ..@ misc        : list()
#&gt;   ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1
#&gt;   .. ..$ : int [1:3] 3 2 3
#&gt;   ..@ commands    :List of 1
#&gt;   .. ..$ FindVariableFeatures.RNA:Formal class 'SeuratCommand' [package "Seurat"] with 5 slots
#&gt;   ..@ tools       : list()</code></pre>
<p><code>liana_wrap</code> calls a number of methods and and each method is run with the provided resources.</p>
<p>Here, we will call SingleCellSignalR (SCA), Connectome, and Squidpy for the sake of computational speed, but any of c(“natmi”, “italk”, “cellchat”) can also be run by simply adding their name to the <code>method</code> parameter</p>
<p>Similarly, here we use only the “OmniPath” CCC resource, but any of the aforementioned ones (available via <code><a href="../reference/get_resources.html">get_resources()</a></code>) can be added to the <code>resource</code> parameter</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run liana</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>liana_test <span class="ot">&lt;-</span> <span class="fu">liana_wrap</span>(testdata,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="fu">c</span>(<span class="st">'sca'</span>, <span class="st">'squidpy'</span>, <span class="st">'connectome'</span>),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">resource =</span> <span class="fu">c</span>(<span class="st">'OmniPath'</span>))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; No such file as table_dge_B.txt in the cluster-analysis folder</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; No such file as table_dge_CD8 T.txt in the cluster-analysis folder</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; No such file as table_dge_NK.txt in the cluster-analysis folder</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Paracrine signaling: </span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Checking for signaling between cell types</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 70 interactions from B to CD8 T</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 82 interactions from B to NK</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 44 interactions from CD8 T to B</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 63 interactions from CD8 T to NK</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 55 interactions from NK to B</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 57 interactions from NK to CD8 T</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Doing B and CD8 T ... OK</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Doing B and NK ... OK</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Doing CD8 T and B ... OK</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Doing CD8 T and NK ... OK</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Doing NK and B ... OK</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Doing NK and CD8 T ... OK</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================</span>                                               <span class="er">|</span>  <span class="dv">33</span>%</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================</span>                       <span class="er">|</span>  <span class="dv">67</span>%</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="co"># liana returns a list of results for each method</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>liana_test <span class="sc">%&gt;%</span> glimpse</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ sca       : tibble [371 × 5] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ source  : chr [1:371] "B" "B" "B" "B" ...</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ ligand  : chr [1:371] "B2M" "B2M" "B2M" "B2M" ...</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ target  : chr [1:371] "CD8 T" "CD8 T" "CD8 T" "CD8 T" ...</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ receptor: chr [1:371] "KLRD1" "KLRC1" "LILRB1" "TFRC" ...</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ LRscore : num [1:371] 0.926 0.921 0.87 0.826 0.826 ...</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ squidpy   : tibble [9,495 × 8] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ ligand        : chr [1:9495] "NRP2" "NRP2" "NRP2" "NRP2" ...</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ receptor      : chr [1:9495] "SEMA4F" "SEMA4F" "SEMA4F" "SEMA4F" ...</span></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ source        : chr [1:9495] "B" "B" "B" "CD8 T" ...</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ target        : chr [1:9495] "B" "CD8 T" "NK" "B" ...</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ means         : num [1:9495] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ pvalue        : num [1:9495] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ uniprot_source: chr [1:9495] "O60462" "O60462" "O60462" "O60462" ...</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ unprot_target : chr [1:9495] "O95754" "O95754" "O95754" "O95754" ...</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ connectome: tibble [9,495 × 8] (S3: tbl_df/tbl/data.frame)</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ source       : chr [1:9495] "NK" "CD8 T" "B" "NK" ...</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ target       : chr [1:9495] "B" "B" "B" "CD8 T" ...</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ ligand       : chr [1:9495] "BMPR2" "BMPR2" "BMPR2" "BMPR2" ...</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ receptor     : chr [1:9495] "ACVR1" "ACVR1" "ACVR1" "ACVR1" ...</span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ weight_norm  : num [1:9495] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ weight_sc    : num [1:9495] 0.0375 0.0375 0.0375 -0.075 -0.075 ...</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ p_val_adj.lig: num [1:9495] NA NA NA NA NA NA NA NA NA 1 ...</span></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ p_val_adj.rec: num [1:9495] 1 1 1 1 1 1 1 1 1 NA ...</span></span></code></pre></div>
</div>
<div id="obtain-consensus-ranks" class="section level2">
<h2 class="hasAnchor">
<a href="#obtain-consensus-ranks" class="anchor"></a>Obtain Consensus Ranks</h2>
<p><code>liana</code> also provides consensus ranks for the results obtained using different methods. By default, liana will provide mean, median, and a aggregate* consensus ranks</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run liana</span>
<span class="va">liana_test</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 9,495 x 13</span>
<span class="co">#&gt;    source ligand target receptor aggregate_rank mean_rank median_rank</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;             &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 B      LGALS9 NK     HAVCR2       0.00000247      59            87</span>
<span class="co">#&gt;  2 NK     PTPRC  B      CD22         0.00000394      79.3         101</span>
<span class="co">#&gt;  3 B      FCER2  CD8 T  CR2          0.0000181      112.          161</span>
<span class="co">#&gt;  4 NK     HLA-B  CD8 T  CD8A         0.0000440     3172.           21</span>
<span class="co">#&gt;  5 NK     B2M    NK     CD247        0.0000440     3168.            7</span>
<span class="co">#&gt;  6 NK     CCL3   CD8 T  CCR5         0.0000497      108            81</span>
<span class="co">#&gt;  7 NK     SELL   B      PODXL2       0.0000535      113            90</span>
<span class="co">#&gt;  8 NK     PDGFD  CD8 T  EPHA1        0.0000721      155           190</span>
<span class="co">#&gt;  9 B      COL18… NK     ITGA5        0.0000737      155           190</span>
<span class="co">#&gt; 10 NK     PDGFD  B      LTK          0.0000753      155           190</span>
<span class="co">#&gt; # … with 9,485 more rows, and 6 more variables: sca.LRscore &lt;dbl&gt;,</span>
<span class="co">#&gt; #   sca.rank &lt;dbl&gt;, squidpy.pvalue &lt;dbl&gt;, squidpy.rank &lt;dbl&gt;,</span>
<span class="co">#&gt; #   connectome.weight_sc &lt;dbl&gt;, connectome.rank &lt;dbl&gt;</span></code></pre></div>
<p>*The aggregate consnesus rank is obtained using the RobustRankAggreg package. The RobustRankAggreg package provides different functions to obtain aggregate ranks, by default the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3278763/"><code>RRA</code></a> method is used. RRA provides rank scores which can be interpreted as p-values. Interactions which ranked consistently higher than random are assigned low scores/p-values.</p>
</div>
<div id="call-liana-with-overwritten-default-settings" class="section level2">
<h2 class="hasAnchor">
<a href="#call-liana-with-overwritten-default-settings" class="anchor"></a>Call <code>liana</code> with overwritten default settings</h2>
<p>In the previous example <code>liana</code> was run with the default for each method which can be obtained via <code>liana_default()</code></p>
<p>Alternatively, one can also overwrite the default settings by simply passing them to the liana wrapper function</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Overwrite SCA default parameters by providing a list of parameters for the method</span>
<span class="co"># Information about the parameters passed to each method can be obtained via</span>
<span class="co"># `call_*` functions - e.g. ?call_sca or ?call_squidpy</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">testdata</span>,
                         method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'sca'</span>, <span class="st">'squidpy'</span><span class="op">)</span>,
                         resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'OmniPath'</span><span class="op">)</span>,
                         sca.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
                           assay <span class="op">=</span> <span class="st">"RNA"</span>,
                           .format <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># don't format according to liana</span>
                           s.score <span class="op">=</span> <span class="fl">0</span>,
                           logFC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; No such file as table_dge_B.txt in the cluster-analysis folder</span>
<span class="co">#&gt; No such file as table_dge_CD8 T.txt in the cluster-analysis folder</span>
<span class="co">#&gt; No such file as table_dge_NK.txt in the cluster-analysis folder</span>
<span class="co">#&gt; Paracrine signaling: </span>
<span class="co">#&gt; Checking for signaling between cell types</span>
<span class="co">#&gt; 70 interactions from B to CD8 T</span>
<span class="co">#&gt; 82 interactions from B to NK</span>
<span class="co">#&gt; 44 interactions from CD8 T to B</span>
<span class="co">#&gt; 63 interactions from CD8 T to NK</span>
<span class="co">#&gt; 55 interactions from NK to B</span>
<span class="co">#&gt; 57 interactions from NK to CD8 T</span>
<span class="co">#&gt; Doing B and CD8 T ... OK</span>
<span class="co">#&gt; Doing B and NK ... OK</span>
<span class="co">#&gt; Doing CD8 T and B ... OK</span>
<span class="co">#&gt; Doing CD8 T and NK ... OK</span>
<span class="co">#&gt; Doing NK and B ... OK</span>
<span class="co">#&gt; Doing NK and CD8 T ... OK</span>

<span class="co"># This returns a list of results for each method</span>
<span class="va">liana_test</span> <span class="op">%&gt;%</span> <span class="va">glimpse</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ sca    :List of 2</span>
<span class="co">#&gt;   ..$ individual-networks:List of 6</span>
<span class="co">#&gt;   .. ..$ B-CD8 T :'data.frame':  70 obs. of  8 variables:</span>
<span class="co">#&gt;   .. ..$ B-NK    :'data.frame':  82 obs. of  8 variables:</span>
<span class="co">#&gt;   .. ..$ CD8 T-B :'data.frame':  44 obs. of  8 variables:</span>
<span class="co">#&gt;   .. ..$ CD8 T-NK:'data.frame':  63 obs. of  8 variables:</span>
<span class="co">#&gt;   .. ..$ NK-B    :'data.frame':  55 obs. of  8 variables:</span>
<span class="co">#&gt;   .. ..$ NK-CD8 T:'data.frame':  57 obs. of  8 variables:</span>
<span class="co">#&gt;   ..$ full-network       :'data.frame':  371 obs. of  4 variables:</span>
<span class="co">#&gt;   .. ..$ ligand          : chr [1:371] "B.B2M" "B.B2M" "B.B2M" "B.B2M" ...</span>
<span class="co">#&gt;   .. ..$ receptor        : chr [1:371] "CD8 T.KLRD1" "CD8 T.KLRC1" "CD8 T.LILRB1" "CD8 T.TFRC" ...</span>
<span class="co">#&gt;   .. ..$ interaction.type: chr [1:371] "paracrine" "paracrine" "paracrine" "paracrine" ...</span>
<span class="co">#&gt;   .. ..$ LRscore         : num [1:371] 0.926 0.921 0.87 0.826 0.826 ...</span>
<span class="co">#&gt;  $ squidpy: tibble [9,495 × 8] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;   ..$ ligand        : chr [1:9495] "NRP2" "NRP2" "NRP2" "NRP2" ...</span>
<span class="co">#&gt;   ..$ receptor      : chr [1:9495] "SEMA4F" "SEMA4F" "SEMA4F" "SEMA4F" ...</span>
<span class="co">#&gt;   ..$ source        : chr [1:9495] "B" "B" "B" "CD8 T" ...</span>
<span class="co">#&gt;   ..$ target        : chr [1:9495] "B" "CD8 T" "NK" "B" ...</span>
<span class="co">#&gt;   ..$ means         : num [1:9495] 0 0 0 0 0 0 0 0 0 0 ...</span>
<span class="co">#&gt;   ..$ pvalue        : num [1:9495] NaN NaN NaN NaN NaN NaN NaN NaN NaN NaN ...</span>
<span class="co">#&gt;   ..$ uniprot_source: chr [1:9495] "O60462" "O60462" "O60462" "O60462" ...</span>
<span class="co">#&gt;   ..$ unprot_target : chr [1:9495] "O95754" "O95754" "O95754" "O95754" ...</span></code></pre></div>
</div>
<div id="call-a-single-method" class="section level2">
<h2 class="hasAnchor">
<a href="#call-a-single-method" class="anchor"></a>Call a single method</h2>
<p>To call a any single method, you can use any of the <code>call_*</code> methods:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># RUN cellchat alone with OmniPath</span>
<span class="va">cc_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/call_cellchat.html">call_cellchat</a></span><span class="op">(</span>
  seurat_object <span class="op">=</span> <span class="va">testdata</span>,
  op_resource <span class="op">=</span> <span class="fu"><a href="../reference/select_resource.html">select_resource</a></span><span class="op">(</span><span class="st">'OmniPath'</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  .normalize <span class="op">=</span> <span class="cn">TRUE</span>,
  nboot <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Create a CellChat object from a data matrix</span>
<span class="co">#&gt; Set cell identities for the new CellChat object</span>
<span class="co">#&gt; The cell groups used for CellChat analysis are  B CD8 T NK</span>
<span class="co">#&gt; Warning: Expected 5 pieces. Missing pieces filled with `NA` in 3 rows [1, 2, 3].</span>
<span class="co">#&gt; Issue identified!! Please check the official Gene Symbol of the following genes:  </span>
<span class="co">#&gt;  2 </span>
<span class="co">#&gt; Issue identified!! Please check the official Gene Symbol of the following genes:  </span>
<span class="co">#&gt;  CYR61 WISP1 CTGF FAM19A4 CD97</span>

<span class="co"># Show CellChat Results</span>
<span class="va">cc_res</span> <span class="op">%&gt;%</span> <span class="va">glimpse</span>
<span class="co">#&gt; Rows: 70</span>
<span class="co">#&gt; Columns: 6</span>
<span class="co">#&gt; $ source   &lt;fct&gt; NK, B, CD8 T, NK, B, CD8 T, NK, B, CD8 T, NK, B, CD8 T, NK, …</span>
<span class="co">#&gt; $ target   &lt;fct&gt; NK, CD8 T, CD8 T, CD8 T, CD8 T, CD8 T, CD8 T, NK, NK, NK, CD…</span>
<span class="co">#&gt; $ ligand   &lt;chr&gt; "TYROBP", "HLA-B", "HLA-B", "HLA-B", "HLA-B", "HLA-B", "HLA-…</span>
<span class="co">#&gt; $ receptor &lt;chr&gt; "KLRD1", "CD8A", "CD8A", "CD8A", "CD3D", "CD3D", "CD3D", "KL…</span>
<span class="co">#&gt; $ prob     &lt;dbl&gt; 0.144223902, 0.154420489, 0.169362205, 0.182686103, 0.265527…</span>
<span class="co">#&gt; $ pval     &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …</span></code></pre></div>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h2>
<p>We appreciate any feedback, so please do not hesitate to open an issue to the <a href="https://github.com/saezlab/liana">liana github page</a>!</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"liana"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To cite liana in publications use:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Dimitrov, D., Türei, D., Boys, C., Nagai, J.S., Flores, R.O.R., Kim,</span>
<span class="co">#&gt;   H., Szalai, B., Costa, I.G., Dugourd, A., Valdeolivas, A. and</span>
<span class="co">#&gt;   Saez-Rodriguez, J., 2021.  Comparison of Resources and Methods to</span>
<span class="co">#&gt;   infer Cell-Cell Communication from Single-cell RNA Data. bioRxiv.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{,</span>
<span class="co">#&gt;     author = {Daniel Dimitrov and Charlotte Boys and Denes Turei and James Nagai S. and Ricardo Ramirez Flores O. and Hyojin Kim and Bence Szalai and Ivan Costa G. and Aurélien Dugourd and Alberto Valdeolivas and Julio Saez-Rodriguez},</span>
<span class="co">#&gt;     title = {Comparison of Resources and Methods to infer Cell-Cell Communication from Single-cell RNA Data},</span>
<span class="co">#&gt;     journal = {BioRxiv},</span>
<span class="co">#&gt;     year = {2021},</span>
<span class="co">#&gt;     doi = {10.1101/2021.05.21.445160},</span>
<span class="co">#&gt;     encoding = {UTF-8},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.0.3 (2020-10-10)
#&gt;  os       Ubuntu 20.04.2 LTS          
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       Europe/Berlin               
#&gt;  date     2021-06-15                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────
#&gt;  package              * version    date       lib
#&gt;  abind                  1.4-5      2016-07-21 [2]
#&gt;  annotate               1.68.0     2020-10-27 [2]
#&gt;  AnnotationDbi          1.52.0     2020-10-27 [2]
#&gt;  assertthat             0.2.1      2019-03-21 [2]
#&gt;  backports              1.2.1      2020-12-09 [2]
#&gt;  bbmle                  1.0.23.1   2020-02-03 [2]
#&gt;  bdsmatrix              1.3-4      2020-01-13 [2]
#&gt;  Biobase              * 2.50.0     2020-10-27 [2]
#&gt;  BiocGenerics         * 0.36.0     2020-10-27 [2]
#&gt;  BiocParallel           1.24.1     2020-11-06 [2]
#&gt;  bit                    4.0.4      2020-08-04 [2]
#&gt;  bit64                  4.0.5      2020-08-30 [2]
#&gt;  bitops                 1.0-6      2013-08-17 [2]
#&gt;  blob                   1.2.1      2020-01-20 [2]
#&gt;  brew                   1.0-6      2011-04-13 [2]
#&gt;  bslib                  0.2.4      2021-01-25 [2]
#&gt;  cachem                 1.0.4      2021-02-13 [2]
#&gt;  Cairo                  1.5-12.2   2020-07-07 [2]
#&gt;  caTools                1.18.1     2021-01-11 [2]
#&gt;  CellChat               1.1.0      2021-05-02 [2]
#&gt;  cellranger             1.1.0      2016-07-27 [2]
#&gt;  checkmate              2.0.0      2020-02-06 [2]
#&gt;  circlize               0.4.12     2021-01-08 [2]
#&gt;  cli                    2.3.0      2021-01-31 [2]
#&gt;  clue                   0.3-58     2020-12-03 [2]
#&gt;  cluster                2.1.0      2019-06-19 [5]
#&gt;  coda                   0.19-4     2020-09-30 [2]
#&gt;  codetools              0.2-16     2018-12-24 [5]
#&gt;  colorspace             2.0-0      2020-11-11 [2]
#&gt;  combinat               0.0-8      2012-10-29 [2]
#&gt;  ComplexHeatmap         2.6.2      2020-11-12 [2]
#&gt;  Connectome             1.0.1      2021-02-21 [2]
#&gt;  conquer                1.0.2      2020-08-27 [2]
#&gt;  cowplot                1.1.1      2020-12-30 [2]
#&gt;  crayon                 1.4.1      2021-02-08 [2]
#&gt;  curl                   4.3        2019-12-02 [2]
#&gt;  data.table             1.14.0     2021-02-21 [2]
#&gt;  DBI                    1.1.1      2021-01-15 [2]
#&gt;  DDRTree                0.1.5      2017-04-30 [2]
#&gt;  DelayedArray           0.16.1     2021-01-22 [2]
#&gt;  deldir                 0.2-10     2021-02-16 [2]
#&gt;  densityClust           0.3        2017-10-24 [2]
#&gt;  desc                   1.2.0      2018-05-01 [2]
#&gt;  DESeq2                 1.30.1     2021-02-19 [2]
#&gt;  DEsingle               1.10.0     2020-10-27 [2]
#&gt;  digest                 0.6.27     2020-10-24 [2]
#&gt;  distillery             1.2        2020-11-21 [2]
#&gt;  docopt                 0.7.1      2020-06-24 [2]
#&gt;  doParallel             1.0.16     2020-10-16 [2]
#&gt;  dplyr                * 1.0.4      2021-02-02 [2]
#&gt;  edgeR                  3.32.1     2021-01-14 [2]
#&gt;  ellipsis               0.3.1      2020-05-15 [2]
#&gt;  evaluate               0.14       2019-05-28 [2]
#&gt;  extRemes               2.1        2020-11-24 [2]
#&gt;  fansi                  0.4.2      2021-01-15 [2]
#&gt;  fastICA                1.2-2      2019-07-08 [2]
#&gt;  fastmap                1.1.0      2021-01-25 [2]
#&gt;  fitdistrplus           1.1-3      2020-12-05 [2]
#&gt;  flexmix                2.3-17     2020-10-12 [2]
#&gt;  FNN                    1.1.3      2019-02-15 [2]
#&gt;  foreach                1.5.1      2020-10-15 [2]
#&gt;  fs                     1.5.0      2020-07-31 [2]
#&gt;  future                 1.21.0     2020-12-10 [2]
#&gt;  future.apply           1.7.0      2021-01-04 [2]
#&gt;  gamlss                 5.2-0      2020-09-12 [2]
#&gt;  gamlss.data            5.1-4      2019-05-15 [2]
#&gt;  gamlss.dist            5.1-7      2020-07-13 [2]
#&gt;  genefilter             1.72.1     2021-01-21 [2]
#&gt;  geneplotter            1.68.0     2020-10-27 [2]
#&gt;  generics               0.1.0      2020-10-31 [2]
#&gt;  GenomeInfoDb           1.26.2     2020-12-08 [2]
#&gt;  GenomeInfoDbData       1.2.4      2020-12-14 [2]
#&gt;  GenomicRanges          1.42.0     2020-10-27 [2]
#&gt;  GetoptLong             1.0.5      2020-12-15 [2]
#&gt;  gg.gap                 1.3        2019-09-30 [2]
#&gt;  ggalluvial             0.12.3     2020-12-05 [2]
#&gt;  ggplot2                3.3.3      2020-12-30 [2]
#&gt;  ggrepel                0.9.1      2021-01-15 [2]
#&gt;  ggridges               0.5.3      2021-01-08 [2]
#&gt;  GlobalOptions          0.1.2      2020-06-10 [2]
#&gt;  globals                0.14.0     2020-11-22 [2]
#&gt;  glue                   1.4.2      2020-08-27 [2]
#&gt;  goftest                1.2-2      2019-12-02 [2]
#&gt;  gplots                 3.1.1      2020-11-28 [2]
#&gt;  gridBase               0.4-7      2014-02-24 [2]
#&gt;  gridExtra              2.3        2017-09-09 [2]
#&gt;  gtable                 0.3.0      2019-03-25 [2]
#&gt;  gtools                 3.8.2      2020-03-31 [2]
#&gt;  hms                    1.0.0      2021-01-13 [2]
#&gt;  HSMMSingleCell         1.10.0     2020-10-29 [2]
#&gt;  htmltools              0.5.1.1    2021-01-22 [2]
#&gt;  htmlwidgets            1.5.3      2020-12-10 [2]
#&gt;  httpuv                 1.5.5      2021-01-13 [2]
#&gt;  httr                   1.4.2      2020-07-20 [2]
#&gt;  ica                    1.0-2      2018-05-24 [2]
#&gt;  igraph                 1.2.6      2020-10-06 [2]
#&gt;  IRanges                2.24.1     2020-12-12 [2]
#&gt;  irlba                  2.3.3      2019-02-05 [2]
#&gt;  iTALK                  0.1.0      2021-02-21 [2]
#&gt;  iterators              1.0.13     2020-10-15 [2]
#&gt;  jquerylib              0.1.3      2020-12-17 [2]
#&gt;  jsonlite               1.7.2      2020-12-09 [2]
#&gt;  KernSmooth             2.23-17    2020-04-26 [5]
#&gt;  knitr                  1.31       2021-01-27 [2]
#&gt;  later                  1.1.0.1    2020-06-05 [2]
#&gt;  lattice                0.20-41    2020-04-02 [5]
#&gt;  lazyeval               0.2.2      2019-03-15 [2]
#&gt;  leiden                 0.3.7      2021-01-26 [2]
#&gt;  liana                * 0.0.1      2021-06-15 [1]
#&gt;  lifecycle              1.0.0      2021-02-15 [2]
#&gt;  limma                  3.46.0     2020-10-27 [2]
#&gt;  listenv                0.8.0      2019-12-05 [2]
#&gt;  Lmoments               1.3-1      2019-03-15 [2]
#&gt;  lmtest                 0.9-38     2020-09-09 [2]
#&gt;  locfit                 1.5-9.4    2020-03-25 [2]
#&gt;  logger                 0.2.0      2021-03-04 [2]
#&gt;  magrittr               2.0.1      2020-11-17 [2]
#&gt;  MASS                   7.3-53     2020-09-09 [5]
#&gt;  MAST                   1.16.0     2020-10-27 [2]
#&gt;  Matrix                 1.2-18     2019-11-27 [5]
#&gt;  MatrixGenerics         1.2.1      2021-01-30 [2]
#&gt;  MatrixModels           0.4-1      2015-08-22 [2]
#&gt;  matrixStats            0.58.0     2021-01-29 [2]
#&gt;  maxLik                 1.4-6      2020-11-24 [2]
#&gt;  memoise                2.0.0      2021-01-26 [2]
#&gt;  mgcv                   1.8-33     2020-08-27 [5]
#&gt;  mime                   0.10       2021-02-13 [2]
#&gt;  miniUI                 0.1.1.1    2018-05-18 [2]
#&gt;  miscTools              0.6-26     2019-12-08 [2]
#&gt;  modeltools             0.2-23     2020-03-05 [2]
#&gt;  monocle                2.18.0     2020-10-27 [2]
#&gt;  multtest               2.46.0     2020-10-27 [2]
#&gt;  munsell                0.5.0      2018-06-12 [2]
#&gt;  mvtnorm                1.1-1      2020-06-09 [2]
#&gt;  network                1.16.1     2020-10-07 [2]
#&gt;  nlme                   3.1-149    2020-08-23 [5]
#&gt;  NMF                    0.23.0     2020-08-01 [2]
#&gt;  nnet                   7.3-14     2020-04-26 [5]
#&gt;  numDeriv               2016.8-1.1 2019-06-06 [2]
#&gt;  OmnipathR              2.99.18    2021-06-09 [2]
#&gt;  parallelly             1.24.0     2021-03-14 [2]
#&gt;  patchwork              1.1.1      2020-12-17 [2]
#&gt;  pbapply                1.4-3      2020-08-18 [2]
#&gt;  pcaMethods             1.82.0     2020-10-27 [2]
#&gt;  pheatmap               1.0.12     2019-01-04 [2]
#&gt;  pillar                 1.4.7      2020-11-20 [2]
#&gt;  pkgconfig              2.0.3      2019-09-22 [2]
#&gt;  pkgdown                1.6.1      2020-09-12 [2]
#&gt;  pkgmaker               0.32.2     2020-10-20 [2]
#&gt;  plotly                 4.9.3      2021-01-10 [2]
#&gt;  plotrix              * 3.8-1      2021-01-21 [2]
#&gt;  plyr                   1.8.6      2020-03-03 [2]
#&gt;  png                    0.1-7      2013-12-03 [2]
#&gt;  polyclip               1.10-0     2019-03-14 [2]
#&gt;  pracma                 2.3.3      2021-01-23 [2]
#&gt;  prettyunits            1.1.1      2020-01-24 [2]
#&gt;  progress               1.2.2      2019-05-16 [2]
#&gt;  promises               1.2.0.1    2021-02-11 [2]
#&gt;  pscl                   1.5.5      2020-03-07 [2]
#&gt;  purrr                * 0.3.4      2020-04-17 [2]
#&gt;  qlcMatrix              0.9.7      2018-04-20 [2]
#&gt;  quantreg               5.83       2021-01-22 [2]
#&gt;  R6                     2.5.0      2020-10-28 [2]
#&gt;  ragg                   1.1.0      2021-02-15 [2]
#&gt;  randomcoloR            1.1.0.1    2019-11-24 [2]
#&gt;  RANN                   2.6.1      2019-01-08 [2]
#&gt;  rappdirs               0.3.3      2021-01-31 [2]
#&gt;  RColorBrewer           1.1-2      2014-12-07 [2]
#&gt;  Rcpp                   1.0.6      2021-01-15 [2]
#&gt;  RcppAnnoy              0.0.18     2020-12-15 [2]
#&gt;  RcppArmadillo          0.10.2.1.0 2021-02-09 [2]
#&gt;  RCurl                  1.98-1.3   2021-03-16 [2]
#&gt;  readr                  1.4.0      2020-10-05 [2]
#&gt;  readxl                 1.3.1      2019-03-13 [2]
#&gt;  registry               0.5-1      2019-03-05 [2]
#&gt;  reshape2               1.4.4      2020-04-09 [2]
#&gt;  reticulate             1.18       2020-10-25 [2]
#&gt;  rjson                  0.2.20     2018-06-08 [2]
#&gt;  rlang                  0.4.10     2020-12-30 [2]
#&gt;  rle                    0.9.2      2020-09-25 [2]
#&gt;  rmarkdown              2.7        2021-02-19 [2]
#&gt;  RMTstat                0.3        2014-11-01 [2]
#&gt;  rngtools               1.5        2020-01-23 [2]
#&gt;  RobustRankAggreg       1.1        2013-06-03 [2]
#&gt;  ROCR                   1.0-11     2020-05-02 [2]
#&gt;  Rook                   1.1-1      2014-10-20 [2]
#&gt;  rpart                  4.1-15     2019-04-12 [5]
#&gt;  rprojroot              2.0.2      2020-11-15 [2]
#&gt;  RSpectra               0.16-0     2019-12-01 [2]
#&gt;  RSQLite                2.2.3      2021-01-24 [2]
#&gt;  rsvd                   1.0.3      2020-02-17 [2]
#&gt;  Rtsne                  0.15       2018-11-10 [2]
#&gt;  S4Vectors              0.28.1     2020-12-09 [2]
#&gt;  sandwich               3.0-0      2020-10-02 [2]
#&gt;  sass                   0.3.1      2021-01-24 [2]
#&gt;  scales                 1.1.1      2020-05-11 [2]
#&gt;  SCAomni                0.0.1.8    2021-02-21 [2]
#&gt;  scattermore            0.7        2020-11-24 [2]
#&gt;  scde                   2.18.0     2020-10-27 [2]
#&gt;  sctransform            0.3.2      2020-12-16 [2]
#&gt;  sessioninfo            1.1.1      2018-11-05 [2]
#&gt;  Seurat               * 3.2.3      2020-12-15 [2]
#&gt;  shape                  1.4.5      2020-09-13 [2]
#&gt;  shiny                  1.6.0      2021-01-25 [2]
#&gt;  SIMLR                  1.16.0     2020-10-27 [2]
#&gt;  SingleCellExperiment   1.12.0     2020-10-27 [2]
#&gt;  slam                   0.1-48     2020-12-03 [2]
#&gt;  sna                    2.6        2020-10-06 [2]
#&gt;  SparseM                1.81       2021-02-18 [2]
#&gt;  sparsesvd              0.2        2019-07-15 [2]
#&gt;  spatstat               1.64-1     2020-05-12 [2]
#&gt;  spatstat.data          2.1-0      2021-03-21 [2]
#&gt;  spatstat.utils         2.1-0      2021-03-15 [2]
#&gt;  statnet.common         4.4.1      2020-10-03 [2]
#&gt;  stringi                1.5.3      2020-09-09 [2]
#&gt;  stringr                1.4.0      2019-02-10 [2]
#&gt;  SummarizedExperiment   1.20.0     2020-10-27 [2]
#&gt;  survival               3.2-7      2020-09-28 [2]
#&gt;  svglite                2.0.0      2021-02-20 [2]
#&gt;  systemfonts            1.0.1      2021-02-09 [2]
#&gt;  tensor                 1.5        2012-05-05 [2]
#&gt;  textshaping            0.3.0      2021-02-10 [2]
#&gt;  tibble               * 3.0.6      2021-01-29 [2]
#&gt;  tidyr                  1.1.2      2020-08-27 [2]
#&gt;  tidyselect             1.1.0      2020-05-11 [2]
#&gt;  utf8                   1.1.4      2018-05-24 [2]
#&gt;  uwot                   0.1.10     2020-12-15 [2]
#&gt;  V8                     3.4.0      2020-11-04 [2]
#&gt;  vctrs                  0.3.6      2020-12-17 [2]
#&gt;  VGAM                   1.1-5      2021-01-14 [2]
#&gt;  viridis                0.5.1      2018-03-29 [2]
#&gt;  viridisLite            0.3.0      2018-02-01 [2]
#&gt;  withr                  2.4.1      2021-01-26 [2]
#&gt;  workflowr            * 1.6.2      2020-04-30 [2]
#&gt;  xfun                   0.21       2021-02-10 [2]
#&gt;  XML                    3.99-0.5   2020-07-23 [2]
#&gt;  xml2                   1.3.2      2020-04-23 [2]
#&gt;  xtable                 1.8-4      2019-04-21 [2]
#&gt;  XVector                0.30.0     2020-10-27 [2]
#&gt;  yaml                   2.2.1      2020-02-01 [2]
#&gt;  zlibbioc               1.36.0     2020-10-27 [2]
#&gt;  zoo                    1.8-8      2020-05-02 [2]
#&gt;  source                               
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Github (sqjin/CellChat@b3ccf96)      
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.0)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.0)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  Github (msraredon/Connectome@4531dd9)
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Github (Coolgenome/iTALK@6d9b390)    
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.0)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.0)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  local                                
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.0)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.0)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Github (saezlab/OmnipathR@ff3ad88)   
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.0)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  Bioconductor                         
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.3)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt;  Bioconductor                         
#&gt;  CRAN (R 4.0.2)                       
#&gt; 
#&gt; [1] /tmp/RtmptC57hb/temp_libpath63b0a1967c2e9
#&gt; [2] /home/dbdimitrov/R/x86_64-pc-linux-gnu-library/4.0
#&gt; [3] /usr/local/lib/R/site-library
#&gt; [4] /usr/lib/R/site-library
#&gt; [5] /usr/lib/R/library</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
