<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>CellChat x Omni Pipe function — call_cellchat • liana</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="CellChat x Omni Pipe function — call_cellchat" />
<meta property="og:description" content="CellChat x Omni Pipe function" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CellChat x Omni Pipe function</h1>
    <small class="dont-index">Source: <a href='https://github.com/saezlab/liana/blob/master/R/cellchat_pipe.R'><code>R/cellchat_pipe.R</code></a></small>
    <div class="hidden name"><code>call_cellchat.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>CellChat x Omni Pipe function</p>
    </div>

    <pre class="usage"><span class='fu'>call_cellchat</span><span class='op'>(</span>
  <span class='va'>op_resource</span>,
  <span class='va'>seurat_object</span>,
  .format <span class='op'>=</span> <span class='cn'>TRUE</span>,
  exclude_anns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>)</span>,
  nboot <span class='op'>=</span> <span class='fl'>100</span>,
  assay <span class='op'>=</span> <span class='st'>"RNA"</span>,
  .seed <span class='op'>=</span> <span class='fl'>1004</span>,
  .normalize <span class='op'>=</span> <span class='cn'>FALSE</span>,
  .do_parallel <span class='op'>=</span> <span class='cn'>FALSE</span>,
  .raw_use <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>op_resource</th>
      <td><p>OmniPath Intercell Resource DN</p></td>
    </tr>
    <tr>
      <th>seurat_object</th>
      <td><p>Seurat object as input</p></td>
    </tr>
    <tr>
      <th>.format</th>
      <td><p>bool whether to format output</p></td>
    </tr>
    <tr>
      <th>exclude_anns</th>
      <td><p>Annotation criteria to be excluded</p></td>
    </tr>
    <tr>
      <th>nboot</th>
      <td><p>number of bootstraps to calculate p-value</p></td>
    </tr>
    <tr>
      <th>.normalize</th>
      <td><p># bool whether to normalize non-normalized data with</p></td>
    </tr>
    <tr>
      <th>.raw_use</th>
      <td><p>whether use the raw data or gene expression data pojectected
to a ppi</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed on to <code><a href='https://rdrr.io/pkg/CellChat/man/subsetCommunication.html'>CellChat::subsetCommunication</a></code></p><dl>
    <dt><code>object</code></dt><dd><p>CellChat object</p></dd>
    <dt><code>net</code></dt><dd><p>Alternative input is a data frame with at least with three columns defining the cell-cell communication network ("source","target","interaction_name")</p></dd>
    <dt><code>slot.name</code></dt><dd><p>the slot name of object: slot.name = "net" when extracting the inferred communications at the level of ligands/receptors; slot.name = "netP" when extracting the inferred communications at the level of signaling pathways</p></dd>
    <dt><code>sources.use</code></dt><dd><p>a vector giving the index or the name of source cell groups</p></dd>
    <dt><code>targets.use</code></dt><dd><p>a vector giving the index or the name of target cell groups.</p></dd>
    <dt><code>signaling</code></dt><dd><p>a character vector giving the name of signaling pathways of interest</p></dd>
    <dt><code>pairLR.use</code></dt><dd><p>a data frame consisting of one column named either "interaction_name" or "pathway_name", defining the interactions of interest</p></dd>
    <dt><code>thresh</code></dt><dd><p>threshold of the p-value for determining significant interaction</p></dd>
    <dt><code>datasets</code></dt><dd><p>select the inferred cell-cell communications from a particular `datasets` when inputing a data frame `net`</p></dd>
    <dt><code>ligand.pvalues</code></dt><dd><p>set threshold for ligand genesligand.pvalues: threshold for pvalues in the differential expression gene analysis (DEG)ligand.logFC: threshold for logFoldChange in the DEG analysis; When ligand.logFC &gt; 0, keep upgulated genes; otherwise, kepp downregulated genesligand.pct.1: threshold for the percent of expressed genes in the defined 'positive' cell group. keep genes with percent greater than ligand.pct.1ligand.pct.2: threshold for the percent of expressed genes in the cells except for the defined 'positive' cell group</p></dd>
    <dt><code>ligand.logFC</code></dt><dd><p>set threshold for ligand genesligand.pvalues: threshold for pvalues in the differential expression gene analysis (DEG)ligand.logFC: threshold for logFoldChange in the DEG analysis; When ligand.logFC &gt; 0, keep upgulated genes; otherwise, kepp downregulated genesligand.pct.1: threshold for the percent of expressed genes in the defined 'positive' cell group. keep genes with percent greater than ligand.pct.1ligand.pct.2: threshold for the percent of expressed genes in the cells except for the defined 'positive' cell group</p></dd>
    <dt><code>ligand.pct.1</code></dt><dd><p>set threshold for ligand genesligand.pvalues: threshold for pvalues in the differential expression gene analysis (DEG)ligand.logFC: threshold for logFoldChange in the DEG analysis; When ligand.logFC &gt; 0, keep upgulated genes; otherwise, kepp downregulated genesligand.pct.1: threshold for the percent of expressed genes in the defined 'positive' cell group. keep genes with percent greater than ligand.pct.1ligand.pct.2: threshold for the percent of expressed genes in the cells except for the defined 'positive' cell group</p></dd>
    <dt><code>ligand.pct.2</code></dt><dd><p>set threshold for ligand genesligand.pvalues: threshold for pvalues in the differential expression gene analysis (DEG)ligand.logFC: threshold for logFoldChange in the DEG analysis; When ligand.logFC &gt; 0, keep upgulated genes; otherwise, kepp downregulated genesligand.pct.1: threshold for the percent of expressed genes in the defined 'positive' cell group. keep genes with percent greater than ligand.pct.1ligand.pct.2: threshold for the percent of expressed genes in the cells except for the defined 'positive' cell group</p></dd>
    <dt><code>receptor.pvalues</code></dt><dd><p>set threshold for receptor genes</p></dd>
    <dt><code>receptor.logFC</code></dt><dd><p>set threshold for receptor genes</p></dd>
    <dt><code>receptor.pct.1</code></dt><dd><p>set threshold for receptor genes</p></dd>
    <dt><code>receptor.pct.2</code></dt><dd><p>set threshold for receptor genes</p></dd>
  
</dl></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A DF of intercellular communication network</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>CellChat's objects are not documented/exported thus the
  whole package has to be imported.</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


