<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>LIANA Tutorial • liana</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="LIANA Tutorial">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/liana_custom_op.html">LIANA Custome literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_intracell.html">LIANA with Intracellular Omnipath</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="liana_tutorial_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>LIANA Tutorial</h1>
                        <h4 class="author">Daniel Dimitrov</h4>
            <address class="author_afil">
      Saezlab, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:daniel.dimitrov@uni-heidelberg.de" class="email">daniel.dimitrov@uni-heidelberg.de</a>
      </address>
                  
            <h4 class="date">2022-02-07</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/master/vignettes/liana_tutorial.Rmd"><code>vignettes/liana_tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>liana_tutorial.Rmd</code></div>

    </div>

    
    
<div id="liana-intro" class="section level1">
<h1 class="hasAnchor">
<a href="#liana-intro" class="anchor"></a><code>liana</code>: Intro</h1>
<p>The continuous developments of single-cell RNA-Seq (scRNA-Seq) have sparked an immense interest in understanding intercellular crosstalk. Multiple tools and resources that aid the investigation of cell-cell communication (CCC) were published recently. However, these methods and resources are usually in a fixed combination of a tool and its corresponding resource, but in principle any resource could be combined with any method.</p>
<p>To this end, we built <code>liana</code> - a framework to decouple the tools from their corresponding resources.</p>
<p><img src="../ligrec_pipe.png"></p>
<div id="load-required-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-required-packages" class="anchor"></a>Load required packages</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://omnipathdb.org/">liana</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="ccc-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#ccc-resources" class="anchor"></a>CCC Resources</h2>
<p><code>liana</code> provides CCC resources obtained and formatted via <a href="https://github.com/saezlab/OmnipathR"><code>OmnipathR</code></a> which are then converted to the appropriate format to each method.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Resource currently included in OmniPathR (and hence `liana`) include:</span>
<span class="fu"><a href="../reference/show_resources.html">show_resources</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Default"          "Baccin2019"       "CellCall"         "CellChatDB"      </span>
<span class="co">#&gt;  [5] "Cellinker"        "CellPhoneDB"      "CellTalkDB"       "connectomeDB2020"</span>
<span class="co">#&gt;  [9] "EMBRACE"          "Guide2Pharma"     "HPMR"             "ICELLNET"        </span>
<span class="co">#&gt; [13] "iTALK"            "Kirouac2010"      "LRdb"             "Ramilowski2015"  </span>
<span class="co">#&gt; [17] "OmniPath"</span>

<span class="co"># A list of resources can be obtained using the `select_resource()` function:</span>
<span class="co"># See `?select_resource()` documentation for further information.</span>
<span class="co"># select_resource(c('OmniPath')) %&gt;% glimpse() </span></code></pre></div>
</div>
<div id="ccc-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#ccc-methods" class="anchor"></a>CCC Methods</h2>
<p>Each of the resources can then be run with any of the following methods:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Resource currently included in OmniPathR (and hence `liana`) include:</span>
<span class="fu"><a href="../reference/show_methods.html">show_methods</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "logfc"           "natmi"           "connectome"      "sca"            </span>
<span class="co">#&gt;  [5] "squidpy"         "cellchat"        "call_sca"        "call_natmi"     </span>
<span class="co">#&gt;  [9] "call_italk"      "call_connectome" "cellphonedb"</span></code></pre></div>
<p>Note that the different algorithms (or scoring measures) used in <code>sca</code>, <code>natmi</code>, <code>connectome</code>, <code>cellphonedb</code>, <code>cytotalk</code>’s crosstalk scores, and <code>logfc</code> were re-implemented in LIANA. Yet, the original method pipelines can be called via the <code>call_*</code> functions.</p>
</div>
<div id="liana-wrapper-function" class="section level2">
<h2 class="hasAnchor">
<a href="#liana-wrapper-function" class="anchor"></a><code>liana</code> wrapper function</h2>
<p>To run <code>liana</code>, we will use a toy PBMCs scRNA-Seq data set, obtained from <a href="https://github.com/satijalab/seurat-data">SeuratData</a>.</p>
<p><code>liana</code> takes <code>Seurat</code> and <code>SingleCellExperiment</code> objects as input, containing processed counts and clustered cells</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">liana_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"liana"</span><span class="op">)</span>
<span class="va">testdata</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">liana_path</span> , <span class="st">"testdata"</span>, <span class="st">"input"</span>, <span class="st">"testdata.rds"</span><span class="op">)</span><span class="op">)</span>

<span class="va">testdata</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Formal class 'Seurat' [package "SeuratObject"] with 13 slots</span>
<span class="co">#&gt;   ..@ assays      :List of 1</span>
<span class="co">#&gt;   .. ..$ RNA:Formal class 'Assay' [package "Seurat"] with 8 slots</span>
<span class="co">#&gt;   ..@ meta.data   :'data.frame': 90 obs. of  4 variables:</span>
<span class="co">#&gt;   .. ..$ orig.ident        : Factor w/ 1 level "pbmc3k": 1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="co">#&gt;   .. ..$ nCount_RNA        : num [1:90] 4903 3914 4973 3281 2641 ...</span>
<span class="co">#&gt;   .. ..$ nFeature_RNA      : int [1:90] 1352 1112 1445 1015 928 937 899 1713 960 888 ...</span>
<span class="co">#&gt;   .. ..$ seurat_annotations: Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 1 3 3 1 1 1 1 ...</span>
<span class="co">#&gt;   ..@ active.assay: chr "RNA"</span>
<span class="co">#&gt;   ..@ active.ident: Factor w/ 3 levels "B","CD8 T","NK": 1 1 1 1 3 3 1 1 1 1 ...</span>
<span class="co">#&gt;   .. ..- attr(*, "names")= chr [1:90] "AAACATTGAGCTAC" "AAACTTGAAAAACG" "AAAGGCCTGTCTAG" "AAAGTTTGGGGTGA" ...</span>
<span class="co">#&gt;   ..@ graphs      : list()</span>
<span class="co">#&gt;   ..@ neighbors   : list()</span>
<span class="co">#&gt;   ..@ reductions  : list()</span>
<span class="co">#&gt;   ..@ images      : list()</span>
<span class="co">#&gt;   ..@ project.name: chr "SeuratProject"</span>
<span class="co">#&gt;   ..@ misc        : list()</span>
<span class="co">#&gt;   ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1</span>
<span class="co">#&gt;   .. ..$ : int [1:3] 3 2 3</span>
<span class="co">#&gt;   ..@ commands    :List of 2</span>
<span class="co">#&gt;   .. ..$ FindVariableFeatures.RNA:Formal class 'SeuratCommand' [package "Seurat"] with 5 slots</span>
<span class="co">#&gt;   .. ..$ NormalizeData.RNA       :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots</span>
<span class="co">#&gt;   ..@ tools       : list()</span></code></pre></div>
<p><code>liana_wrap</code> calls a number of methods and and each method is run with the provided resource(s).</p>
<p>We will now call all methods that are currently available in liana.</p>
<p>Here we use only the “OmniPath” CCC resource, but any of the aforementioned ones (available via <code><a href="../reference/show_resources.html">show_resources()</a></code>) can be added to the <code>resource</code> parameter</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run liana</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">testdata</span><span class="op">)</span>
<span class="co">#&gt; Warning in .filter_sce(sce): 3465 genes and/or 0 cells were removed as they had</span>
<span class="co">#&gt; no counts!</span>

<span class="co"># Liana returns a list of results, each element of which corresponds to a method</span>
<span class="va">liana_test</span> <span class="op">%&gt;%</span> <span class="va">glimpse</span></code></pre></div>
<p>LIANA currently provides a mixture of re-implemented methods and pipelines which externally call specific LR methods. By default, LIANA will call only the internal scoring function (or those that are re-implemented in LIANA).</p>
<p>However, one can use LIANA to also run the original methods. For more information and validation with the original methods see <a href="https://saezlab.github.io/liana/articles/liana_devel.html">LIANA++</a> and <a href="https://saezlab.github.io/liana/articles/liana_remethod.html">validate scores</a>, respectively.</p>
</div>
<div id="aggregate-and-obiain-consensus-ranks" class="section level2">
<h2 class="hasAnchor">
<a href="#aggregate-and-obiain-consensus-ranks" class="anchor"></a>Aggregate and Obiain Consensus Ranks</h2>
<p><code>liana</code> also provides consensus ranks for the results obtained using different methods. By default, <code>liana</code> will provide mean, median, and aggregate* consensus ranks</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We can aggregate these results into a tibble with consensus ranks</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span> <span class="va">liana_test</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Now aggregating natmi</span>
<span class="co">#&gt; Now aggregating connectome</span>
<span class="co">#&gt; Now aggregating logfc</span>
<span class="co">#&gt; Now aggregating sca</span>
<span class="co">#&gt; Now aggregating cellphonedb</span>
<span class="co">#&gt; Aggregating Ranks</span>

<span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="va">liana_test</span><span class="op">)</span>
<span class="co">#&gt; Rows: 996</span>
<span class="co">#&gt; Columns: 17</span>
<span class="co">#&gt; $ source                 &lt;chr&gt; "B", "B", "B", "B", "B", "B", "B", "B", "B", "B…</span>
<span class="co">#&gt; $ ligand                 &lt;chr&gt; "HLA-DRA", "HLA-DRA", "HLA-DRA", "HLA-DMA", "HL…</span>
<span class="co">#&gt; $ target                 &lt;chr&gt; "NK", "B", "B", "B", "B", "B", "B", "B", "B", "…</span>
<span class="co">#&gt; $ receptor               &lt;chr&gt; "CD63", "CD37", "CD19", "CD74", "CD74", "CD37",…</span>
<span class="co">#&gt; $ aggregate_rank         &lt;dbl&gt; 4.602527e-07, 5.217006e-07, 3.260154e-06, 3.966…</span>
<span class="co">#&gt; $ mean_rank              &lt;dbl&gt; 13.2, 18.4, 48.0, 20.0, 26.2, 27.2, 53.4, 32.2,…</span>
<span class="co">#&gt; $ median_rank            &lt;dbl&gt; 2, 5, 19, 5, 8, 11, 13, 12, 25, 42, 27, 23, 3, …</span>
<span class="co">#&gt; $ natmi.edge_specificity &lt;dbl&gt; 0.4806804, 0.3655873, 0.6433933, 0.3961174, 0.3…</span>
<span class="co">#&gt; $ natmi.rank             &lt;dbl&gt; 39, 73, 19, 60, 62, 94, 13, 26, 22, 75, 135, 34…</span>
<span class="co">#&gt; $ connectome.weight_sc   &lt;dbl&gt; 1.0496186, 0.9928272, 0.8734601, 1.0355778, 0.9…</span>
<span class="co">#&gt; $ connectome.rank        &lt;dbl&gt; 1, 5, 19, 2, 7, 11, 13, 12, 25, 30, 29, 23, 3, …</span>
<span class="co">#&gt; $ logfc.logfc_comb       &lt;dbl&gt; 2.5312796, 2.7312529, 2.0594963, 2.1383984, 1.9…</span>
<span class="co">#&gt; $ logfc.rank             &lt;dbl&gt; 2, 1, 7, 5, 8, 6, 25, 4, 28, 42, 9, 23, 3, 35, …</span>
<span class="co">#&gt; $ sca.LRscore            &lt;dbl&gt; 0.9348366, 0.9442782, 0.8701017, 0.9293454, 0.9…</span>
<span class="co">#&gt; $ sca.rank               &lt;dbl&gt; 23, 12, 194, 32, 53, 24, 215, 118, 258, 79, 27,…</span>
<span class="co">#&gt; $ cellphonedb.pvalue     &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…</span>
<span class="co">#&gt; $ cellphonedb.rank       &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…</span></code></pre></div>
<p>Voila! That’s it. A very brief intro to LIANA and how to obtain the scoring functions† for each method implemented in it, as well as an aggregate_rank* which serves as a consensus across methods.</p>
<p>(†) Note that here we focus on the scores recommended to be used to prioritize interaction in a single sample system. Most of these, with the exception of SingleCellSignalR’s LRscore, take the specificity of the cluster pair into account.</p>
<p>(*) The aggregate consensus rank (<code>aggregate_rank</code>) is obtained using the RobustRankAggreg package and its <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3278763/"><code>RRA</code></a> method.</p>
<p><code>RRA</code> scores can be interpreted as p-values and interactions which are ranked consistently higher than random are assigned low scores/p-values.</p>
</div>
<div id="simple-dotplot" class="section level2">
<h2 class="hasAnchor">
<a href="#simple-dotplot" class="anchor"></a>Simple DotPlot</h2>
<p>We will now plot the results. By default, we use the <code>LRscore</code> from SingleCellSignalR to represent the magnitude of expression of the ligand and receptor, and NATMI’s <code>specificity weights</code> to show how specific a given interaction is to the <code>source</code>(L) and <code>target</code>(R) cell types.</p>
<p>In this case, we plot the top 25 interactions with the lowest aggregate_rank scores, and we plot interactions in which B cells are the <code>source</code> (express the ~ligands) and the other 3 cell types in this toy dataset are the <code>target</code> (express the ~receptors).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">liana_test</span> <span class="op">%&gt;%</span> 
  <span class="fu">top_n</span><span class="op">(</span><span class="fl">25</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">aggregate_rank</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/liana_dotplot.html">liana_dotplot</a></span><span class="op">(</span>source_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B"</span><span class="op">)</span>,
                target_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NK"</span>, <span class="st">"CD8 T"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="liana_tutorial_files/figure-html/liana_dotplot-1.png" width="1056"> Note that missing dots are interactions which are not expressed in at least 10% of the cells (by default) in both cell clusters.</p>
</div>
<div id="run-any-method-of-choice-" class="section level2">
<h2 class="hasAnchor">
<a href="#run-any-method-of-choice-" class="anchor"></a>Run any method of choice.</h2>
<p>We will now run only the CellPhoneDB’s permutation-based algorithm. We will also lower the number of permutations that we wish to perform for the sake of computational time. Note that one can also parallelize the CPDB algorithm implemented in LIANA (in this case we don’t, as this would only make sense when working with large datasets).</p>
<p>Also, here we will use a <code>SingleCellExperiment</code> object as input. In reality, LIANA converts <code>Seurat</code> objects to SingleCellExperiment and is to a large extend based on the BioConductor single-cell infrastructure.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load Sce testdata</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">liana_path</span> , <span class="st">"testdata"</span>, <span class="st">"input"</span>, <span class="st">"testsce.rds"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># RUN CPDB alone</span>
<span class="va">cpdb_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">sce</span>,
                        method <span class="op">=</span> <span class="st">'cellphonedb'</span>,
                        resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CellPhoneDB'</span><span class="op">)</span>,
                        permutation.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>nperms<span class="op">=</span><span class="fl">100</span>,
                                                  parallelize<span class="op">=</span><span class="cn">FALSE</span>,
                                                  workers<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Plot toy results</span>
<span class="va">cpdb_test</span> <span class="op">%&gt;%</span>
  <span class="co"># filter(pvalue &lt;= 0.05) %&gt;% # only keep interactions with p-val &lt;= 0.05</span>
  <span class="fu">mutate</span><span class="op">(</span>pvalue <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># invert size (low p-value/high specificity = larger dot size)</span>
  <span class="fu"><a href="../reference/liana_dotplot.html">liana_dotplot</a></span><span class="op">(</span>source_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"c"</span><span class="op">)</span>,
                target_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"c"</span>, <span class="st">"a"</span>, <span class="st">"b"</span><span class="op">)</span>,
                specificity <span class="op">=</span> <span class="st">"pvalue"</span>,
                magnitude <span class="op">=</span> <span class="st">"lr.mean"</span><span class="op">)</span></code></pre></div>
<p><img src="liana_tutorial_files/figure-html/cpdb_test-1.png" width="1056"></p>
</div>
<div id="singlecellsignalr-cytotalk-natmi-and-connectome-scores-w-complexes" class="section level2">
<h2 class="hasAnchor">
<a href="#singlecellsignalr-cytotalk-natmi-and-connectome-scores-w-complexes" class="anchor"></a>SingleCellSignalR, CytoTalk, NATMI, and Connectome scores /w Complexes</h2>
<p>The re-implementation of the aforementioned methods in LIANA enables us to make use of multimeric complex information as provided by the e.g. CellPhoneDB, CellChatDB, and ICELLNET resources</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run liana re-implementations with the CellPhoneDB resource</span>
<span class="va">complex_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">testdata</span>,
                           method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'natmi'</span>, <span class="st">'sca'</span>, <span class="st">'logfc'</span><span class="op">)</span>,
                           resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CellPhoneDB'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in .filter_sce(sce): 3465 genes and/or 0 cells were removed as they had</span>
<span class="co">#&gt; no counts!</span>

<span class="va">complex_test</span> <span class="op">%&gt;%</span> <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 233 x 13</span>
<span class="co">#&gt;    source ligand target receptor aggregate_rank mean_rank median_rank</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;             &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 CD8 T  LCK    CD8 T  CD8B           0.000765      10.7          11</span>
<span class="co">#&gt;  2 NK     MIF    B      CD74           0.000851      25.7           2</span>
<span class="co">#&gt;  3 B      MIF    B      CD74           0.00191       41             3</span>
<span class="co">#&gt;  4 NK     HLA-C  NK     KIR2DL3        0.00279       19.3          18</span>
<span class="co">#&gt;  5 CD8 T  MIF    B      CD74           0.00338       36.3           4</span>
<span class="co">#&gt;  6 NK     PTPRC  B      CD22           0.00764       29.3          29</span>
<span class="co">#&gt;  7 CD8 T  PTPRC  B      CD22           0.00849       26            29</span>
<span class="co">#&gt;  8 NK     CD58   CD8 T  CD2            0.0103        30             7</span>
<span class="co">#&gt;  9 B      COPA   B      CD74           0.0104        21.7          31</span>
<span class="co">#&gt; 10 CD8 T  CD48   NK     CD244          0.0125        22.7          21</span>
<span class="co">#&gt; # … with 223 more rows, and 6 more variables: natmi.edge_specificity &lt;dbl&gt;,</span>
<span class="co">#&gt; #   natmi.rank &lt;dbl&gt;, sca.LRscore &lt;dbl&gt;, sca.rank &lt;dbl&gt;,</span>
<span class="co">#&gt; #   logfc.logfc_comb &lt;dbl&gt;, logfc.rank &lt;dbl&gt;</span></code></pre></div>
</div>
<div id="call-liana-with-overwritten-default-settings" class="section level2">
<h2 class="hasAnchor">
<a href="#call-liana-with-overwritten-default-settings" class="anchor"></a>Call <code>liana</code> with overwritten default settings</h2>
<p>By default <code>liana</code> is run with the default for each method which can be obtained via <code>liana_default()</code></p>
<p>Alternatively, one can also overwrite the default settings by simply passing them to the liana wrapper function</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Overwrite CellChat default parameters by providing a list of parameters</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">testdata</span>,
                         method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'cellchat'</span>, <span class="st">'sca'</span><span class="op">)</span>,
                         resource <span class="op">=</span> <span class="st">'OmniPath'</span>,
                         cellchat.params <span class="op">=</span> 
                           <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
                             nboot <span class="op">=</span> <span class="fl">10</span>,
                             exclude_anns <span class="op">=</span> <span class="cn">NULL</span>,
                             thresh <span class="op">=</span> <span class="fl">1</span>,
                             assay <span class="op">=</span> <span class="st">"RNA"</span>,
                             .normalize <span class="op">=</span> <span class="cn">FALSE</span>,
                             .do_parallel <span class="op">=</span> <span class="cn">FALSE</span>,
                             .raw_use <span class="op">=</span> <span class="cn">TRUE</span>
                             <span class="op">)</span>,
                         liana_call.params<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>complex_policy <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span>
                         <span class="op">)</span>
<span class="co">#&gt; Warning in .filter_sce(sce): 3465 genes and/or 0 cells were removed as they had</span>
<span class="co">#&gt; no counts!</span>
<span class="co">#&gt; The cell groups used for CellChat analysis are  B CD8 T NK</span>
<span class="co">#&gt; Warning: Expected 5 pieces. Missing pieces filled with `NA` in 3 rows [1, 2, 3].</span>
<span class="co">#&gt; Issue identified!! Please check the official Gene Symbol of the following genes:  </span>
<span class="co">#&gt;  2 </span>
<span class="co">#&gt; Issue identified!! Please check the official Gene Symbol of the following genes:  </span>
<span class="co">#&gt;  HIDE1</span>

<span class="co"># This returns a list of results for each method</span>
<span class="va">liana_test</span> <span class="op">%&gt;%</span> <span class="va">glimpse</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ cellchat: tibble [245 × 6] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;   ..$ source  : Factor w/ 3 levels "B","CD8 T","NK": 1 3 1 2 3 1 2 3 1 2 ...</span>
<span class="co">#&gt;   ..$ target  : Factor w/ 3 levels "B","CD8 T","NK": 3 3 2 2 2 1 1 1 2 2 ...</span>
<span class="co">#&gt;   ..$ ligand  : chr [1:245] "TYROBP" "TYROBP" "HLA-B" "HLA-B" ...</span>
<span class="co">#&gt;   ..$ receptor: chr [1:245] "KLRD1" "KLRD1" "CD8A" "CD8A" ...</span>
<span class="co">#&gt;   ..$ prob    : num [1:245] 0.0105 0.1442 0.1544 0.1694 0.1827 ...</span>
<span class="co">#&gt;   ..$ pval    : num [1:245] 0.2 0 0 0 0 1 1 1 0 0 ...</span>
<span class="co">#&gt;  $ sca     : tibble [996 × 10] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;   ..$ source          : chr [1:996] "B" "B" "B" "B" ...</span>
<span class="co">#&gt;   ..$ target          : chr [1:996] "B" "B" "B" "B" ...</span>
<span class="co">#&gt;   ..$ ligand.complex  : chr [1:996] "ADAM10" "ITGB1BP1" "ITGB1BP1" "ITGB1BP1" ...</span>
<span class="co">#&gt;   ..$ ligand          : chr [1:996] "ADAM10" "ITGB1BP1" "ITGB1BP1" "ITGB1BP1" ...</span>
<span class="co">#&gt;   ..$ receptor.complex: chr [1:996] "CD44" "ITGB2" "ITGB1" "ITGB7" ...</span>
<span class="co">#&gt;   ..$ receptor        : chr [1:996] "CD44" "ITGB2" "ITGB1" "ITGB7" ...</span>
<span class="co">#&gt;   ..$ ligand.expr     : num [1:996] 0.12 0.322 0.322 0.322 2.914 ...</span>
<span class="co">#&gt;   ..$ receptor.expr   : num [1:996] 0.655 0.76 0.156 0.362 0.499 ...</span>
<span class="co">#&gt;   ..$ global_mean     : num [1:996] 0.199 0.199 0.199 0.199 0.199 ...</span>
<span class="co">#&gt;   ..$ LRscore         : num [1:996] 0.585 0.713 0.53 0.632 0.858 ...</span></code></pre></div>
<p>Note that with <code>liana_call.params</code> we change the way that we account for complexes. By default this is set to min0/min (as done for expression in CellPhoneDB/Squidpy), which assigns these statistics for each complex to those closest to 0. Here we change it to mean, but any alternative function can be passed and other perfectly viable approaches could also include e.g. the Trimean of CellChat.</p>
<p>Please refer to the <code><a href="../reference/liana_wrap.html">?liana_wrap</a></code> documentation for more information on all the parameters that can be tuned in liana. Also, one can obtain a list with all default parameters by calling the <code><a href="../reference/liana_defaults.html">liana_defaults()</a></code> function.</p>
<div id="information-about-the-parameters-passed-to-each-method-can-be-obtained-via-liana_call-and-via" class="section level6">
<h6>Information about the parameters passed to each method can be obtained via <code>liana_call</code> and via</h6>
<p><code>call_*</code> functions - e.g. ?call_cellchat or ?call_squidpy. A list with the default parameters themselves can be obtained via <code><a href="../reference/liana_defaults.html">liana_defaults()</a></code></p>
</div>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<pre><code>#&gt; 
#&gt; To cite liana in publications use:
#&gt; 
#&gt;   Dimitrov, D., Türei, D., Boys, C., Nagai, J.S., Flores, R.O.R., Kim,
#&gt;   H., Szalai, B., Costa, I.G., Dugourd, A., Valdeolivas, A. and
#&gt;   Saez-Rodriguez, J., 2021.  Comparison of Resources and Methods to
#&gt;   infer Cell-Cell Communication from Single-cell RNA Data. bioRxiv.
#&gt; 
#&gt; A BibTeX entry for LaTeX users is
#&gt; 
#&gt;   @Article{,
#&gt;     author = {Daniel Dimitrov and Charlotte Boys and Denes Turei and James Nagai S. and Ricardo Ramirez Flores O. and Hyojin Kim and Bence Szalai and Ivan Costa G. and Aurélien Dugourd and Alberto Valdeolivas and Julio Saez-Rodriguez},
#&gt;     title = {Comparison of Resources and Methods to infer Cell-Cell Communication from Single-cell RNA Data},
#&gt;     journal = {BioRxiv},
#&gt;     year = {2021},
#&gt;     doi = {10.1101/2021.05.21.445160},
#&gt;     encoding = {UTF-8},
#&gt;   }</code></pre>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.0.3 (2020-10-10)
#&gt;  os       Ubuntu 20.04.3 LTS          
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       Europe/Berlin               
#&gt;  date     2022-02-07                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package              * version  date       lib source                            
#&gt;  abind                  1.4-5    2016-07-21 [2] CRAN (R 4.0.2)                    
#&gt;  assertthat             0.2.1    2019-03-21 [2] CRAN (R 4.0.2)                    
#&gt;  backports              1.2.1    2020-12-09 [2] CRAN (R 4.0.2)                    
#&gt;  beachmat               2.6.4    2020-12-20 [2] Bioconductor                      
#&gt;  beeswarm               0.2.3    2016-04-25 [2] CRAN (R 4.0.2)                    
#&gt;  Biobase              * 2.50.0   2020-10-27 [2] Bioconductor                      
#&gt;  BiocGenerics         * 0.36.0   2020-10-27 [2] Bioconductor                      
#&gt;  BiocManager            1.30.10  2019-11-16 [2] CRAN (R 4.0.2)                    
#&gt;  BiocNeighbors          1.8.2    2020-12-07 [2] Bioconductor                      
#&gt;  BiocParallel           1.24.1   2020-11-06 [2] Bioconductor                      
#&gt;  BiocSingular           1.6.0    2020-10-27 [2] Bioconductor                      
#&gt;  BiocStyle            * 2.18.1   2020-11-24 [2] Bioconductor                      
#&gt;  bitops                 1.0-6    2013-08-17 [2] CRAN (R 4.0.2)                    
#&gt;  bluster                1.0.0    2020-10-27 [2] Bioconductor                      
#&gt;  bookdown               0.21     2020-10-13 [2] CRAN (R 4.0.2)                    
#&gt;  broom                  0.7.8    2021-06-24 [2] CRAN (R 4.0.3)                    
#&gt;  bslib                  0.2.4    2021-01-25 [2] CRAN (R 4.0.3)                    
#&gt;  cachem                 1.0.4    2021-02-13 [2] CRAN (R 4.0.3)                    
#&gt;  Cairo                  1.5-12.2 2020-07-07 [2] CRAN (R 4.0.3)                    
#&gt;  CellChat               1.1.0    2021-05-02 [2] Github (sqjin/CellChat@b3ccf96)   
#&gt;  cellranger             1.1.0    2016-07-27 [2] CRAN (R 4.0.2)                    
#&gt;  checkmate              2.0.0    2020-02-06 [2] CRAN (R 4.0.3)                    
#&gt;  circlize               0.4.12   2021-01-08 [2] CRAN (R 4.0.3)                    
#&gt;  cli                    2.5.0    2021-04-26 [2] CRAN (R 4.0.3)                    
#&gt;  clue                   0.3-58   2020-12-03 [2] CRAN (R 4.0.3)                    
#&gt;  cluster                2.1.0    2019-06-19 [5] CRAN (R 4.0.0)                    
#&gt;  coda                   0.19-4   2020-09-30 [2] CRAN (R 4.0.3)                    
#&gt;  codetools              0.2-16   2018-12-24 [5] CRAN (R 4.0.0)                    
#&gt;  colorspace             2.0-0    2020-11-11 [2] CRAN (R 4.0.2)                    
#&gt;  ComplexHeatmap         2.6.2    2020-11-12 [2] Bioconductor                      
#&gt;  cowplot                1.1.1    2020-12-30 [2] CRAN (R 4.0.3)                    
#&gt;  crayon                 1.4.1    2021-02-08 [2] CRAN (R 4.0.3)                    
#&gt;  curl                   4.3.2    2021-06-23 [2] CRAN (R 4.0.3)                    
#&gt;  data.table             1.14.0   2021-02-21 [2] CRAN (R 4.0.3)                    
#&gt;  DBI                    1.1.1    2021-01-15 [2] CRAN (R 4.0.3)                    
#&gt;  dbplyr                 2.1.0    2021-02-03 [2] CRAN (R 4.0.3)                    
#&gt;  DelayedArray           0.16.1   2021-01-22 [2] Bioconductor                      
#&gt;  DelayedMatrixStats     1.12.3   2021-02-03 [2] Bioconductor                      
#&gt;  deldir                 0.2-10   2021-02-16 [2] CRAN (R 4.0.3)                    
#&gt;  desc                   1.2.0    2018-05-01 [2] CRAN (R 4.0.2)                    
#&gt;  digest                 0.6.27   2020-10-24 [2] CRAN (R 4.0.2)                    
#&gt;  doParallel             1.0.16   2020-10-16 [2] CRAN (R 4.0.2)                    
#&gt;  dplyr                * 1.0.7    2021-06-18 [2] CRAN (R 4.0.3)                    
#&gt;  dqrng                  0.2.1    2019-05-17 [2] CRAN (R 4.0.2)                    
#&gt;  edgeR                  3.32.1   2021-01-14 [2] Bioconductor                      
#&gt;  ellipsis               0.3.2    2021-04-29 [2] CRAN (R 4.0.3)                    
#&gt;  entropy                1.2.1    2014-11-14 [2] CRAN (R 4.0.3)                    
#&gt;  evaluate               0.14     2019-05-28 [2] CRAN (R 4.0.2)                    
#&gt;  fansi                  0.4.2    2021-01-15 [2] CRAN (R 4.0.3)                    
#&gt;  farver                 2.0.3    2020-01-16 [2] CRAN (R 4.0.2)                    
#&gt;  fastmap                1.1.0    2021-01-25 [2] CRAN (R 4.0.3)                    
#&gt;  fitdistrplus           1.1-3    2020-12-05 [2] CRAN (R 4.0.2)                    
#&gt;  FNN                    1.1.3    2019-02-15 [2] CRAN (R 4.0.2)                    
#&gt;  forcats              * 0.5.1    2021-01-27 [2] CRAN (R 4.0.3)                    
#&gt;  foreach                1.5.1    2020-10-15 [2] CRAN (R 4.0.2)                    
#&gt;  fs                     1.5.0    2020-07-31 [2] CRAN (R 4.0.2)                    
#&gt;  future                 1.21.0   2020-12-10 [2] CRAN (R 4.0.3)                    
#&gt;  future.apply           1.7.0    2021-01-04 [2] CRAN (R 4.0.3)                    
#&gt;  generics               0.1.0    2020-10-31 [2] CRAN (R 4.0.2)                    
#&gt;  GenomeInfoDb           1.26.2   2020-12-08 [2] Bioconductor                      
#&gt;  GenomeInfoDbData       1.2.4    2020-12-14 [2] Bioconductor                      
#&gt;  GenomicRanges          1.42.0   2020-10-27 [2] Bioconductor                      
#&gt;  GetoptLong             1.0.5    2020-12-15 [2] CRAN (R 4.0.3)                    
#&gt;  gg.gap                 1.3      2019-09-30 [2] CRAN (R 4.0.3)                    
#&gt;  ggalluvial             0.12.3   2020-12-05 [2] CRAN (R 4.0.3)                    
#&gt;  ggbeeswarm             0.6.0    2017-08-07 [2] CRAN (R 4.0.2)                    
#&gt;  ggplot2              * 3.3.5    2021-06-25 [2] CRAN (R 4.0.3)                    
#&gt;  ggrepel                0.9.1    2021-01-15 [2] CRAN (R 4.0.3)                    
#&gt;  ggridges               0.5.3    2021-01-08 [2] CRAN (R 4.0.3)                    
#&gt;  GlobalOptions          0.1.2    2020-06-10 [2] CRAN (R 4.0.3)                    
#&gt;  globals                0.14.0   2020-11-22 [2] CRAN (R 4.0.2)                    
#&gt;  glue                   1.4.2    2020-08-27 [2] CRAN (R 4.0.2)                    
#&gt;  goftest                1.2-2    2019-12-02 [2] CRAN (R 4.0.2)                    
#&gt;  gridBase               0.4-7    2014-02-24 [2] CRAN (R 4.0.3)                    
#&gt;  gridExtra              2.3      2017-09-09 [2] CRAN (R 4.0.2)                    
#&gt;  gtable                 0.3.0    2019-03-25 [2] CRAN (R 4.0.2)                    
#&gt;  haven                  2.3.1    2020-06-01 [2] CRAN (R 4.0.2)                    
#&gt;  highr                  0.8      2019-03-20 [2] CRAN (R 4.0.2)                    
#&gt;  hms                    1.0.0    2021-01-13 [2] CRAN (R 4.0.3)                    
#&gt;  htmltools              0.5.1.1  2021-01-22 [2] CRAN (R 4.0.3)                    
#&gt;  htmlwidgets            1.5.3    2020-12-10 [2] CRAN (R 4.0.2)                    
#&gt;  httpuv                 1.5.5    2021-01-13 [2] CRAN (R 4.0.3)                    
#&gt;  httr                   1.4.2    2020-07-20 [2] CRAN (R 4.0.2)                    
#&gt;  ica                    1.0-2    2018-05-24 [2] CRAN (R 4.0.2)                    
#&gt;  igraph                 1.2.6    2020-10-06 [2] CRAN (R 4.0.2)                    
#&gt;  IRanges                2.24.1   2020-12-12 [2] Bioconductor                      
#&gt;  irlba                  2.3.3    2019-02-05 [2] CRAN (R 4.0.2)                    
#&gt;  iterators              1.0.13   2020-10-15 [2] CRAN (R 4.0.2)                    
#&gt;  jquerylib              0.1.3    2020-12-17 [2] CRAN (R 4.0.3)                    
#&gt;  jsonlite               1.7.2    2020-12-09 [2] CRAN (R 4.0.2)                    
#&gt;  KernSmooth             2.23-17  2020-04-26 [5] CRAN (R 4.0.0)                    
#&gt;  knitr                  1.31     2021-01-27 [2] CRAN (R 4.0.3)                    
#&gt;  labeling               0.4.2    2020-10-20 [2] CRAN (R 4.0.2)                    
#&gt;  later                  1.1.0.1  2020-06-05 [2] CRAN (R 4.0.2)                    
#&gt;  lattice                0.20-41  2020-04-02 [5] CRAN (R 4.0.0)                    
#&gt;  lazyeval               0.2.2    2019-03-15 [2] CRAN (R 4.0.2)                    
#&gt;  leiden                 0.3.7    2021-01-26 [2] CRAN (R 4.0.3)                    
#&gt;  liana                * 0.0.6    2022-02-07 [1] local                             
#&gt;  lifecycle              1.0.0    2021-02-15 [2] CRAN (R 4.0.3)                    
#&gt;  limma                  3.46.0   2020-10-27 [2] Bioconductor                      
#&gt;  listenv                0.8.0    2019-12-05 [2] CRAN (R 4.0.2)                    
#&gt;  lmtest                 0.9-38   2020-09-09 [2] CRAN (R 4.0.2)                    
#&gt;  locfit                 1.5-9.4  2020-03-25 [2] CRAN (R 4.0.2)                    
#&gt;  logger                 0.2.0    2021-03-04 [2] CRAN (R 4.0.3)                    
#&gt;  lubridate              1.7.9.2  2020-11-13 [2] CRAN (R 4.0.2)                    
#&gt;  magrittr             * 2.0.1    2020-11-17 [2] CRAN (R 4.0.2)                    
#&gt;  MASS                   7.3-53   2020-09-09 [5] CRAN (R 4.0.2)                    
#&gt;  Matrix                 1.3-4    2021-06-01 [2] CRAN (R 4.0.3)                    
#&gt;  MatrixGenerics         1.2.1    2021-01-30 [2] Bioconductor                      
#&gt;  matrixStats            0.58.0   2021-01-29 [2] CRAN (R 4.0.3)                    
#&gt;  memoise                2.0.0    2021-01-26 [2] CRAN (R 4.0.3)                    
#&gt;  mgcv                   1.8-33   2020-08-27 [5] CRAN (R 4.0.2)                    
#&gt;  mime                   0.10     2021-02-13 [2] CRAN (R 4.0.3)                    
#&gt;  miniUI                 0.1.1.1  2018-05-18 [2] CRAN (R 4.0.2)                    
#&gt;  modelr                 0.1.8    2020-05-19 [2] CRAN (R 4.0.2)                    
#&gt;  munsell                0.5.0    2018-06-12 [2] CRAN (R 4.0.2)                    
#&gt;  network                1.16.1   2020-10-07 [2] CRAN (R 4.0.2)                    
#&gt;  nlme                   3.1-149  2020-08-23 [5] CRAN (R 4.0.2)                    
#&gt;  NMF                    0.23.0   2020-08-01 [2] CRAN (R 4.0.3)                    
#&gt;  OmnipathR              3.3.5    2021-12-12 [2] Github (saezlab/OmnipathR@30e0133)
#&gt;  parallelly             1.24.0   2021-03-14 [2] CRAN (R 4.0.3)                    
#&gt;  patchwork              1.1.1    2020-12-17 [2] CRAN (R 4.0.3)                    
#&gt;  pbapply                1.4-3    2020-08-18 [2] CRAN (R 4.0.2)                    
#&gt;  pillar                 1.6.1    2021-05-16 [2] CRAN (R 4.0.3)                    
#&gt;  pkgconfig              2.0.3    2019-09-22 [2] CRAN (R 4.0.2)                    
#&gt;  pkgdown                1.6.1    2020-09-12 [2] CRAN (R 4.0.3)                    
#&gt;  pkgmaker               0.32.2   2020-10-20 [2] CRAN (R 4.0.3)                    
#&gt;  plotly                 4.9.3    2021-01-10 [2] CRAN (R 4.0.3)                    
#&gt;  plyr                   1.8.6    2020-03-03 [2] CRAN (R 4.0.2)                    
#&gt;  png                    0.1-7    2013-12-03 [2] CRAN (R 4.0.2)                    
#&gt;  polyclip               1.10-0   2019-03-14 [2] CRAN (R 4.0.2)                    
#&gt;  prettyunits            1.1.1    2020-01-24 [2] CRAN (R 4.0.2)                    
#&gt;  progress               1.2.2    2019-05-16 [2] CRAN (R 4.0.2)                    
#&gt;  promises               1.2.0.1  2021-02-11 [2] CRAN (R 4.0.3)                    
#&gt;  purrr                * 0.3.4    2020-04-17 [2] CRAN (R 4.0.2)                    
#&gt;  R6                     2.5.0    2020-10-28 [2] CRAN (R 4.0.2)                    
#&gt;  ragg                   1.1.0    2021-02-15 [2] CRAN (R 4.0.3)                    
#&gt;  RANN                   2.6.1    2019-01-08 [2] CRAN (R 4.0.2)                    
#&gt;  rappdirs               0.3.3    2021-01-31 [2] CRAN (R 4.0.3)                    
#&gt;  RColorBrewer           1.1-2    2014-12-07 [2] CRAN (R 4.0.2)                    
#&gt;  Rcpp                   1.0.6    2021-01-15 [2] CRAN (R 4.0.3)                    
#&gt;  RcppAnnoy              0.0.18   2020-12-15 [2] CRAN (R 4.0.3)                    
#&gt;  RCurl                  1.98-1.3 2021-03-16 [2] CRAN (R 4.0.3)                    
#&gt;  readr                * 2.1.0    2021-11-11 [2] CRAN (R 4.0.3)                    
#&gt;  readxl                 1.3.1    2019-03-13 [2] CRAN (R 4.0.2)                    
#&gt;  registry               0.5-1    2019-03-05 [2] CRAN (R 4.0.3)                    
#&gt;  reprex                 1.0.0    2021-01-27 [2] CRAN (R 4.0.3)                    
#&gt;  reshape2               1.4.4    2020-04-09 [2] CRAN (R 4.0.2)                    
#&gt;  reticulate             1.20     2021-05-03 [2] CRAN (R 4.0.3)                    
#&gt;  rjson                  0.2.20   2018-06-08 [2] CRAN (R 4.0.3)                    
#&gt;  rlang                  0.4.11   2021-04-30 [2] CRAN (R 4.0.3)                    
#&gt;  rle                    0.9.2    2020-09-25 [2] CRAN (R 4.0.3)                    
#&gt;  rmarkdown              2.7      2021-02-19 [2] CRAN (R 4.0.3)                    
#&gt;  rngtools               1.5      2020-01-23 [2] CRAN (R 4.0.2)                    
#&gt;  RobustRankAggreg       1.1      2013-06-03 [2] CRAN (R 4.0.3)                    
#&gt;  ROCR                   1.0-11   2020-05-02 [2] CRAN (R 4.0.2)                    
#&gt;  rpart                  4.1-15   2019-04-12 [5] CRAN (R 4.0.0)                    
#&gt;  rprojroot              2.0.2    2020-11-15 [2] CRAN (R 4.0.2)                    
#&gt;  RSpectra               0.16-0   2019-12-01 [2] CRAN (R 4.0.2)                    
#&gt;  rstudioapi             0.13     2020-11-12 [2] CRAN (R 4.0.2)                    
#&gt;  rsvd                   1.0.3    2020-02-17 [2] CRAN (R 4.0.2)                    
#&gt;  Rtsne                  0.15     2018-11-10 [2] CRAN (R 4.0.2)                    
#&gt;  rvest                  0.3.6    2020-07-25 [2] CRAN (R 4.0.2)                    
#&gt;  S4Vectors              0.28.1   2020-12-09 [2] Bioconductor                      
#&gt;  sass                   0.3.1    2021-01-24 [2] CRAN (R 4.0.3)                    
#&gt;  scales                 1.1.1    2020-05-11 [2] CRAN (R 4.0.2)                    
#&gt;  scater                 1.18.5   2021-02-16 [2] Bioconductor                      
#&gt;  scattermore            0.7      2020-11-24 [2] CRAN (R 4.0.3)                    
#&gt;  scran                  1.18.5   2021-02-04 [2] Bioconductor                      
#&gt;  sctransform            0.3.2    2020-12-16 [2] CRAN (R 4.0.3)                    
#&gt;  scuttle                1.0.4    2020-12-17 [2] Bioconductor                      
#&gt;  sessioninfo            1.1.1    2018-11-05 [2] CRAN (R 4.0.2)                    
#&gt;  Seurat                 4.0.3    2021-06-10 [2] CRAN (R 4.0.3)                    
#&gt;  SeuratObject           4.0.2    2021-06-09 [2] CRAN (R 4.0.3)                    
#&gt;  shape                  1.4.5    2020-09-13 [2] CRAN (R 4.0.3)                    
#&gt;  shiny                  1.6.0    2021-01-25 [2] CRAN (R 4.0.3)                    
#&gt;  SingleCellExperiment   1.12.0   2020-10-27 [2] Bioconductor                      
#&gt;  sna                    2.6      2020-10-06 [2] CRAN (R 4.0.3)                    
#&gt;  sparseMatrixStats      1.2.1    2021-02-02 [2] Bioconductor                      
#&gt;  spatstat.core          2.0-0    2021-03-23 [2] CRAN (R 4.0.3)                    
#&gt;  spatstat.data          2.1-0    2021-03-21 [2] CRAN (R 4.0.3)                    
#&gt;  spatstat.geom          2.0-1    2021-03-22 [2] CRAN (R 4.0.3)                    
#&gt;  spatstat.sparse        2.0-0    2021-03-16 [2] CRAN (R 4.0.3)                    
#&gt;  spatstat.utils         2.1-0    2021-03-15 [2] CRAN (R 4.0.3)                    
#&gt;  statmod                1.4.35   2020-10-19 [2] CRAN (R 4.0.2)                    
#&gt;  statnet.common         4.4.1    2020-10-03 [2] CRAN (R 4.0.3)                    
#&gt;  stringi                1.5.3    2020-09-09 [2] CRAN (R 4.0.2)                    
#&gt;  stringr              * 1.4.0    2019-02-10 [2] CRAN (R 4.0.2)                    
#&gt;  SummarizedExperiment   1.20.0   2020-10-27 [2] Bioconductor                      
#&gt;  survival               3.2-7    2020-09-28 [2] CRAN (R 4.0.2)                    
#&gt;  svglite                2.0.0    2021-02-20 [2] CRAN (R 4.0.3)                    
#&gt;  systemfonts            1.0.1    2021-02-09 [2] CRAN (R 4.0.3)                    
#&gt;  tensor                 1.5      2012-05-05 [2] CRAN (R 4.0.2)                    
#&gt;  textshaping            0.3.0    2021-02-10 [2] CRAN (R 4.0.3)                    
#&gt;  tibble               * 3.1.2    2021-05-16 [2] CRAN (R 4.0.3)                    
#&gt;  tidyr                * 1.1.3    2021-03-03 [2] CRAN (R 4.0.3)                    
#&gt;  tidyselect             1.1.0    2020-05-11 [2] CRAN (R 4.0.2)                    
#&gt;  tidyverse            * 1.3.0    2019-11-21 [2] CRAN (R 4.0.3)                    
#&gt;  tzdb                   0.1.2    2021-07-20 [2] CRAN (R 4.0.3)                    
#&gt;  utf8                   1.1.4    2018-05-24 [2] CRAN (R 4.0.2)                    
#&gt;  uwot                   0.1.10   2020-12-15 [2] CRAN (R 4.0.3)                    
#&gt;  vctrs                  0.3.8    2021-04-29 [2] CRAN (R 4.0.3)                    
#&gt;  vipor                  0.4.5    2017-03-22 [2] CRAN (R 4.0.2)                    
#&gt;  viridis                0.5.1    2018-03-29 [2] CRAN (R 4.0.2)                    
#&gt;  viridisLite            0.3.0    2018-02-01 [2] CRAN (R 4.0.2)                    
#&gt;  withr                  2.4.1    2021-01-26 [2] CRAN (R 4.0.3)                    
#&gt;  xfun                   0.21     2021-02-10 [2] CRAN (R 4.0.3)                    
#&gt;  xml2                   1.3.2    2020-04-23 [2] CRAN (R 4.0.2)                    
#&gt;  xtable                 1.8-4    2019-04-21 [2] CRAN (R 4.0.2)                    
#&gt;  XVector                0.30.0   2020-10-27 [2] Bioconductor                      
#&gt;  yaml                   2.2.1    2020-02-01 [2] CRAN (R 4.0.2)                    
#&gt;  zlibbioc               1.36.0   2020-10-27 [2] Bioconductor                      
#&gt;  zoo                    1.8-8    2020-05-02 [2] CRAN (R 4.0.2)                    
#&gt; 
#&gt; [1] /tmp/RtmpUYDfzk/temp_libpath32ca4bd7ccea
#&gt; [2] /home/dbdimitrov/R/x86_64-pc-linux-gnu-library/4.0
#&gt; [3] /usr/local/lib/R/site-library
#&gt; [4] /usr/lib/R/site-library
#&gt; [5] /usr/lib/R/library</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
