<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>LIANA - NicheNet • liana</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="LIANA - NicheNet">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_intracell.html">LIANA with Intracellular Omnipath</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_ortho.html">Orthology Conversion with LIANA</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="liana_nichenet_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>LIANA - NicheNet</h1>
                        <h4 class="author">Martin Garrido Rodriguez-Cordoba</h4>
            
            <h4 class="date">2022-05-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/master/vignettes/liana_nichenet.Rmd"><code>vignettes/liana_nichenet.Rmd</code></a></small>
      <div class="hidden name"><code>liana_nichenet.Rmd</code></div>

    </div>

    
    
<div id="before-you-start" class="section level2">
<h2 class="hasAnchor">
<a href="#before-you-start" class="anchor"></a>Before you start</h2>
<p>This vignette is an example of how to combine LIANA’s hypotheses with those produced by NicheNet. The biological results that are obtained here are highly dependent on the hypothesis in mind, which is described in <a href="https://www.nature.com/articles/s41592-019-0667-5">NicheNet’s original article</a>. Before you start, we strongly recommend to have a look to the <a href="https://github.com/saeyslab/nichenetr">NicheNet’s repository</a>, as the data and the analyses that are run here were extracted from its vignettes. We acknowledge NicheNet’s authors for the extensive and reproducible documentation that they provide in this repository.</p>
</div>
<div id="background" class="section level2">
<h2 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h2>
<p><a href="https://saezlab.github.io/liana/index.html">LIANA</a> (LIgand-receptor ANalysis frAmework) is a framework able to prioritize ligand-receptor interactions from single-cell transcriptomics using different resources and methods. It allows users to systematically generate hypotheses about which ligands from a given cell type is binding receptors on another. In contrast to LIANA, NicheNet aims to deepen in the intra-cellular mechanisms that connect a ligand with a set of transcriptional targets, making an extensive usage of prior knowledge from multiple sources. <strong>LIANA and NicheNet are not mutually exclusive, but rather complementary, given that they aim to explore different aspects of inter and intra cellular communication</strong>.</p>
<p><img src="liana_nichenet.svg"></p>
<p>Because of this, in this vignette, we show how to use LIANA in combination with NicheNet using the data and the biological scenario described in <a href="https://github.com/saeyslab/nichenetr/blob/master/vignettes/ligand_activity_geneset.md">NicheNet’s vignette</a>. Briefly, the biological question here is: Which ligands expressed by cancer-associated fibroblasts (CAFs) can induce a specific gene program in neighboring malignant cells? (Using data from Puram et al. 2017).</p>
</div>
<div id="load-packages-and-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages-and-data" class="anchor"></a>Load packages and data</h2>
<p>We first load required libraries</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/liana/">liana</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/saeyslab/nichenetr">nichenetr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel">ggrepel</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">300</span><span class="op">)</span> <span class="co"># required for downloading single-cell expression on slow connection</span></code></pre></div>
<p>Then, we load and prepare the single-cell data, NicheNet’s model weights, and the gene set of interest. The latest is composed by genes that are known to participate in the partial epithelial-mesenchymal transition (p-EMT) program, as defined in <a href="https://github.com/saeyslab/nichenetr/blob/master/vignettes/ligand_activity_geneset.md">NicheNet’s vignette</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># single-cell expression matrix described in Puram et al. 2017</span>
<span class="va">hnscc_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/3260758/files/hnscc_expression.rds"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># model weights</span>
<span class="va">ligand_target_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/3260758/files/ligand_target_matrix.rds"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Furthermore, we restrict the single-cell data to the two cell types of interest for this example, which are the cancer associated fibroblasts (CAFs) and the tumor cells.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">expression</span> <span class="op">&lt;-</span> <span class="va">hnscc_expression</span><span class="op">$</span><span class="va">expression</span>
<span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">hnscc_expression</span><span class="op">$</span><span class="va">sample_info</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/make.names.html">make.names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span><span class="op">)</span>

<span class="co"># filter samples based on vignette's information and add cell type</span>
<span class="va">tumors_remove</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HN10"</span>, <span class="st">"HN"</span>, <span class="st">"HN12"</span>, <span class="st">"HN13"</span>, <span class="st">"HN24"</span>, <span class="st">"HN7"</span>, <span class="st">"HN8"</span>, <span class="st">"HN23"</span><span class="op">)</span>
<span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">sample_info</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span> <span class="op">!</span><span class="op">(</span><span class="va">tumor</span> <span class="op">%in%</span> <span class="va">tumors_remove</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">Lymph.node</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># fix some cell type identity names</span>
  <span class="fu">mutate</span><span class="op">(</span>cell_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classified..as.cancer.cell</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"Tumor"</span>, <span class="va">non.cancer.cell.type</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Tumor"</span>, <span class="st">"CAF"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># cell ID as rownames</span>
<span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">cell</span>

<span class="co"># subset expression to selected cells</span>
<span class="va">expression</span> <span class="op">&lt;-</span> <span class="va">expression</span><span class="op">[</span><span class="va">sample_info</span><span class="op">$</span><span class="va">cell</span>, <span class="op">]</span>

<span class="co"># gene set of interest</span>
<span class="va">geneset_oi</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/3260758/files/pemt_signature.txt"</span><span class="op">)</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span><span class="op">)</span>, col_names <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="va">.</span><span class="op">[</span><span class="va">.</span> <span class="op">%in%</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames</a></span><span class="op">(</span><span class="va">ligand_target_matrix</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="run-liana" class="section level2">
<h2 class="hasAnchor">
<a href="#run-liana" class="anchor"></a>Run LIANA</h2>
<p>In the first step, we run LIANA to systematically score all the ligand-receptor interactions between all the cell types included in the dataset. To do so, we first need to create a Seurat object from data:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create seurat object</span>
<span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/CreateAssayObject.html">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">expm1</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/CreateSeuratObject.html">CreateSeuratObject</a></span><span class="op">(</span><span class="va">.</span>, meta.data <span class="op">=</span> <span class="va">sample_info</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># set cell identity to cell type</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">seurat_object</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">cell_type</span></code></pre></div>
<p>And then we can execute LIANA using default parameters. After LIANA execution, we employ the function <code><a href="../reference/liana_aggregate.html">liana_aggregate()</a></code> to summarize the output of different methods and to obtain a single score for each interaction.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">liana_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>By default, LIANA will score the ligand-receptor interactions in all the possible directions within the two cell types of interest. This includes: Autocrine signaling (e.g. CAFs -&gt; CAFs), CAFs -&gt; Tumor cells and Tumor cells -&gt; CAFs. As we are only interested in the CAFs -&gt; Tumor cell direction, we filter the results and visualize the top 50 interactions according to the consensus/aggregate rank across methods. The aggregate rank itself can be interpreted as the significance of preferential enrichment for the interactions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># filter results to cell types of interest</span>
<span class="va">caf_tumor_results</span> <span class="op">&lt;-</span> <span class="va">liana_results</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">source</span> <span class="op">==</span> <span class="st">"CAF"</span> <span class="op">&amp;</span> <span class="va">target</span> <span class="op">==</span> <span class="st">"Tumor"</span><span class="op">)</span>

<span class="co"># filter results to top N interactions</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">50</span>
<span class="va">top_n_caf_tumor</span> <span class="op">&lt;-</span> <span class="va">caf_tumor_results</span> <span class="op">%&gt;%</span>
  <span class="fu">arrange</span><span class="op">(</span><span class="va">aggregate_rank</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">slice_head</span><span class="op">(</span>n <span class="op">=</span> <span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>id <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ligand</span>, <span class="st">" -&gt; "</span>, <span class="va">receptor</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># visualize median rank</span>
<span class="va">top_n_caf_tumor</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">aggregate_rank</span>, x <span class="op">=</span> <span class="va">id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Interaction"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"LIANA's aggregate rank"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span>, angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="liana_nichenet_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div id="run-nichenet-using-lianas-ligands" class="section level2">
<h2 class="hasAnchor">
<a href="#run-nichenet-using-lianas-ligands" class="anchor"></a>Run NicheNet using LIANA’s ligands</h2>
<p>The key aspect of combining LIANA with NicheNet is that we can use the ligands prioritized by LIANA as the set of potential ligands for NicheNet. Instead of evaluating all the expressed ligands for which the receptor is also expressed in the receiver cell type, we will only explore those that were prioritized by the methods included in LIANA. Hence, we select the ligands that form the interactions previously shown.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get ligands and filter to those included in NicheNet's ligand-target matrix</span>
<span class="va">ligands</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">top_n_caf_tumor</span><span class="op">$</span><span class="va">ligand</span><span class="op">)</span>
<span class="va">ligands</span> <span class="op">&lt;-</span> <span class="va">ligands</span><span class="op">[</span><span class="va">ligands</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ligand_target_matrix</span><span class="op">)</span><span class="op">]</span>
<span class="va">ligands</span></code></pre></div>
<pre><code>##  [1] "COL1A1" "FN1"    "THBS2"  "CXCL12" "COL4A1" "THBS1"  "THBS4"  "LAMB1" 
##  [9] "LAMA2"  "COMP"   "LAMB2"  "HSPG2"</code></pre>
<p>Before running NicheNet, we also need to define a list of background genes. To do so, we employ the threshold defined in <a href="https://github.com/saeyslab/nichenetr/blob/master/vignettes/ligand_activity_geneset.md">NicheNet’s vignette</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">background_genes</span> <span class="op">&lt;-</span> <span class="va">expression</span><span class="op">[</span><span class="va">sample_info</span><span class="op">$</span><span class="va">cell</span><span class="op">[</span><span class="va">sample_info</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">"Tumor"</span><span class="op">]</span>, <span class="op">]</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fl">2</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fl">10</span><span class="op">*</span><span class="op">(</span><span class="fl">2</span><span class="op">**</span><span class="va">x</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fl">2</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="va">.</span><span class="op">[</span><span class="va">.</span> <span class="op">&gt;=</span> <span class="fl">4</span><span class="op">]</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>And execute NicheNet to predict the ligand activities using the pEMT gene set previously mentioned</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nichenet_activities</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/nichenetr/man/predict_ligand_activities.html">predict_ligand_activities</a></span><span class="op">(</span>
  geneset <span class="op">=</span> <span class="va">geneset_oi</span>,
  background_expressed_genes <span class="op">=</span> <span class="va">background_genes</span>,
  ligand_target_matrix <span class="op">=</span> <span class="va">ligand_target_matrix</span>, potential_ligands <span class="op">=</span> <span class="va">ligands</span>
<span class="op">)</span></code></pre></div>
<p>As a result, we obtain the NicheNet’s activity predictions for the ligands previously prioritized using LIANA. In a final step, we will visualize the ligand-receptor scores of LIANA and the ligand activity score of NicheNet in a single figure.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># prepare data for visualization</span>
<span class="va">vis_liana_nichenet</span> <span class="op">&lt;-</span> <span class="va">top_n_caf_tumor</span> <span class="op">%&gt;%</span>
  <span class="fu">inner_join</span><span class="op">(</span><span class="va">nichenet_activities</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ligand"</span> <span class="op">=</span> <span class="st">"test_ligand"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">arrange</span><span class="op">(</span><span class="va">pearson</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>ligand <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">ligand</span><span class="op">)</span><span class="op">)</span>

<span class="co"># prepare NicheNet figure</span>
<span class="va">nichenet_scores_plot</span> <span class="op">&lt;-</span> <span class="va">vis_liana_nichenet</span> <span class="op">%&gt;%</span>
  <span class="fu">group_by</span><span class="op">(</span><span class="va">ligand</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">pearson</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">ligand</span>, x <span class="op">=</span> <span class="va">pearson</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"NicheNet"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Pearson's score"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        axis.ticks.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        axis.line.y <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>,
        plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="co"># prepare LIANA figure</span>
<span class="va">liana_receptor_heatmap</span> <span class="op">&lt;-</span> <span class="va">vis_liana_nichenet</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">ligand</span>, x <span class="op">=</span> <span class="va">receptor</span>, fill <span class="op">=</span> <span class="va">aggregate_rank</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tile</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"LIANA"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ylab</span><span class="op">(</span><span class="st">"Ligand"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Receptor"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
        plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,
        panel.grid.major <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="co"># combine plots</span>
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">liana_receptor_heatmap</span>, <span class="va">nichenet_scores_plot</span>,
          align <span class="op">=</span> <span class="st">"h"</span>, nrow <span class="op">=</span> <span class="fl">1</span>, rel_widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.8</span>,<span class="fl">0.3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="liana_nichenet_files/figure-html/unnamed-chunk-10-1.png" width="768"></p>
</div>
<div id="conclusion" class="section level2">
<h2 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h2>
<p>In this vignette, we exemplify how to use LIANA’s predictions as NicheNet’s input. Although both methods are complementary, there is one point that should not be forgotten: LIANA predicts ligand-receptor interaction pairs. However, NicheNet score for a given ligand comes from how likely is to reach a set of given targets from it. A ligand can have a great Pearson correlation score to regulate a given set of targets, but we do not actually know if it is mediated by the receptor that we predicted using LIANA. Given this, <strong>the combination of methods to predict cell-cell communication (like LIANA) with tools that are able to model intracellular signaling using prior knowledge (e.g. NicheNet) constitute a promising approach to deepen in the signaling mechanisms implicated in the biological process under study.</strong></p>
</div>
<div id="r-session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#r-session-information" class="anchor"></a>R Session information</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] cowplot_1.1.1      ggrepel_0.9.1      SeuratObject_4.0.2 Seurat_4.0.3      
##  [5] nichenetr_1.0.0    liana_0.1.2        forcats_0.5.1      stringr_1.4.0     
##  [9] dplyr_1.0.7        purrr_0.3.4        readr_2.1.0        tidyr_1.1.3       
## [13] tibble_3.1.2       ggplot2_3.3.5      tidyverse_1.3.0    BiocStyle_2.18.1  
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.3              scattermore_0.7            
##   [3] ModelMetrics_1.2.2.2        ragg_1.1.0                 
##   [5] bit64_4.0.5                 knitr_1.31                 
##   [7] irlba_2.3.3                 DelayedArray_0.16.1        
##   [9] data.table_1.14.0           rpart_4.1-15               
##  [11] RCurl_1.98-1.3              generics_0.1.0             
##  [13] BiocGenerics_0.36.0         RANN_2.6.1                 
##  [15] future_1.21.0               DiagrammeR_1.0.6.1         
##  [17] bit_4.0.4                   tzdb_0.1.2                 
##  [19] spatstat.data_2.1-0         xml2_1.3.2                 
##  [21] lubridate_1.7.9.2           httpuv_1.5.5               
##  [23] SummarizedExperiment_1.20.0 assertthat_0.2.1           
##  [25] viridis_0.5.1               gower_0.2.2                
##  [27] xfun_0.21                   hms_1.0.0                  
##  [29] jquerylib_0.1.3             evaluate_0.14              
##  [31] promises_1.2.0.1            fansi_0.4.2                
##  [33] progress_1.2.2              caTools_1.18.1             
##  [35] dbplyr_2.1.0                readxl_1.3.1               
##  [37] igraph_1.3.0                DBI_1.1.1                  
##  [39] htmlwidgets_1.5.3           spatstat.geom_2.0-1        
##  [41] stats4_4.0.3                ellipsis_0.3.2             
##  [43] backports_1.2.1             bookdown_0.21              
##  [45] deldir_0.2-10               sparseMatrixStats_1.2.1    
##  [47] MatrixGenerics_1.2.1        vctrs_0.3.8                
##  [49] SingleCellExperiment_1.12.0 Biobase_2.50.0             
##  [51] Cairo_1.5-12.2              ROCR_1.0-11                
##  [53] abind_1.4-5                 caret_6.0-88               
##  [55] cachem_1.0.4                withr_2.4.1                
##  [57] vroom_1.5.6                 checkmate_2.0.0            
##  [59] sctransform_0.3.2           fdrtool_1.2.16             
##  [61] prettyunits_1.1.1           scran_1.18.5               
##  [63] goftest_1.2-2               cluster_2.1.0              
##  [65] lazyeval_0.2.2              crayon_1.4.1               
##  [67] labeling_0.4.2              edgeR_3.32.1               
##  [69] recipes_0.1.16              pkgconfig_2.0.3            
##  [71] GenomeInfoDb_1.26.2         vipor_0.4.5                
##  [73] nlme_3.1-149                nnet_7.3-14                
##  [75] rlang_0.4.11                globals_0.14.0             
##  [77] lifecycle_1.0.0             miniUI_0.1.1.1             
##  [79] modelr_0.1.8                rsvd_1.0.3                 
##  [81] randomForest_4.6-14         cellranger_1.1.0           
##  [83] rprojroot_2.0.2             polyclip_1.10-0            
##  [85] matrixStats_0.58.0          lmtest_0.9-38              
##  [87] Matrix_1.4-1                zoo_1.8-8                  
##  [89] beeswarm_0.2.3              base64enc_0.1-3            
##  [91] reprex_1.0.0                ggridges_0.5.3             
##  [93] GlobalOptions_0.1.2         png_0.1-7                  
##  [95] viridisLite_0.3.0           rjson_0.2.20               
##  [97] bitops_1.0-6                visNetwork_2.0.9           
##  [99] pROC_1.17.0.1               KernSmooth_2.23-17         
## [101] DelayedMatrixStats_1.12.3   shape_1.4.5                
## [103] parallelly_1.24.0           jpeg_0.1-8.1               
## [105] S4Vectors_0.28.1            beachmat_2.6.4             
## [107] scales_1.1.1                memoise_2.0.0              
## [109] magrittr_2.0.1              plyr_1.8.6                 
## [111] ica_1.0-2                   RobustRankAggreg_1.1       
## [113] zlibbioc_1.36.0             compiler_4.0.3             
## [115] dqrng_0.2.1                 RColorBrewer_1.1-2         
## [117] clue_0.3-58                 fitdistrplus_1.1-3         
## [119] cli_2.5.0                   XVector_0.30.0             
## [121] listenv_0.8.0               patchwork_1.1.1            
## [123] pbapply_1.4-3               htmlTable_2.1.0            
## [125] Formula_1.2-4               MASS_7.3-53                
## [127] mgcv_1.8-33                 tidyselect_1.1.0           
## [129] stringi_1.5.3               textshaping_0.3.0          
## [131] highr_0.8                   yaml_2.2.1                 
## [133] BiocSingular_1.6.0          locfit_1.5-9.4             
## [135] latticeExtra_0.6-29         grid_4.0.3                 
## [137] sass_0.3.1                  tools_4.0.3                
## [139] future.apply_1.7.0          parallel_4.0.3             
## [141] circlize_0.4.12             rstudioapi_0.13            
## [143] foreign_0.8-80              bluster_1.0.0              
## [145] foreach_1.5.1               gridExtra_2.3              
## [147] prodlim_2019.11.13          farver_2.0.3               
## [149] Rtsne_0.15                  digest_0.6.27              
## [151] BiocManager_1.30.10         lava_1.6.9                 
## [153] shiny_1.6.0                 Rcpp_1.0.6                 
## [155] GenomicRanges_1.42.0        broom_0.7.8                
## [157] scuttle_1.0.4               later_1.1.0.1              
## [159] RcppAnnoy_0.0.18            httr_1.4.2                 
## [161] ComplexHeatmap_2.6.2        colorspace_2.0-0           
## [163] rvest_0.3.6                 fs_1.5.0                   
## [165] tensor_1.5                  reticulate_1.20            
## [167] IRanges_2.24.1              splines_4.0.3              
## [169] uwot_0.1.10                 statmod_1.4.35             
## [171] OmnipathR_3.3.21            spatstat.utils_2.1-0       
## [173] scater_1.18.5               pkgdown_1.6.1              
## [175] plotly_4.9.3                systemfonts_1.0.1          
## [177] xtable_1.8-4                jsonlite_1.7.2             
## [179] timeDate_3043.102           ipred_0.9-11               
## [181] R6_2.5.0                    Hmisc_4.4-2                
## [183] pillar_1.6.1                htmltools_0.5.1.1          
## [185] mime_0.10                   glue_1.4.2                 
## [187] fastmap_1.1.0               BiocParallel_1.24.1        
## [189] BiocNeighbors_1.8.2         class_7.3-17               
## [191] codetools_0.2-16            utf8_1.1.4                 
## [193] lattice_0.20-41             bslib_0.2.4                
## [195] spatstat.sparse_2.0-0       logger_0.2.0               
## [197] ggbeeswarm_0.6.0            curl_4.3.2                 
## [199] leiden_0.3.7                survival_3.2-7             
## [201] limma_3.46.0                rmarkdown_2.7              
## [203] desc_1.2.0                  munsell_0.5.0              
## [205] e1071_1.7-4                 GetoptLong_1.0.5           
## [207] GenomeInfoDbData_1.2.4      iterators_1.0.13           
## [209] haven_2.3.1                 reshape2_1.4.4             
## [211] gtable_0.3.0                spatstat.core_2.0-0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
