<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="liana">
<title>LIANA - NicheNet • liana</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="LIANA - NicheNet">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">liana</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/liana_cc2tensor.html">Context Factorisation with tensor-cell2cell</a>
    <a class="dropdown-item" href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    <a class="dropdown-item" href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    <a class="dropdown-item" href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    <a class="dropdown-item" href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    <a class="dropdown-item" href="../articles/liana_tutorial.html">LIANA Tutorial</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/liana/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>LIANA - NicheNet</h1>
                        <h4 data-toc-skip class="author">Martin Garrido
Rodriguez-Cordoba</h4>
            
            <h4 data-toc-skip class="date">2023-02-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/HEAD/vignettes/liana_nichenet.Rmd" class="external-link"><code>vignettes/liana_nichenet.Rmd</code></a></small>
      <div class="d-none name"><code>liana_nichenet.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="before-you-start">Before you start<a class="anchor" aria-label="anchor" href="#before-you-start"></a>
</h2>
<p>This vignette is an example of how to combine LIANA’s hypotheses with
those produced by NicheNet. The biological results that are obtained
here are highly dependent on the hypothesis in mind, which is described
in <a href="https://www.nature.com/articles/s41592-019-0667-5" class="external-link">NicheNet’s
original article</a>. Before you start, we strongly recommend to have a
look to the <a href="https://github.com/saeyslab/nichenetr" class="external-link">NicheNet’s
repository</a>, as the data and the analyses that are run here were
extracted from its vignettes. We acknowledge NicheNet’s authors for the
extensive and reproducible documentation that they provide in this
repository.</p>
</div>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p><a href="https://saezlab.github.io/liana/index.html" class="external-link">LIANA</a>
(LIgand-receptor ANalysis frAmework) is a framework able to prioritize
ligand-receptor interactions from single-cell transcriptomics using
different resources and methods. It allows users to systematically
generate hypotheses about which ligands from a given cell type is
binding receptors on another. In contrast to LIANA, NicheNet aims to
deepen in the intra-cellular mechanisms that connect a ligand with a set
of transcriptional targets, making an extensive usage of prior knowledge
from multiple sources. <strong>LIANA and NicheNet are not mutually
exclusive, but in certain scenarios could be rather complementary, given
that they aim to explore different aspects of inter- and intra-cellular
communication</strong>.</p>
<p><img src="liana_nichenet.svg"></p>
<p>Because of this, in this vignette, we show how to use LIANA in
combination with NicheNet using the data and the biological scenario
described in <a href="https://github.com/saeyslab/nichenetr/blob/master/vignettes/ligand_activity_geneset.md" class="external-link">NicheNet’s
vignette</a>. Briefly, the biological question here is: Which ligands
expressed by cancer-associated fibroblasts (CAFs) can induce a specific
gene program in neighboring malignant cells? (Using data from Puram et
al. 2017).</p>
</div>
<div class="section level2">
<h2 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h2>
<p>We first install NicheNet and load required libraries</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">'ggpubr'</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'ggpubr'</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span>, repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/slowkow/ggrepel" class="external-link">'ggrepel'</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'ggrepel'</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span>, repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://wilkelab.org/cowplot/" class="external-link">'cowplot'</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'cowplot'</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span>, repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://remotes.r-lib.org" class="external-link">'remotes'</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'remotes'</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span>, repos <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/saeyslab/nichenetr" class="external-link">'nichenetr'</a></span><span class="op">)</span><span class="op">)</span> <span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"saeyslab/nichenetr"</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/liana/" class="external-link">liana</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/saeyslab/nichenetr" class="external-link">nichenetr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">600</span><span class="op">)</span> <span class="co"># required to download expression data /w slow connection</span></span></code></pre></div>
<p>Then, we load and prepare the single-cell data, NicheNet’s model
weights, and the gene set of interest. The latest is composed by genes
that are known to participate in the partial epithelial-mesenchymal
transition (p-EMT) program, as defined in <a href="https://github.com/saeyslab/nichenetr/blob/master/vignettes/ligand_activity_geneset.md" class="external-link">NicheNet’s
vignette</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># single-cell expression matrix described in Puram et al. 2017</span></span>
<span><span class="va">hnscc_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/3260758/files/hnscc_expression.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># model weights</span></span>
<span><span class="va">ligand_target_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/3260758/files/ligand_target_matrix.rds"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Furthermore, we restrict the single-cell data to the two cell types
of interest for this example, which are the cancer associated
fibroblasts (CAFs) and the tumor cells.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="va">hnscc_expression</span><span class="op">$</span><span class="va">expression</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">hnscc_expression</span><span class="op">$</span><span class="va">sample_info</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/make.names.html" class="external-link">make.names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter samples based on vignette's information and add cell type</span></span>
<span><span class="va">tumors_remove</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HN10"</span>, <span class="st">"HN"</span>, <span class="st">"HN12"</span>, <span class="st">"HN13"</span>, <span class="st">"HN24"</span>, <span class="st">"HN7"</span>, <span class="st">"HN8"</span>, <span class="st">"HN23"</span><span class="op">)</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">sample_info</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span> <span class="op">!</span><span class="op">(</span><span class="va">tumor</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">tumors_remove</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">Lymph.node</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># fix some cell type identity names</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cell_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">classified..as.cancer.cell</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"Tumor"</span>, <span class="va">non.cancer.cell.type</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Tumor"</span>, <span class="st">"CAF"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># cell ID as rownames</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_info</span><span class="op">$</span><span class="va">cell</span></span>
<span></span>
<span><span class="co"># subset expression to selected cells</span></span>
<span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="va">expression</span><span class="op">[</span><span class="va">sample_info</span><span class="op">$</span><span class="va">cell</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># gene set of interest</span></span>
<span><span class="va">geneset_oi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/3260758/files/pemt_signature.txt"</span><span class="op">)</span>, col_types <span class="op">=</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/cols.html" class="external-link">cols</a></span><span class="op">(</span><span class="op">)</span>, col_names <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="va">.</span><span class="op">[</span><span class="va">.</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ligand_target_matrix</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-liana">Run LIANA<a class="anchor" aria-label="anchor" href="#run-liana"></a>
</h2>
<p>In the first step, we run LIANA to systematically score all the
ligand-receptor interactions between all the cell types included in the
dataset. To do so, we first need to create a Seurat object from
data:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create seurat object</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">expm1</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">.</span>, meta.data <span class="op">=</span> <span class="va">sample_info</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set cell identity to cell type</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">seurat_object</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">cell_type</span></span></code></pre></div>
<p>And then we can execute LIANA using default parameters. After LIANA
execution, we employ the function <code><a href="../reference/liana_aggregate.html">liana_aggregate()</a></code> to
summarize the output of different methods and to obtain a single score
for each interaction.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">liana_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>By default, LIANA will score the ligand-receptor interactions in all
the possible directions within the two cell types of interest. This
includes: Autocrine signaling (e.g. CAFs -&gt; CAFs), CAFs -&gt; Tumor
cells and Tumor cells -&gt; CAFs. As we are only interested in the CAFs
-&gt; Tumor cell direction, we filter the results and visualize the top
50 interactions according to the consensus/aggregate rank across
methods. The aggregate rank itself can be interpreted as the
significance of preferential enrichment for the interactions.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter results to cell types of interest</span></span>
<span><span class="va">caf_tumor_results</span> <span class="op">&lt;-</span> <span class="va">liana_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">source</span> <span class="op">==</span> <span class="st">"CAF"</span> <span class="op">&amp;</span> <span class="va">target</span> <span class="op">==</span> <span class="st">"Tumor"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>ligand<span class="op">=</span><span class="va">ligand.complex</span>, receptor<span class="op">=</span><span class="va">receptor.complex</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter results to top N interactions</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">top_n_caf_tumor</span> <span class="op">&lt;-</span> <span class="va">caf_tumor_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">aggregate_rank</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">n</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_inorder.html" class="external-link">fct_inorder</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ligand</span>, <span class="st">" -&gt; "</span>, <span class="va">receptor</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize median rank</span></span>
<span><span class="va">top_n_caf_tumor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">aggregate_rank</span>, x <span class="op">=</span> <span class="va">id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Interaction"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"LIANA's aggregate rank"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span>, angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_nichenet_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="run-nichenet-using-lianas-ligands">Run NicheNet using LIANA’s ligands<a class="anchor" aria-label="anchor" href="#run-nichenet-using-lianas-ligands"></a>
</h2>
<p>The key aspect of combining LIANA with NicheNet is that we can use
the ligands prioritized by LIANA as the set of potential ligands for
NicheNet. Instead of evaluating all the expressed ligands for which the
receptor is also expressed in the receiver cell type, we will only
explore those that were prioritized by the methods included in LIANA.
Hence, we select the ligands that form the interactions previously
shown.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get ligands and filter to those included in NicheNet's ligand-target matrix</span></span>
<span><span class="va">ligands</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">top_n_caf_tumor</span><span class="op">$</span><span class="va">ligand</span><span class="op">)</span></span>
<span><span class="va">ligands</span> <span class="op">&lt;-</span> <span class="va">ligands</span><span class="op">[</span><span class="va">ligands</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ligand_target_matrix</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">ligands</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "COL1A1" "FN1"    "THBS2"  "CXCL12" "COL4A1" "THBS1"  "THBS4"  "LAMB1" </span></span>
<span><span class="co">##  [9] "IGF1"   "LAMA2"  "COMP"   "LAMB2"</span></span></code></pre>
<p>Before running NicheNet, we also need to define a list of background
genes. To do so, we employ the threshold defined in <a href="https://github.com/saeyslab/nichenetr/blob/master/vignettes/ligand_activity_geneset.md" class="external-link">NicheNet’s
vignette</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">background_genes</span> <span class="op">&lt;-</span> <span class="va">expression</span><span class="op">[</span><span class="va">sample_info</span><span class="op">$</span><span class="va">cell</span><span class="op">[</span><span class="va">sample_info</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">"Tumor"</span><span class="op">]</span>, <span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">2</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fl">10</span><span class="op">*</span><span class="op">(</span><span class="fl">2</span><span class="op">**</span><span class="va">x</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">2</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="va">.</span><span class="op">[</span><span class="va">.</span> <span class="op">&gt;=</span> <span class="fl">4</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>And execute NicheNet to predict the ligand activities using the pEMT
gene set previously mentioned</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nichenet_activities</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/nichenetr/man/predict_ligand_activities.html" class="external-link">predict_ligand_activities</a></span><span class="op">(</span></span>
<span>  geneset <span class="op">=</span> <span class="va">geneset_oi</span>,</span>
<span>  background_expressed_genes <span class="op">=</span> <span class="va">background_genes</span>,</span>
<span>  ligand_target_matrix <span class="op">=</span> <span class="va">ligand_target_matrix</span>, potential_ligands <span class="op">=</span> <span class="va">ligands</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>As a result, we obtain the NicheNet’s activity predictions for the
ligands previously prioritized using LIANA. In a final step, we will
visualize the ligand-receptor scores of LIANA and the ligand activity
score of NicheNet in a single figure.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># prepare data for visualization</span></span>
<span><span class="va">vis_liana_nichenet</span> <span class="op">&lt;-</span> <span class="va">top_n_caf_tumor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">nichenet_activities</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ligand"</span> <span class="op">=</span> <span class="st">"test_ligand"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">pearson</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>ligand <span class="op">=</span> <span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_inorder.html" class="external-link">fct_inorder</a></span><span class="op">(</span><span class="va">ligand</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare NicheNet figure</span></span>
<span><span class="va">nichenet_scores_plot</span> <span class="op">&lt;-</span> <span class="va">vis_liana_nichenet</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">ligand</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">pearson</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">ligand</span>, x <span class="op">=</span> <span class="va">pearson</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"NicheNet"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Pearson's score"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.line.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>,</span>
<span>        plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare LIANA figure</span></span>
<span><span class="va">liana_receptor_heatmap</span> <span class="op">&lt;-</span> <span class="va">vis_liana_nichenet</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">ligand</span>, x <span class="op">=</span> <span class="va">receptor</span>, fill <span class="op">=</span> <span class="va">aggregate_rank</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"LIANA"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Ligand"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Receptor"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>        plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># combine plots</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">liana_receptor_heatmap</span>, <span class="va">nichenet_scores_plot</span>,</span>
<span>          align <span class="op">=</span> <span class="st">"h"</span>, nrow <span class="op">=</span> <span class="fl">1</span>, rel_widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.8</span>,<span class="fl">0.3</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_nichenet_files/figure-html/unnamed-chunk-11-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>In this vignette, we exemplify how to use LIANA’s predictions as
NicheNet’s input. Although both methods are complementary, there is one
point that should not be forgotten: LIANA predicts ligand-receptor
interaction pairs. However, NicheNet score for a given ligand comes from
how likely is to reach a set of given targets from it. A ligand can have
a great Pearson correlation score to regulate a given set of targets,
but we do not actually know if it is mediated by the receptor that we
predicted using LIANA. Given this, <strong>the combination of methods to
predict cell-cell communication (like LIANA) with tools that are able to
model intracellular signaling using prior knowledge (e.g. NicheNet)
constitute a promising approach to deepen in the signaling mechanisms
implicated in the biological process under study.</strong></p>
</div>
<div class="section level2">
<h2 id="r-session-information">R Session information<a class="anchor" aria-label="anchor" href="#r-session-information"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.1.2 (2021-11-01)</span></span>
<span><span class="co">## Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: /home/dbdimitrov/anaconda3/envs/liana4.1/lib/libopenblasp-r0.3.18.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] sp_1.5-0           SeuratObject_4.1.0 Seurat_4.1.1       liana_0.1.11      </span></span>
<span><span class="co">##  [5] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4       </span></span>
<span><span class="co">##  [9] readr_2.1.2        tidyr_1.2.0        tibble_3.1.8       tidyverse_1.3.2   </span></span>
<span><span class="co">## [13] nichenetr_1.1.1    remotes_2.4.2      cowplot_1.1.1      ggrepel_0.9.1     </span></span>
<span><span class="co">## [17] ggpubr_0.4.0       ggplot2_3.3.6      BiocStyle_2.22.0  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rsvd_1.0.5                  Hmisc_4.7-0                </span></span>
<span><span class="co">##   [3] ica_1.0-3                   class_7.3-20               </span></span>
<span><span class="co">##   [5] foreach_1.5.2               lmtest_0.9-40              </span></span>
<span><span class="co">##   [7] rprojroot_2.0.3             crayon_1.5.1               </span></span>
<span><span class="co">##   [9] spatstat.core_2.4-4         MASS_7.3-58                </span></span>
<span><span class="co">##  [11] nlme_3.1-158                backports_1.4.1            </span></span>
<span><span class="co">##  [13] reprex_2.0.1                basilisk_1.9.12            </span></span>
<span><span class="co">##  [15] rlang_1.0.4                 XVector_0.34.0             </span></span>
<span><span class="co">##  [17] caret_6.0-92                ROCR_1.0-11                </span></span>
<span><span class="co">##  [19] readxl_1.4.0                irlba_2.3.5                </span></span>
<span><span class="co">##  [21] limma_3.50.3                scater_1.22.0              </span></span>
<span><span class="co">##  [23] filelock_1.0.2              BiocParallel_1.28.3        </span></span>
<span><span class="co">##  [25] rjson_0.2.21                bit64_4.0.5                </span></span>
<span><span class="co">##  [27] glue_1.6.2                  sctransform_0.3.3          </span></span>
<span><span class="co">##  [29] vipor_0.4.5                 parallel_4.1.2             </span></span>
<span><span class="co">##  [31] spatstat.sparse_3.0-0       BiocGenerics_0.40.0        </span></span>
<span><span class="co">##  [33] spatstat.geom_3.0-3         haven_2.5.0                </span></span>
<span><span class="co">##  [35] tidyselect_1.1.2            SummarizedExperiment_1.24.0</span></span>
<span><span class="co">##  [37] fitdistrplus_1.1-8          zoo_1.8-10                 </span></span>
<span><span class="co">##  [39] xtable_1.8-4                magrittr_2.0.3             </span></span>
<span><span class="co">##  [41] evaluate_0.15               scuttle_1.4.0              </span></span>
<span><span class="co">##  [43] cli_3.3.0                   zlibbioc_1.40.0            </span></span>
<span><span class="co">##  [45] rstudioapi_0.13             miniUI_0.1.1.1             </span></span>
<span><span class="co">##  [47] bslib_0.4.0                 logger_0.2.2               </span></span>
<span><span class="co">##  [49] rpart_4.1.16                shiny_1.7.2                </span></span>
<span><span class="co">##  [51] BiocSingular_1.10.0         xfun_0.31                  </span></span>
<span><span class="co">##  [53] clue_0.3-61                 cluster_2.1.3              </span></span>
<span><span class="co">##  [55] caTools_1.18.2              listenv_0.8.0              </span></span>
<span><span class="co">##  [57] png_0.1-7                   future_1.27.0              </span></span>
<span><span class="co">##  [59] ipred_0.9-13                withr_2.5.0                </span></span>
<span><span class="co">##  [61] bitops_1.0-7                plyr_1.8.7                 </span></span>
<span><span class="co">##  [63] cellranger_1.1.0            hardhat_1.2.0              </span></span>
<span><span class="co">##  [65] e1071_1.7-11                dqrng_0.3.0                </span></span>
<span><span class="co">##  [67] pROC_1.18.0                 pillar_1.8.0               </span></span>
<span><span class="co">##  [69] GlobalOptions_0.1.2         cachem_1.0.6               </span></span>
<span><span class="co">##  [71] fs_1.5.2                    GetoptLong_1.0.5           </span></span>
<span><span class="co">##  [73] DelayedMatrixStats_1.16.0   vctrs_0.4.1                </span></span>
<span><span class="co">##  [75] ellipsis_0.3.2              generics_0.1.3             </span></span>
<span><span class="co">##  [77] lava_1.6.10                 tools_4.1.2                </span></span>
<span><span class="co">##  [79] foreign_0.8-82              beeswarm_0.4.0             </span></span>
<span><span class="co">##  [81] munsell_0.5.0               proxy_0.4-27               </span></span>
<span><span class="co">##  [83] DelayedArray_0.20.0         fastmap_1.1.0              </span></span>
<span><span class="co">##  [85] compiler_4.1.2              abind_1.4-5                </span></span>
<span><span class="co">##  [87] httpuv_1.6.5                plotly_4.10.0              </span></span>
<span><span class="co">##  [89] rgeos_0.5-9                 GenomeInfoDbData_1.2.7     </span></span>
<span><span class="co">##  [91] prodlim_2019.11.13          gridExtra_2.3              </span></span>
<span><span class="co">##  [93] edgeR_3.36.0                lattice_0.20-45            </span></span>
<span><span class="co">##  [95] dir.expiry_1.2.0            deldir_1.0-6               </span></span>
<span><span class="co">##  [97] visNetwork_2.1.0            utf8_1.2.2                 </span></span>
<span><span class="co">##  [99] later_1.3.0                 recipes_1.0.1              </span></span>
<span><span class="co">## [101] jsonlite_1.8.0              scales_1.2.0               </span></span>
<span><span class="co">## [103] ScaledMatrix_1.2.0          pbapply_1.5-0              </span></span>
<span><span class="co">## [105] carData_3.0-5               sparseMatrixStats_1.6.0    </span></span>
<span><span class="co">## [107] lazyeval_0.2.2              promises_1.2.0.1           </span></span>
<span><span class="co">## [109] car_3.1-0                   doParallel_1.0.17          </span></span>
<span><span class="co">## [111] latticeExtra_0.6-30         goftest_1.2-3              </span></span>
<span><span class="co">## [113] spatstat.utils_3.0-1        reticulate_1.25            </span></span>
<span><span class="co">## [115] checkmate_2.1.0             rmarkdown_2.14             </span></span>
<span><span class="co">## [117] pkgdown_2.0.6               textshaping_0.3.6          </span></span>
<span><span class="co">## [119] statmod_1.4.36              Rtsne_0.16                 </span></span>
<span><span class="co">## [121] Biobase_2.54.0              uwot_0.1.11                </span></span>
<span><span class="co">## [123] igraph_1.3.0                survival_3.3-1             </span></span>
<span><span class="co">## [125] yaml_2.3.5                  systemfonts_1.0.4          </span></span>
<span><span class="co">## [127] htmltools_0.5.3             memoise_2.0.1              </span></span>
<span><span class="co">## [129] locfit_1.5-9.6              IRanges_2.28.0             </span></span>
<span><span class="co">## [131] viridisLite_0.4.0           digest_0.6.29              </span></span>
<span><span class="co">## [133] assertthat_0.2.1            mime_0.12                  </span></span>
<span><span class="co">## [135] rappdirs_0.3.3              future.apply_1.9.0         </span></span>
<span><span class="co">## [137] data.table_1.14.2           S4Vectors_0.32.4           </span></span>
<span><span class="co">## [139] ragg_1.2.2                  DiagrammeR_1.0.9           </span></span>
<span><span class="co">## [141] labeling_0.4.2              splines_4.1.2              </span></span>
<span><span class="co">## [143] Formula_1.2-4               googledrive_2.0.0          </span></span>
<span><span class="co">## [145] RCurl_1.98-1.7              broom_1.0.0                </span></span>
<span><span class="co">## [147] hms_1.1.1                   modelr_0.1.8               </span></span>
<span><span class="co">## [149] colorspace_2.0-3            base64enc_0.1-3            </span></span>
<span><span class="co">## [151] BiocManager_1.30.18         ggbeeswarm_0.6.0           </span></span>
<span><span class="co">## [153] GenomicRanges_1.46.1        shape_1.4.6                </span></span>
<span><span class="co">## [155] nnet_7.3-17                 sass_0.4.2                 </span></span>
<span><span class="co">## [157] Rcpp_1.0.8.3                bookdown_0.27              </span></span>
<span><span class="co">## [159] RANN_2.6.1                  circlize_0.4.15            </span></span>
<span><span class="co">## [161] fansi_1.0.3                 tzdb_0.3.0                 </span></span>
<span><span class="co">## [163] parallelly_1.32.1           ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">## [165] R6_2.5.1                    grid_4.1.2                 </span></span>
<span><span class="co">## [167] ggridges_0.5.3              lifecycle_1.0.1            </span></span>
<span><span class="co">## [169] bluster_1.4.0               curl_4.3.2                 </span></span>
<span><span class="co">## [171] ggsignif_0.6.3              googlesheets4_1.0.0        </span></span>
<span><span class="co">## [173] leiden_0.4.2                jquerylib_0.1.4            </span></span>
<span><span class="co">## [175] Matrix_1.4-1                desc_1.4.1                 </span></span>
<span><span class="co">## [177] RcppAnnoy_0.0.19            RColorBrewer_1.1-3         </span></span>
<span><span class="co">## [179] iterators_1.0.14            gower_1.0.0                </span></span>
<span><span class="co">## [181] htmlwidgets_1.5.4           beachmat_2.10.0            </span></span>
<span><span class="co">## [183] polyclip_1.10-0             rvest_1.0.2                </span></span>
<span><span class="co">## [185] ComplexHeatmap_2.10.0       mgcv_1.8-40                </span></span>
<span><span class="co">## [187] globals_0.15.1              htmlTable_2.4.1            </span></span>
<span><span class="co">## [189] patchwork_1.1.1             spatstat.random_3.0-1      </span></span>
<span><span class="co">## [191] progressr_0.10.1            codetools_0.2-18           </span></span>
<span><span class="co">## [193] matrixStats_0.62.0          lubridate_1.8.0            </span></span>
<span><span class="co">## [195] metapod_1.2.0               randomForest_4.7-1.1       </span></span>
<span><span class="co">## [197] prettyunits_1.1.1           SingleCellExperiment_1.16.0</span></span>
<span><span class="co">## [199] dbplyr_2.2.1                basilisk.utils_1.9.4       </span></span>
<span><span class="co">## [201] GenomeInfoDb_1.30.1         gtable_0.3.0               </span></span>
<span><span class="co">## [203] DBI_1.1.3                   stats4_4.1.2               </span></span>
<span><span class="co">## [205] highr_0.9                   tensor_1.5                 </span></span>
<span><span class="co">## [207] httr_1.4.3                  KernSmooth_2.23-20         </span></span>
<span><span class="co">## [209] stringi_1.7.6               vroom_1.5.7                </span></span>
<span><span class="co">## [211] progress_1.2.2              farver_2.1.1               </span></span>
<span><span class="co">## [213] reshape2_1.4.4              viridis_0.6.2              </span></span>
<span><span class="co">## [215] fdrtool_1.2.17              timeDate_4021.104          </span></span>
<span><span class="co">## [217] xml2_1.3.3                  BiocNeighbors_1.12.0       </span></span>
<span><span class="co">## [219] OmnipathR_3.5.5             interp_1.1-3               </span></span>
<span><span class="co">## [221] scattermore_0.8             bit_4.0.4                  </span></span>
<span><span class="co">## [223] scran_1.22.1                jpeg_0.1-9                 </span></span>
<span><span class="co">## [225] MatrixGenerics_1.6.0        spatstat.data_3.0-0        </span></span>
<span><span class="co">## [227] pkgconfig_2.0.3             gargle_1.2.0               </span></span>
<span><span class="co">## [229] rstatix_0.7.0               knitr_1.39</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
