<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Context Factorisation with tensor-cell2cell • liana</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Context Factorisation with tensor-cell2cell">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/liana_cc2tensor.html">Context Factorisation with tensor-cell2cell</a>
    </li>
    <li>
      <a href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_ortho.html"> LIANA Orthology Conversion</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/liana/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org" class="external-link">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="liana_cc2tensor_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Context Factorisation with tensor-cell2cell</h1>
                        <h4 data-toc-skip class="author">Daniel Dimitrov</h4>
            <address class="author_afil">
      Saezlab, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:daniel.dimitrov@uni-heidelberg.de" class="email">daniel.dimitrov@uni-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2022-11-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/HEAD/vignettes/liana_cc2tensor.Rmd" class="external-link"><code>vignettes/liana_cc2tensor.Rmd</code></a></small>
      <div class="hidden name"><code>liana_cc2tensor.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Tensor decomposition as proposed in the <a href="https://www.nature.com/articles/s41467-022-31369-2#ref-CR5" class="external-link">tensor_cell2cell</a> paper, enables us to decipher context-driven intercellular communication by simultaneously accounting for an unlimited number of “contexts”. These contexts could represent samples coming from longtidinal sampling points, multiple conditions, or cellular niches.</p>
<p>The power of tensor-cell2cell is in its ability to decompose latent patterns of intercellular communication in an untargeted manner, in theory being able to handle cell-cell communication results coming from any experimental design, regardless of its complexity.</p>
<p>Simply put, tensor_cell2cell uses LIANA’s output <code>by sample</code> to build a 4D tensor, represented by 1) contexts, 2) interactions, 3) sender, and 4) receiver cell types. This tensor is then decomposed into a set of factors, which can be interpreted as low dimensionality latent variables (vectors) that capture the CCC patterns across contexts.</p>
<p>Here, we will combine <code>LIANA</code> with <code>tensor_cell2cell</code> to decipher potential ligand-receptor interaction changes. As a simple example, we will look at ~14000 PBMCs from 8 donors, each before and after IFN-beta stimulation (GSE96583; obtained via <a href="https://bioconductor.org/packages/release/bioc/html/ExperimentHub.html" class="external-link">ExperimentHub</a> &amp; <a href="https://bioconductor.org/packages/release/data/experiment/html/muscData.html" class="external-link">muscData</a>). Note that by focusing on PBMCs, for the purpose of this tutorial, we assume that coordinated events occur among them.</p>
<p>This tutorial was heavily influenced by the <a href="https://earmingol.github.io/cell2cell/tutorials/ASD/01-Tensor-Factorization-ASD/" class="external-link">tutorials of tensor_cell2cell</a>.</p>
<p>Any usage of <code>liana</code> x <code>tensor_cell2cell</code> should logically cite both articles, and in particular <code>tensor_cell2cell</code> (see reference at the bottom).</p>
<p>Load required libraries</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/liana/" class="external-link">liana</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="request-data-and-preprocess">Request Data and Preprocess<a class="anchor" aria-label="anchor" href="#request-data-and-preprocess"></a>
</h3>
<div class="section level5">
<h5 id="request-data">Request Data<a class="anchor" aria-label="anchor" href="#request-data"></a>
</h5>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># Get Data</span></span>
<span><span class="op">(</span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH2259"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SingleCellExperiment </span></span>
<span><span class="co">## dim: 35635 29065 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(35635): MIR1302-10 FAM138A ... MT-ND6 MT-CYB</span></span>
<span><span class="co">## rowData names(2): ENSEMBL SYMBOL</span></span>
<span><span class="co">## colnames(29065): AAACATACAATGCC-1 AAACATACATTTCC-1 ... TTTGCATGGTTTGG-1</span></span>
<span><span class="co">##   TTTGCATGTCTTAC-1</span></span>
<span><span class="co">## colData names(5): ind stim cluster cell multiplets</span></span>
<span><span class="co">## reducedDimNames(1): TSNE</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span></code></pre>
</div>
<div class="section level5">
<h5 id="preprocess">Preprocess<a class="anchor" aria-label="anchor" href="#preprocess"></a>
</h5>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># basic feature filtering</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">5</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># basic outlier filtering</span></span>
<span><span class="va">qc</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html" class="external-link">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove cells with few or many detected genes</span></span>
<span><span class="va">ol</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">isOutlier</a></span><span class="op">(</span>metric <span class="op">=</span> <span class="va">qc</span><span class="op">$</span><span class="va">detected</span>, nmads <span class="op">=</span> <span class="fl">2</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="op">!</span><span class="va">ol</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Remove doublets</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="va">sce</span><span class="op">$</span><span class="va">multiplets</span><span class="op">==</span><span class="st">"singlet"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Set rownames to symbols</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span></span>
<span></span>
<span><span class="co"># log-transform</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a label unique for every sample</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">context</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">stim</span>, <span class="va">sce</span><span class="op">$</span><span class="va">ind</span>, sep<span class="op">=</span><span class="st">"|"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="ensure-consitency-across-cell-identities">Ensure Consitency across Cell identities<a class="anchor" aria-label="anchor" href="#ensure-consitency-across-cell-identities"></a>
</h5>
<p>To obtain consistent CCC patterns across samples, we need to make sure that the cell identities are stable. Namely,</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot</span></span>
<span><span class="va">sce</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/get_abundance_summary.html">get_abundance_summary</a></span><span class="op">(</span>sample_col <span class="op">=</span> <span class="st">"context"</span>,</span>
<span>                        idents_col <span class="op">=</span> <span class="st">"cell"</span>, </span>
<span>                        min_cells <span class="op">=</span> <span class="fl">10</span>, <span class="co"># min cells per sample</span></span>
<span>                        min_samples <span class="op">=</span> <span class="fl">3</span>, <span class="co"># min samples</span></span>
<span>                        min_prop <span class="op">=</span> <span class="fl">0.2</span> <span class="co"># min prop of samples</span></span>
<span>                        <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_abundance_summary.html">plot_abundance_summary</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-4-1.png" width="1536"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter non abundant celltypes</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">liana</span><span class="fu">::</span><span class="fu"><a href="../reference/filter_nonabundant_celltypes.html">filter_nonabundant_celltypes</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>                                           sample_col <span class="op">=</span> <span class="st">"context"</span>,</span>
<span>                                           idents_col <span class="op">=</span> <span class="st">"cell"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="run-liana-for-on-each-individual-sample-">Run liana for on each individual sample.<a class="anchor" aria-label="anchor" href="#run-liana-for-on-each-individual-sample-"></a>
</h3>
<p>In order to construct the tensor, we first need to obtain CCC predictions for each sample. In this case, we use SingleCellSignalR scores, as they are regularized, and theory directly comparable between any dataset. One can use any method with <strong>non-negative scores</strong> from LIANA /w cell2cell_tensor as they were previously shown to yield consistent results (Armingol &amp; Baghdassarian, 2022).</p>
<p>Note that <code>liana_bysample</code> works with <code>SingleCellExperiment</code> alone, if you wish to use Seurat, please use the <code>as.SingleCellExperiment</code> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run LIANA by sample</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_bysample.html">liana_bysample</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>                      sample_col <span class="op">=</span> <span class="st">"context"</span>,</span>
<span>                      idents_col <span class="op">=</span> <span class="st">"cell"</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"sca"</span>, <span class="co"># we use SingleCellSignalR's score alone</span></span>
<span>                      expr_prop <span class="op">=</span> <span class="fl">0</span>, <span class="co"># expression proportion threshold</span></span>
<span>                      inplace<span class="op">=</span><span class="cn">TRUE</span> <span class="co"># saves to sce</span></span>
<span>                      <span class="op">)</span></span></code></pre></div>
<p>We expect to see a successful LIANA run for each sample/context.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">liana_res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           Length Class  Mode</span></span>
<span><span class="co">## ctrl|101  12     tbl_df list</span></span>
<span><span class="co">## ctrl|1015 12     tbl_df list</span></span>
<span><span class="co">## ctrl|1016 12     tbl_df list</span></span>
<span><span class="co">## ctrl|1039 12     tbl_df list</span></span>
<span><span class="co">## ctrl|107  12     tbl_df list</span></span>
<span><span class="co">## ctrl|1244 12     tbl_df list</span></span>
<span><span class="co">## ctrl|1256 12     tbl_df list</span></span>
<span><span class="co">## ctrl|1488 12     tbl_df list</span></span>
<span><span class="co">## stim|101  12     tbl_df list</span></span>
<span><span class="co">## stim|1015 12     tbl_df list</span></span>
<span><span class="co">## stim|1016 12     tbl_df list</span></span>
<span><span class="co">## stim|1039 12     tbl_df list</span></span>
<span><span class="co">## stim|107  12     tbl_df list</span></span>
<span><span class="co">## stim|1244 12     tbl_df list</span></span>
<span><span class="co">## stim|1256 12     tbl_df list</span></span>
<span><span class="co">## stim|1488 12     tbl_df list</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="cell-cell-communication-tensor-decomposition">Cell-cell Communication Tensor Decomposition<a class="anchor" aria-label="anchor" href="#cell-cell-communication-tensor-decomposition"></a>
</h2>
<p>Here, we call tensor_cell2cell.</p>
<p>This function will first format the ligand-receptor scores per sample into a 4 Dimensional tensor.</p>
<p>It will then estimate the number of factors to which the tensor will be decomposed (set rank to <code>NULL</code>, for the sake of computational speed, I have pre-calculated the <code>rank</code> and I explicitly set it to 7 here). Optimal rank estimation can be computationally demanding, thus in this case the number ranks was pre-determined.</p>
<p>One can think of this as a higher-order non-negative matrix factorization where the factors can be used to reconstruct the tensor. We refer the user to the publication of <a href="https://www.nature.com/articles/s41467-022-31369-2" class="external-link">tensor-cell2cell</a> for further information.</p>
<p>Note that by default LIANA will set up a conda environment with <code>basilisk</code> (if conda_env is NULL), the user can alternatively specify the name of a <code>conda_env</code> with cell2cell to be called via reticulate.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/liana_tensor_c2c.html">liana_tensor_c2c</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>                        score_col <span class="op">=</span> <span class="st">'LRscore'</span>,</span>
<span>                        rank <span class="op">=</span> <span class="fl">7</span>,  <span class="co"># set to None to estimate!</span></span>
<span>                        how<span class="op">=</span><span class="st">'outer'</span>  <span class="co">#  defines how the tensor is built</span></span>
<span>                        <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Setting up Conda Environment with Basilisk</span></span></code></pre>
<pre><code><span><span class="co">## + '/home/runner/.cache/R/basilisk/1.11.2/0/bin/conda' 'create' '--yes' '--prefix' '/home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell' 'python=3.8.8' '--quiet' '-c' 'conda-forge'</span></span></code></pre>
<pre><code><span><span class="co">## + '/home/runner/.cache/R/basilisk/1.11.2/0/bin/conda' 'install' '--yes' '--prefix' '/home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell' 'python=3.8.8'</span></span></code></pre>
<pre><code><span><span class="co">## + '/home/runner/.cache/R/basilisk/1.11.2/0/bin/conda' 'install' '--yes' '--prefix' '/home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell' '-c' 'conda-forge' 'python=3.8.8' 'python=3.8.8' 'pandas=1.4.2' 'rpy2=3.4.5'</span></span></code></pre>
<pre><code><span><span class="co">## Building the tensor...</span></span></code></pre>
<pre><code><span><span class="co">## Decomposing the tensor...</span></span></code></pre>
<p>The <code>how</code> parameter plays is essential in how we treat missing cell types and interactions. In this scenario, we use <code>outer</code>, which will only decompose CCC across <em>any</em> cell identities and interactions in the dataset. Alternatively, one could change it to ‘inner’ then CCC paterns will be decomposed patterns across cell identities &amp; interactions shared between <em>all</em> samples/contexts.</p>
<p><code>cell2cell_tensor</code> also accepts additional parameters to further tune tensor decomposition, but these are out of scope for this tutorial. Stay tuned for a set of tutorials with comprehensive application of liana with tensor!</p>
<div class="section level3">
<h3 id="results-overview">Results Overview<a class="anchor" aria-label="anchor" href="#results-overview"></a>
</h3>
<p>Upon a seccusseful run cell2cell_tensor will decompose the tensor into a set of factors, or four vectors corresponding to the initial dimensions of the tensor.</p>
<ul>
<li>
<code>contexts</code> - the factor scores assigned to each sample/context</li>
<li>
<code>interactions</code> - the interaction loadings per factor</li>
<li>
<code>senders</code> - loadings for sender/source cell identities</li>
<li>
<code>receivers</code> - loadings for receivers/target cell identities</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the factors</span></span>
<span><span class="va">factors</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_c2c_factors.html">get_c2c_factors</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show them</span></span>
<span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">factors</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 4</span></span>
<span><span class="co">##  $ contexts    : tibble [16 × 8] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">##   ..$ context : Factor w/ 16 levels "ctrl|101","ctrl|1015",..: 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   ..$ Factor.1: num [1:16] 0.246 0.281 0.237 0.221 0.232 ...</span></span>
<span><span class="co">##   ..$ Factor.2: num [1:16] 0.232 0.241 0.236 0.26 0.245 ...</span></span>
<span><span class="co">##   ..$ Factor.3: num [1:16] 0.255 0.294 0.29 0.178 0.202 ...</span></span>
<span><span class="co">##   ..$ Factor.4: num [1:16] 0.01122 0.03064 0.06389 0.00963 0.0202 ...</span></span>
<span><span class="co">##   ..$ Factor.5: num [1:16] 0.29 0.383 0.325 0.255 0.255 ...</span></span>
<span><span class="co">##   ..$ Factor.6: num [1:16] 0.28 0.329 0.289 0.238 0.251 ...</span></span>
<span><span class="co">##   ..$ Factor.7: num [1:16] 0.246 0.336 0.288 0.197 0.171 ...</span></span>
<span><span class="co">##   ..- attr(*, "pandas.index")=Index(['ctrl|101', 'ctrl|1015', 'ctrl|1016', 'ctrl|1039', 'ctrl|107',</span></span>
<span><span class="co">##        'ctrl|1244', 'ctrl|1256', 'ctrl|1488', 'stim|101', 'stim|1015',</span></span>
<span><span class="co">##        'stim|1016', 'stim|1039', 'stim|107', 'stim|1244', 'stim|1256',</span></span>
<span><span class="co">##        'stim|1488'],</span></span>
<span><span class="co">##       dtype='object')</span></span>
<span><span class="co">##  $ interactions: tibble [1,599 × 8] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">##   ..$ lr      : Factor w/ 1599 levels "A2M^LRP1","ACTR2^ADRB2",..: 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   ..$ Factor.1: num [1:1599] 5.68e-03 9.65e-03 2.61e-02 8.13e-31 3.19e-02 ...</span></span>
<span><span class="co">##   ..$ Factor.2: num [1:1599] 1.66e-08 8.15e-03 2.86e-02 9.22e-02 3.69e-20 ...</span></span>
<span><span class="co">##   ..$ Factor.3: num [1:1599] 7.80e-05 5.51e-02 7.81e-16 1.41e-02 1.11e-39 ...</span></span>
<span><span class="co">##   ..$ Factor.4: num [1:1599] 3.39e-09 5.40e-02 2.34e-02 1.31e-03 2.14e-02 ...</span></span>
<span><span class="co">##   ..$ Factor.5: num [1:1599] 1.87e-03 2.69e-02 1.38e-06 5.99e-13 1.98e-06 ...</span></span>
<span><span class="co">##   ..$ Factor.6: num [1:1599] 1.34e-09 3.12e-02 2.82e-03 1.24e-02 6.31e-08 ...</span></span>
<span><span class="co">##   ..$ Factor.7: num [1:1599] 1.98e-11 4.17e-02 5.93e-03 3.53e-03 4.57e-03 ...</span></span>
<span><span class="co">##   ..- attr(*, "pandas.index")=Index(['A2M^LRP1', 'ACTR2^ADRB2', 'ACTR2^LDLR', 'ADAM10^AXL', 'ADAM10^CADM1',</span></span>
<span><span class="co">##        'ADAM10^CD44', 'ADAM10^GPNMB', 'ADAM10^IL6R', 'ADAM10^MET',</span></span>
<span><span class="co">##        'ADAM10^NOTCH1',</span></span>
<span><span class="co">##        ...</span></span>
<span><span class="co">##        'WNT7A^FZD9_LRP6', 'WNT7A^LDLR', 'WNT7A^RECK', 'YBX1^NOTCH1',</span></span>
<span><span class="co">##        'ZG16B^CXCR4', 'ZG16B^TLR2', 'ZG16B^TLR4', 'ZG16B^TLR5', 'ZG16B^TLR6',</span></span>
<span><span class="co">##        'ZP3^MERTK'],</span></span>
<span><span class="co">##       dtype='object', length=1599)</span></span>
<span><span class="co">##  $ senders     : tibble [8 × 8] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">##   ..$ celltype: Factor w/ 8 levels "B cells","CD14+ Monocytes",..: 1 2 3 4 5 6 7 8</span></span>
<span><span class="co">##   ..$ Factor.1: num [1:8] 0.356 0.392 0.333 0.323 0.365 ...</span></span>
<span><span class="co">##   ..$ Factor.2: num [1:8] 0.394 0.367 0.325 0.307 0.4 ...</span></span>
<span><span class="co">##   ..$ Factor.3: num [1:8] 0.389 0.354 0.358 0.318 0.369 ...</span></span>
<span><span class="co">##   ..$ Factor.4: num [1:8] 0.39 0.392 0.316 0.28 0.403 ...</span></span>
<span><span class="co">##   ..$ Factor.5: num [1:8] 0.47 0.291 0.399 0.26 0.392 ...</span></span>
<span><span class="co">##   ..$ Factor.6: num [1:8] 1.18e-02 7.72e-01 1.21e-02 2.32e-06 1.80e-01 ...</span></span>
<span><span class="co">##   ..$ Factor.7: num [1:8] 0.10381 0.02506 0.4003 0.58482 0.00678 ...</span></span>
<span><span class="co">##   ..- attr(*, "pandas.index")=Index(['B cells', 'CD14+ Monocytes', 'CD4 T cells', 'CD8 T cells',</span></span>
<span><span class="co">##        'Dendritic cells', 'FCGR3A+ Monocytes', 'Megakaryocytes', 'NK cells'],</span></span>
<span><span class="co">##       dtype='object')</span></span>
<span><span class="co">##  $ receivers   : tibble [8 × 8] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">##   ..$ celltype: Factor w/ 8 levels "B cells","CD14+ Monocytes",..: 1 2 3 4 5 6 7 8</span></span>
<span><span class="co">##   ..$ Factor.1: num [1:8] 2.22e-06 7.58e-01 7.14e-02 8.84e-05 1.46e-01 ...</span></span>
<span><span class="co">##   ..$ Factor.2: num [1:8] 0.00846 0.1065 0.74094 0.49777 0.00018 ...</span></span>
<span><span class="co">##   ..$ Factor.3: num [1:8] 0.00863 0.01187 0.01306 0.49121 0.00252 ...</span></span>
<span><span class="co">##   ..$ Factor.4: num [1:8] 0.468 0.327 0.277 0.263 0.49 ...</span></span>
<span><span class="co">##   ..$ Factor.5: num [1:8] 0.5992 0.3197 0.2067 0.0412 0.4111 ...</span></span>
<span><span class="co">##   ..$ Factor.6: num [1:8] 0.339 0.389 0.393 0.351 0.314 ...</span></span>
<span><span class="co">##   ..$ Factor.7: num [1:8] 0.363 0.388 0.435 0.364 0.272 ...</span></span>
<span><span class="co">##   ..- attr(*, "pandas.index")=Index(['B cells', 'CD14+ Monocytes', 'CD4 T cells', 'CD8 T cells',</span></span>
<span><span class="co">##        'Dendritic cells', 'FCGR3A+ Monocytes', 'Megakaryocytes', 'NK cells'],</span></span>
<span><span class="co">##       dtype='object')</span></span></code></pre>
<p>Here, we examine the behavior of the different dimensions across the factors. When we look at the contexts (samples) loadings <code>Factor.4</code> seems to be notably different between the groups. We also see that the loadings for the “Sender” and “Receiver” cells have a relatively uniform distribution, suggesting that most cell types are involved in the CCC events that distinguish the conditions.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot overview</span></span>
<span><span class="fu"><a href="../reference/plot_c2c_overview.html">plot_c2c_overview</a></span><span class="op">(</span><span class="va">sce</span>, group_col<span class="op">=</span><span class="st">"stim"</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-8-1.png" width="1536"></p>
</div>
<div class="section level3">
<h3 id="statistical-comparison-of-communication-patterns">Statistical comparison of Communication Patterns<a class="anchor" aria-label="anchor" href="#statistical-comparison-of-communication-patterns"></a>
</h3>
<p>Here, we see that the communication patterns (or context loadings) identified statistically significant patterns before and after stimulation.</p>
<p>These factors thus represent differences in the ligand-receptor interactions as well as cell types participating in cell-cell communication before and after IFN-beta stimulation.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get all boxplots</span></span>
<span><span class="va">all_facts_boxes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_context_boxplot.html">plot_context_boxplot</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>                                        group_col <span class="op">=</span> <span class="st">"stim"</span>,</span>
<span>                                        test<span class="op">=</span><span class="st">"t.test"</span>, <span class="co"># applicable only to two groups</span></span>
<span>                                        paired<span class="op">=</span><span class="cn">TRUE</span> <span class="co">#! Is this the case for your data?</span></span>
<span>                                        <span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine all boxplots</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span></span>
<span>    <span class="va">all_facts_boxes</span>,</span>
<span>    ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">'bottom'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.tag <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">'bold'</span>, size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span>
<span>          <span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-9-1.png" width="1056"></p>
</div>
<div class="section level3">
<h3 id="contexts-heatmap">Contexts Heatmap<a class="anchor" aria-label="anchor" href="#contexts-heatmap"></a>
</h3>
<p>Again, we see a clear seperation between STIM and CTRL, further suggesting the relevance of the changes in the inferred interactions.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_context_heat.html">plot_context_heat</a></span><span class="op">(</span><span class="va">sce</span>, group_col <span class="op">=</span> <span class="st">"stim"</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-10-1.png" width="1056"></p>
</div>
</div>
<div class="section level2">
<h2 id="cell-cell-pairs-with-high-potential-of-interaction">Cell-cell pairs with high potential of interaction<a class="anchor" aria-label="anchor" href="#cell-cell-pairs-with-high-potential-of-interaction"></a>
</h2>
<p>Here we use the product of the <code>source</code> and <code>target</code> loadings to generate the heatmap of potential celltype-celltype pair relationships, which contribute to <code>Factor.4</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_c2c_cells.html">plot_c2c_cells</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>               factor_of_int <span class="op">=</span> <span class="st">"Factor.4"</span>,</span>
<span>               name <span class="op">=</span> <span class="st">"Loadings \nProduct"</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/ccpairs-1.png" width="1056"></p>
<div class="section level3">
<h3 id="gini-coefficients-of-factor-specific-communicating-sender-and-receiver-cell-types">Gini Coefficients of Factor-specific Communicating Sender and Receiver cell types<a class="anchor" aria-label="anchor" href="#gini-coefficients-of-factor-specific-communicating-sender-and-receiver-cell-types"></a>
</h3>
<p>Gini coefficients range from 0 to 1, and are measures of dispersion, typically used to measure inequality.</p>
<p>Here, the Gini coefficient is used to measure the imbalance of communication in terms of the cell types.</p>
<p>Gini coefficient of 1 would suggests that there is a single cell type contributing to the communication patterns within a factor, while a value of 0 suggest perfect equality between the cell type.</p>
<p>When we focus on <code>Factor.4</code>, we see that the both source/sender and target/receiver cell types have nearly uniform contributions.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get loadings for source/sender Cell types</span></span>
<span><span class="fu"><a href="../reference/calculate_gini.html">calculate_gini</a></span><span class="op">(</span><span class="va">factors</span><span class="op">$</span><span class="va">senders</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 7 × 2</span></span></span>
<span><span class="co">##   factor     gini</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Factor.1 0.044<span style="text-decoration: underline;">6</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Factor.2 0.067<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Factor.3 0.039<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Factor.4 0.079<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Factor.5 0.145 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Factor.6 0.765 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> Factor.7 0.625</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get loadings for target/receiver Cell types</span></span>
<span><span class="fu"><a href="../reference/calculate_gini.html">calculate_gini</a></span><span class="op">(</span><span class="va">factors</span><span class="op">$</span><span class="va">receivers</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 7 × 2</span></span></span>
<span><span class="co">##   factor     gini</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Factor.1 0.793 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Factor.2 0.699 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Factor.3 0.869 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Factor.4 0.164 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Factor.5 0.350 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Factor.6 0.083<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> Factor.7 0.109</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="lr-loadings-heatmap">LR loadings Heatmap<a class="anchor" aria-label="anchor" href="#lr-loadings-heatmap"></a>
</h2>
<p>We can see the LRs involved across contexts.</p>
<p>Though, perhaps since in this case Factors 4 and 5 are those associated with the stimulation, perhaps it’s best to focus on those.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_lr_heatmap.html">plot_lr_heatmap</a></span><span class="op">(</span><span class="va">sce</span>,  n <span class="op">=</span> <span class="fl">5</span>, cluster_columns<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-13-1.png" width="1056"></p>
</div>
<div class="section level2">
<h2 id="lr-loadings-pathway-enrichment">LR loadings Pathway Enrichment<a class="anchor" aria-label="anchor" href="#lr-loadings-pathway-enrichment"></a>
</h2>
<p>Here, we will use pathways from <a href="https://github.com/saezlab/progeny" class="external-link">PROGENy</a>, together with <a href="https://github.com/saezlab/decoupleR" class="external-link">decoupleR</a>, to do an enrichment analysis on the LR loadings.</p>
<p><code>PROGENy</code> represent a data-driven signatures for 14 pathways, while <code>decoupleR</code> is an framework with multiple enrichment methods.</p>
<p>Since PROGENy was originally intended to be used with genes, we need to reformat it to match ligand-receptor predictions. Namely, <code>generate_lr_geneset</code> will assign a LR to a specific pathway, only if both the ligand and receptor, as well as all their subunits, are associated with the same pathway. And also in the case of weighted gene sets (like progeny), all entities have the same weight sign.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># obtain progeny gene sets</span></span>
<span><span class="va">progeny</span> <span class="op">&lt;-</span> <span class="fu">decoupleR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/decoupleR/reference/get_progeny.html" class="external-link">get_progeny</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="st">'human'</span>, top<span class="op">=</span><span class="fl">5000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">p_value</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># convert to LR sets</span></span>
<span><span class="va">progeny_lr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_lr_geneset.html">generate_lr_geneset</a></span><span class="op">(</span><span class="va">factors</span><span class="op">$</span><span class="va">interactions</span>,</span>
<span>                                  resource <span class="op">=</span> <span class="va">progeny</span><span class="op">)</span></span>
<span></span>
<span><span class="va">progeny_lr</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 941 × 3</span></span></span>
<span><span class="co">##    lr             set         mor</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ACTR2^LDLR     EGFR      1.13 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ADAM10^CADM1   MAPK     -<span style="color: #BB0000;">1.04</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ADAM10^GPNMB   EGFR     -<span style="color: #BB0000;">1.99</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ADAM10^GPNMB   MAPK     -<span style="color: #BB0000;">2.79</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ADAM10^NOTCH1  EGFR     -<span style="color: #BB0000;">0.925</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ADAM10^NOTCH1  MAPK     -<span style="color: #BB0000;">0.925</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ADAM10^NOTCH2  TGFb      1.25 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ADAM10^TSPAN15 MAPK     -<span style="color: #BB0000;">0.925</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ADAM12^ITGB1   Androgen -<span style="color: #BB0000;">0.677</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ADAM12^ITGB1   TGFb      2.15 </span></span>
<span><span class="co">## <span style="color: #949494;"># … with 931 more rows</span></span></span></code></pre>
<div class="section level5">
<h5 id="enrichment-dotplot">Enrichment Dotplot<a class="anchor" aria-label="anchor" href="#enrichment-dotplot"></a>
</h5>
<p>Here, we see that LRs associated with the JAK-STAT gene in progeny are enriched in <code>Factor.4</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># interaction loadings to matrix</span></span>
<span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">factors</span><span class="op">$</span><span class="va">interactions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"lr"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># run enrichment analysis with decoupler</span></span>
<span><span class="co"># (we fit a univariate linear model for each gene set)</span></span>
<span><span class="co"># We don't consider genesets with minsize &lt; 10</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">decoupleR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/decoupleR/reference/run_ulm.html" class="external-link">run_ulm</a></span><span class="op">(</span>mat <span class="op">=</span> <span class="va">mat</span>,</span>
<span>                          network <span class="op">=</span> <span class="va">progeny_lr</span>,</span>
<span>                          .source <span class="op">=</span> <span class="st">"set"</span>,</span>
<span>                          .target <span class="op">=</span> <span class="st">"lr"</span>,</span>
<span>                          minsize<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>p_adj <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">p_value</span>, method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># sig/isnig flag</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>significant <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="st">"signif."</span>, <span class="st">"not"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">source</span>, y<span class="op">=</span><span class="va">condition</span>, shape<span class="op">=</span><span class="va">significant</span>,</span>
<span>             colour<span class="op">=</span><span class="va">score</span>, size<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">p_value</span><span class="op">+</span><span class="fl">1e-36</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_colour_gradient2</a></span><span class="op">(</span>high <span class="op">=</span> <span class="st">"red"</span>, low<span class="op">=</span><span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size_continuous</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">21</span>, <span class="fl">16</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Pathway"</span>,</span>
<span>       y<span class="op">=</span><span class="st">"Factor"</span>,</span>
<span>       colour<span class="op">=</span><span class="st">"Activity"</span></span>
<span>       <span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-15-1.png" width="1152"></p>
</div>
<div class="section level5">
<h5 id="lrs-driving-enrichment-of-jak-stat-in-factor-4">LRs driving enrichment of JAK-STAT in Factor.4<a class="anchor" aria-label="anchor" href="#lrs-driving-enrichment-of-jak-stat-in-factor-4"></a>
</h5>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot LRs associated with Estrogen</span></span>
<span><span class="va">lrs</span> <span class="op">&lt;-</span>  <span class="va">factors</span><span class="op">$</span><span class="va">interactions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">progeny_lr</span>, by<span class="op">=</span><span class="st">"lr"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">set</span><span class="op">==</span><span class="st">"JAK-STAT"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">lr</span>, <span class="va">set</span>, <span class="va">mor</span>, loading <span class="op">=</span> <span class="va">Factor.3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>lr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"\\^"</span><span class="op">)</span><span class="op">)</span>, <span class="st">" -&gt; "</span>, <span class="va">lr</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>weight <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">mor</span> <span class="op">&gt;=</span> <span class="fl">0</span>, <span class="st">"positive"</span>, <span class="st">"negative"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">lrs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># only label those that are &gt; x</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>lr <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">loading</span><span class="op">&gt;=</span><span class="fl">0.001</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">mor</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span>, <span class="va">lr</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">mor</span>, y<span class="op">=</span><span class="va">loading</span>, colour<span class="op">=</span><span class="va">weight</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># label only top 20</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">stat_smooth</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"lm"</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_label_repel</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">lr</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"royalblue3"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Pathway Weight"</span>, y<span class="op">=</span><span class="st">"LR Loading"</span><span class="op">)</span></span></code></pre></div>
<p><img src="liana_cc2tensor_files/figure-html/unnamed-chunk-16-1.png" width="1152"></p>
</div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>To Cite tensor_cell2cell: Armingol, E., Baghdassarian, H.M., Martino, C., Perez-Lopez, A., Aamodt, C., Knight, R. and Lewis, N.E., 2022. Context-aware deconvolution of cell–cell communication with Tensor-cell2cell. Nature Communications, 13(1), pp.1-15.</p>
<p>To Cite LIANA:</p>
<pre><code><span><span class="co">## Warning in person1(given = given[[i]], family = family[[i]], middle =</span></span>
<span><span class="co">## middle[[i]], : It is recommended to use 'given' instead of 'middle'.</span></span>
<span></span>
<span><span class="co">## Warning in person1(given = given[[i]], family = family[[i]], middle =</span></span>
<span><span class="co">## middle[[i]], : It is recommended to use 'given' instead of 'middle'.</span></span>
<span></span>
<span><span class="co">## Warning in person1(given = given[[i]], family = family[[i]], middle =</span></span>
<span><span class="co">## middle[[i]], : It is recommended to use 'given' instead of 'middle'.</span></span>
<span></span>
<span><span class="co">## Warning in person1(given = given[[i]], family = family[[i]], middle =</span></span>
<span><span class="co">## middle[[i]], : It is recommended to use 'given' instead of 'middle'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## To cite liana in publications use:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Dimitrov, D., Türei, D., Garrido-Rodriguez M., Burmedi P.L., Nagai,</span></span>
<span><span class="co">##   J.S., Boys, C., Flores, R.O.R., Kim, H., Szalai, B., Costa, I.G.,</span></span>
<span><span class="co">##   Valdeolivas, A., Dugourd, A. and Saez-Rodriguez, J. Comparison of</span></span>
<span><span class="co">##   methods and resources for cell-cell communication inference from</span></span>
<span><span class="co">##   single-cell RNA-Seq data. Nat Commun 13, 3224 (2022).</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   @Article{,</span></span>
<span><span class="co">##     author = {Daniel Dimitrov and Denes Turei and Martin Garrido-Rodriguez and Paul Burmedi L. and James Nagai S. and Charlotte Boys and Ricardo Ramirez Flores O. and Hyojin Kim and Bence Szalai and Ivan Costa G. and Alberto Valdeolivas and Aurélien Dugourd and Julio Saez-Rodriguez},</span></span>
<span><span class="co">##     title = {Comparison of methods and resources for cell-cell communication inference from single-cell RNA-Seq data},</span></span>
<span><span class="co">##     journal = {Nature Communications},</span></span>
<span><span class="co">##     year = {2022},</span></span>
<span><span class="co">##     doi = {10.1038/s41467-022-30755-0},</span></span>
<span><span class="co">##     encoding = {UTF-8},</span></span>
<span><span class="co">##   }</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">##  os       Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">##  system   x86_64, linux-gnu</span></span>
<span><span class="co">##  ui       X11</span></span>
<span><span class="co">##  language en</span></span>
<span><span class="co">##  collate  C.UTF-8</span></span>
<span><span class="co">##  ctype    C.UTF-8</span></span>
<span><span class="co">##  tz       UTC</span></span>
<span><span class="co">##  date     2022-11-18</span></span>
<span><span class="co">##  pandoc   2.7.3 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  AnnotationDbi            1.58.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationHub          * 3.4.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  assertthat               0.2.1    <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  backports                1.4.1    <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  basilisk                 1.11.2   <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (LTLA/basilisk@be72de1)</span></span></span>
<span><span class="co">##  basilisk.utils           1.9.4    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (LTLA/basilisk.utils@b3ab58d)</span></span></span>
<span><span class="co">##  beachmat                 2.12.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  beeswarm                 0.4.0    <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  Biobase                * 2.56.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocFileCache          * 2.4.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocGenerics           * 0.42.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocManager              1.30.19  <span style="color: #555555;">2022-10-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  BiocNeighbors            1.14.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocParallel             1.30.4   <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocSingular             1.12.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocStyle              * 2.24.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocVersion              3.15.2   <span style="color: #555555;">2022-03-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  Biostrings               2.64.1   <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bit                      4.0.5    <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  bit64                    4.0.5    <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  bitops                   1.0-7    <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  blob                     1.2.3    <span style="color: #555555;">2022-04-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  bluster                  1.6.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bookdown                 0.30     <span style="color: #555555;">2022-11-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  broom                    1.0.1    <span style="color: #555555;">2022-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  bslib                    0.4.1    <span style="color: #555555;">2022-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  cachem                   1.0.6    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  cellranger               1.1.0    <span style="color: #555555;">2016-07-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  checkmate                2.1.0    <span style="color: #555555;">2022-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  circlize                 0.4.15   <span style="color: #555555;">2022-05-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  cli                      3.4.1    <span style="color: #555555;">2022-09-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  clue                     0.3-62   <span style="color: #555555;">2022-10-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  cluster                  2.1.4    <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  codetools                0.2-18   <span style="color: #555555;">2020-11-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  colorspace               2.0-3    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ComplexHeatmap           2.12.1   <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  crayon                   1.5.2    <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  curl                     4.3.3    <span style="color: #555555;">2022-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  DBI                      1.1.3    <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  dbplyr                 * 2.2.1    <span style="color: #555555;">2022-06-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  decoupleR                2.3.2    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (saezlab/decoupleR@56dc1a3)</span></span></span>
<span><span class="co">##  DelayedArray             0.22.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  DelayedMatrixStats       1.18.2   <span style="color: #555555;">2022-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  desc                     1.4.2    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  digest                   0.6.30   <span style="color: #555555;">2022-10-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  dir.expiry               1.4.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  doParallel               1.0.17   <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  dplyr                  * 1.0.10   <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  dqrng                    0.3.0    <span style="color: #555555;">2021-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  edgeR                    3.38.4   <span style="color: #555555;">2022-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ellipsis                 0.3.2    <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  evaluate                 0.18     <span style="color: #555555;">2022-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ExperimentHub          * 2.4.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  fansi                    1.0.3    <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  farver                   2.1.1    <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  fastmap                  1.1.0    <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  filelock                 1.0.2    <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  forcats                * 0.5.2    <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  foreach                  1.5.2    <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  fs                       1.5.2    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  future                   1.29.0   <span style="color: #555555;">2022-11-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  future.apply             1.10.0   <span style="color: #555555;">2022-11-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  gargle                   1.2.1    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  generics                 0.1.3    <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  GenomeInfoDb           * 1.32.4   <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomeInfoDbData         1.2.8    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicRanges          * 1.48.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GetoptLong               1.0.5    <span style="color: #555555;">2020-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ggbeeswarm               0.6.0    <span style="color: #555555;">2017-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ggplot2                * 3.4.0    <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ggrepel                  0.9.2    <span style="color: #555555;">2022-11-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  GlobalOptions            0.1.2    <span style="color: #555555;">2020-06-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  globals                  0.16.1   <span style="color: #555555;">2022-08-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  glue                     1.6.2    <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  googledrive              2.0.0    <span style="color: #555555;">2021-07-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  googlesheets4            1.0.1    <span style="color: #555555;">2022-08-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  gridExtra                2.3      <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  gtable                   0.3.1    <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  haven                    2.5.1    <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  here                     1.0.1    <span style="color: #555555;">2020-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  highr                    0.9      <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  hms                      1.1.2    <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  htmltools                0.5.3    <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  httpuv                   1.6.6    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  httr                     1.4.4    <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  igraph                   1.3.5    <span style="color: #555555;">2022-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  interactiveDisplayBase   1.34.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  IRanges                * 2.30.1   <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  irlba                    2.3.5.1  <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  iterators                1.0.14   <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  jquerylib                0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  jsonlite                 1.8.3    <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  KEGGREST                 1.36.3   <span style="color: #555555;">2022-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  knitr                    1.41     <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  labeling                 0.4.2    <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  later                    1.3.0    <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  lattice                  0.20-45  <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  liana                  * 0.1.8    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">##  lifecycle                1.0.3    <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  limma                    3.52.4   <span style="color: #555555;">2022-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  listenv                  0.8.0    <span style="color: #555555;">2019-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  locfit                   1.5-9.6  <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  logger                   0.2.2    <span style="color: #555555;">2021-10-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  lubridate                1.9.0    <span style="color: #555555;">2022-11-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  magrittr               * 2.0.3    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  Matrix                   1.5-3    <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  MatrixGenerics         * 1.8.1    <span style="color: #555555;">2022-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  matrixStats            * 0.63.0   <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  memoise                  2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  metapod                  1.4.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  mgcv                     1.8-41   <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  mime                     0.12     <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  modelr                   0.1.10   <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  munsell                  0.5.0    <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  muscData               * 1.10.0   <span style="color: #555555;">2022-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  nlme                     3.1-160  <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">##  OmnipathR                3.5.25   <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (saezlab/OmnipathR@2a734a1)</span></span></span>
<span><span class="co">##  parallelly               1.32.1   <span style="color: #555555;">2022-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  patchwork              * 1.1.2    <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  pillar                   1.8.1    <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  pkgconfig                2.0.3    <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  pkgdown                  2.0.6    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  png                      0.1-7    <span style="color: #555555;">2013-12-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  prettyunits              1.1.1    <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  progress                 1.2.2    <span style="color: #555555;">2019-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  progressr                0.11.0   <span style="color: #555555;">2022-09-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  promises                 1.2.0.1  <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  purrr                  * 0.3.5    <span style="color: #555555;">2022-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  R6                       2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ragg                     1.2.4    <span style="color: #555555;">2022-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rappdirs                 0.3.3    <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  RColorBrewer             1.1-3    <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  Rcpp                     1.0.9    <span style="color: #555555;">2022-07-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  RCurl                    1.98-1.9 <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  readr                  * 2.1.3    <span style="color: #555555;">2022-10-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  readxl                   1.4.1    <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  reprex                   2.0.2    <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  reticulate             * 1.26     <span style="color: #555555;">2022-08-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rjson                    0.2.21   <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rlang                    1.0.6    <span style="color: #555555;">2022-09-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rmarkdown                2.18     <span style="color: #555555;">2022-11-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rprojroot                2.0.3    <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  RSQLite                  2.2.18   <span style="color: #555555;">2022-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rsvd                     1.0.5    <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  rvest                    1.0.3    <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  S4Vectors              * 0.34.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  sass                     0.4.2    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  ScaledMatrix             1.4.1    <span style="color: #555555;">2022-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  scales                   1.2.1    <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  scater                   1.24.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  scran                    1.24.1   <span style="color: #555555;">2022-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  scuttle                  1.6.3    <span style="color: #555555;">2022-08-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  sessioninfo              1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  SeuratObject             4.1.3    <span style="color: #555555;">2022-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  shape                    1.4.6    <span style="color: #555555;">2021-05-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  shiny                    1.7.3    <span style="color: #555555;">2022-10-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  SingleCellExperiment   * 1.18.1   <span style="color: #555555;">2022-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  sp                       1.5-1    <span style="color: #555555;">2022-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  sparseMatrixStats        1.8.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  statmod                  1.4.37   <span style="color: #555555;">2022-08-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  stringi                  1.7.8    <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  stringr                * 1.4.1    <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  SummarizedExperiment   * 1.26.1   <span style="color: #555555;">2022-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  systemfonts              1.0.4    <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  textshaping              0.3.6    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  tibble                 * 3.1.8    <span style="color: #555555;">2022-07-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  tidyr                  * 1.2.1    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  tidyselect               1.2.0    <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  tidyverse              * 1.3.2    <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  timechange               0.1.1    <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  tzdb                     0.3.0    <span style="color: #555555;">2022-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  utf8                     1.2.2    <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  vctrs                    0.5.1    <span style="color: #555555;">2022-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  vipor                    0.4.5    <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  viridis                  0.6.2    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  viridisLite              0.4.1    <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  vroom                    1.6.0    <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  withr                    2.5.0    <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  xfun                     0.35     <span style="color: #555555;">2022-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  xml2                     1.3.3    <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  xtable                   1.8-4    <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  XVector                  0.36.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  yaml                     2.3.6    <span style="color: #555555;">2022-10-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">##  zlibbioc                 1.42.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] /home/runner/work/_temp/Library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [2] /opt/R/4.2.2/lib/R/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Python configuration ───────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  python:         /home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell/bin/python</span></span>
<span><span class="co">##  libpython:      /home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell/lib/libpython3.8.so</span></span>
<span><span class="co">##  pythonhome:     /home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell:/home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell</span></span>
<span><span class="co">##  version:        3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)  [GCC 9.3.0]</span></span>
<span><span class="co">##  numpy:          /home/runner/.cache/R/basilisk/1.11.2/liana/0.1.8/liana_cell2cell/lib/python3.8/site-packages/numpy</span></span>
<span><span class="co">##  numpy_version:  1.23.4</span></span>
<span><span class="co">##  </span></span>
<span><span class="co">##  NOTE: Python version was forced by use_python function</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
