<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>LIANA Custom literature and database knowledge from OmniPath • liana</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="LIANA Custom literature and database knowledge from OmniPath">
<meta property="og:description" content="liana">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">liana</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/liana_custom_op.html">LIANA Custom literature and database knowledge from OmniPath</a>
    </li>
    <li>
      <a href="../articles/liana_devel.html">LIANA++ for Benchmarks and Developers</a>
    </li>
    <li>
      <a href="../articles/liana_intracell.html">LIANA with Intracellular Omnipath</a>
    </li>
    <li>
      <a href="../articles/liana_nichenet.html">LIANA - NicheNet</a>
    </li>
    <li>
      <a href="../articles/liana_ortho.html">Orthology Conversion with LIANA</a>
    </li>
    <li>
      <a href="../articles/liana_tutorial.html">LIANA Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/liana/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="liana_custom_op_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>LIANA Custom literature and database knowledge from OmniPath</h1>
                        <h4 class="author">Daniel Dimitrov</h4>
            <address class="author_afil">
      Saezlab, Heidelberg University<br><a class="author_email" href="mailto:#"></a><a href="mailto:daniel.dimitrov@uni-heidelberg.de" class="email">daniel.dimitrov@uni-heidelberg.de</a>
      </address>
                  
            <h4 class="date">2022-05-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/liana/blob/master/vignettes/liana_custom_op.Rmd"><code>vignettes/liana_custom_op.Rmd</code></a></small>
      <div class="hidden name"><code>liana_custom_op.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>All ligand-receptor interaction prediction methods in LIANA require a resource of literature and database knowledge about intercellular signaling. LIANA, together with <a href="https://omnipathdb.org/">OmniPath web service</a>, provide a number functionalities to obtain this prior knowledge.</p>
<p>Here we show how to customize and quality filter the prior knowledge from OmniPath, and how to use LIANA with a custom database resource.</p>
<div id="intercellular-communication-data-in-omnipath" class="section level2">
<h2 class="hasAnchor">
<a href="#intercellular-communication-data-in-omnipath" class="anchor"></a>Intercellular communication data in OmniPath</h2>
<p>OmniPath combines data from more than 20 resources to depict the roles of proteins in intercellular communication, and it uses around 50 network resources to connect these proteins by signaling interactions. In its intercellular communication annotation (intercell) database, OmniPath aims to cover the broadest range of information, and contains numerous false positive records. Hence, for using in prediction methods, as we do in LIANA, quality filtering OmniPath is highly recommended. We do quality filtering based on 1) the amount of evidences (number of resources and references), 2) the consensus between resources for each annotation record, 3) the localization of proteins (e.g. secreted, plasma membrane). The design of the OmniPath database is described in <a href="https://www.embopress.org/doi/full/10.15252/msb.20209923">Turei et al. Mol Syst Biol (2021)17:e9923</a>, while you can read more about the quality filtering options <a href="https://saezlab.github.io/OmnipathR/reference/import_intercell_network.html">here</a>, <a href="https://saezlab.github.io/OmnipathR/reference/filter_intercell_network.html">here</a> and <a href="https://saezlab.github.io/OmnipathR/articles/bioc_workshop.html#intercellular-communication-roles">here</a>.</p>
<p>We further provide a list of curated interactions, which we define as those which come from manually-curated resources in the context of CCC, and come with a corresponding Pubmed ID, available via <a href="https://saezlab.github.io/OmnipathR/reference/curated_ligand_receptor_interactions.html">OmnipathR::curated_ligand_receptor_interactions</a>.</p>
<div id="complexes" class="section level3">
<h3 class="hasAnchor">
<a href="#complexes" class="anchor"></a>Complexes</h3>
<p>Some resources in OmniPath, such as CellChatDB, CellPhoneDB and ICELLNET, provide information about protein complexes in intercellular communication. Besides these, most of the protein complex annotations in OmniPath are in silico inferred, based on the members of the complex. Network interactions of protein complexes are very sparse, as only few resources provide this kind of data.</p>
</div>
<div id="connecting-liana-and-omnipath" class="section level3">
<h3 class="hasAnchor">
<a href="#connecting-liana-and-omnipath" class="anchor"></a>Connecting LIANA and OmniPath</h3>
<p>OmniPath data is distributed by the web service at <a href="https://omnipathdb.org/">OmniPath web service</a>, and the <a href="https://saezlab.github.io/OmnipathR/"><code>OmnipathR</code> R/Bioconductor package</a> offer a direct access in R. In <code>OmnipathR</code>, the <a href="https://saezlab.github.io/OmnipathR/reference/import_intercell_network.html"><code>import_intercell_network</code></a> and <a href="https://saezlab.github.io/OmnipathR/reference/filter_intercell_network.html">filter_intercell_network</a> functions represent the interface to the intercellular communication network. In <code>liana</code>, the <a href="https://saezlab.github.io/liana/reference/generate_omni.html"><code>generate_omni</code></a> function wraps the <code>Omnipath</code> functions and ensures the output is suitable for the downstream methods in LIANA. <a href="https://saezlab.github.io/liana/reference/get_curated_omni.html"><code>get_curated_omni</code></a> function which is used to generate the curated consensus (default) resource that LIANA uses.</p>
<p>We suggest using the latest OmnipathR version for any resource-generation purposes.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'saezlab/OmnipathR'</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="loading-prerequisites" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-prerequisites" class="anchor"></a>Loading prerequisites</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/OmnipathR/">OmnipathR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/liana/">liana</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org">purrr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span>

<span class="va">liana_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"liana"</span><span class="op">)</span>
<span class="va">testdata</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">liana_path</span> , <span class="st">"testdata"</span>, <span class="st">"input"</span>, <span class="st">"testdata.rds"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="curated-omnipath-consensus" class="section level2">
<h2 class="hasAnchor">
<a href="#curated-omnipath-consensus" class="anchor"></a>Curated OmniPath Consensus</h2>
<p>By default, LIANA uses the <code>Consensus</code> resource, which is composed solely from the consensus interactions set coming from the manually-curated (in the context of CCC) “CellPhoneDB”, “CellChatDB”, “ICELLNET”, “connectomeDB2020”, and “CellTalkDB” resources. These interactions are also filtered by localisation and curation.</p>
<p>For more information about how we generate this resource please refer to the <code>get_curated_omni</code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">consensus_omni</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_resource.html">select_resource</a></span><span class="op">(</span><span class="st">"Consensus"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span>
  <span class="va">glimpse</span></code></pre></div>
<pre><code>## Rows: 4,733
## Columns: 10
## $ source                    &lt;chr&gt; "O00182", "O00182", "O00182", "O00182", "O00…
## $ target                    &lt;chr&gt; "P08575", "P08581", "P16070", "Q07954", "Q08…
## $ source_genesymbol         &lt;chr&gt; "LGALS9", "LGALS9", "LGALS9", "LGALS9", "LGA…
## $ target_genesymbol         &lt;chr&gt; "PTPRC", "MET", "CD44", "LRP1", "CD47", "PTP…
## $ category_intercell_source &lt;chr&gt; "cell_surface_ligand", "cell_surface_ligand"…
## $ database_intercell_source &lt;chr&gt; "CellPhoneDB", "CellPhoneDB", "CellPhoneDB",…
## $ category_intercell_target &lt;chr&gt; "receptor", "receptor", "receptor", "recepto…
## $ database_intercell_target &lt;chr&gt; "connectomeDB2020;CellChatDB;CellTalkDB", "c…
## $ sources                   &lt;chr&gt; "CellChatDB;Cellinker", "CellPhoneDB;InnateD…
## $ references                &lt;chr&gt; "CellChatDB:30120235;Cellinker:30120235", ""…</code></pre>
<p>One could also obtain a list with all manually-curated interactions with corresponding PubMed IDs in the context of CCC from OmnipathR. We will use this list to double-check the curation effort of our Consensus resource.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Obtain purely curated resources</span>
<span class="va">curated</span> <span class="op">&lt;-</span> <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/curated_ligand_receptor_interactions.html">curated_ligand_receptor_interactions</a></span><span class="op">(</span>
  cellphonedb <span class="op">=</span> <span class="cn">TRUE</span>,
  cellinker <span class="op">=</span> <span class="cn">TRUE</span>,
  talklr <span class="op">=</span> <span class="cn">TRUE</span>,
  signalink <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">liana</span><span class="fu">:::</span><span class="fu"><a href="../reference/decomplexify.html">decomplexify</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="va">source_genesymbol</span>, <span class="va">target_genesymbol</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>curated_flag <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="va">consensus_omni</span> <span class="op">%&gt;%</span>
  <span class="fu">liana</span><span class="fu">:::</span><span class="fu"><a href="../reference/decomplexify.html">decomplexify</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># dissociates complexes</span>
  <span class="fu">select</span><span class="op">(</span><span class="va">source_genesymbol</span>, <span class="va">target_genesymbol</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">left_join</span><span class="op">(</span><span class="va">curated</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"source_genesymbol"</span>, <span class="st">"target_genesymbol"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>curated_flag <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="fu">replace_na</span><span class="op">(</span><span class="va">curated_flag</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>total <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">group_by</span><span class="op">(</span><span class="va">curated_flag</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>num <span class="op">=</span>  <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="va">curated_flag</span>, <span class="va">total</span>, <span class="va">num</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>perc <span class="op">=</span> <span class="va">num</span><span class="op">/</span><span class="va">total</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 2 x 4
## # Rowwise: 
##   curated_flag total   num  perc
##   &lt;fct&gt;        &lt;int&gt; &lt;int&gt; &lt;dbl&gt;
## 1 1             4859  4425 91.1 
## 2 0             4859   434  8.93</code></pre>
<p>We see that our Consensus resource is highly (&gt;90%) curated.</p>
<p>It is, nevertheless, worth keeping in mind that this check obtains the information from the same resources, so please consider the percentages with a grain of salt :). Yet this approach provides us a reasonable quality reference, applicable to any custom-made resource.</p>
<p><strong>Note that we plan to continuously update this resource, and we welcome any feedback <a href="https://forms.gle/LhTUZjNdiHz8RZZ86">here</a>.</strong></p>
</div>
<div id="surface-or-membrane-bound-interactions-only" class="section level2">
<h2 class="hasAnchor">
<a href="#surface-or-membrane-bound-interactions-only" class="anchor"></a>Surface or Membrane-bound Interactions only</h2>
<p>We could also restrict our query to cell-surface (i.e. membrane bound) interactions, and also alter some further parameters: we set a more stringent threshold for consensus of resources on localization, we include also protein complexes, and we simplify the data frame to keep only the most relevant columns.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pm_omni</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_omni.html">generate_omni</a></span><span class="op">(</span>
    loc_consensus_percentile <span class="op">=</span> <span class="fl">51</span>, <span class="co"># increase localisation consensus threshold</span>
    consensus_percentile <span class="op">=</span> <span class="cn">NULL</span>,
    <span class="co"># include only PM-bound proteins</span>
    transmitter_topology <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
        <span class="st">'plasma_membrane_transmembrane'</span>,
        <span class="st">'plasma_membrane_peripheral'</span>
    <span class="op">)</span>,
    receiver_topology <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
        <span class="st">'plasma_membrane_transmembrane'</span>,
        <span class="st">'plasma_membrane_peripheral'</span>
    <span class="op">)</span>,
    min_curation_effort <span class="op">=</span> <span class="fl">1</span>,
    ligrecextra <span class="op">=</span> <span class="cn">FALSE</span>,
    remove_complexes <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># keep complexes</span>
    simplify <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># do simplify</span>
<span class="op">)</span>


<span class="co"># check categories of ligands (category_intercell_source)</span>
<span class="va">pm_omni</span><span class="op">$</span><span class="va">category_intercell_source</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "activating_cofactor" "adhesion"            "cell_adhesion"      
## [4] "cell_surface_enzyme" "cell_surface_ligand" "ecm"                
## [7] "inhibitory_cofactor" "ligand"</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># and receptors (category_intercell_target)</span>
<span class="va">pm_omni</span><span class="op">$</span><span class="va">category_intercell_target</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "adhesion"          "receptor"          "adherens_junction"
## [4] "cell_adhesion"     "ion_channel"</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># remove some categories</span>
<span class="va">pm_omni</span> <span class="op">%&lt;&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span>
    <span class="op">!</span><span class="va">category_intercell_source</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
        <span class="st">'activating_cofactor'</span>,
        <span class="st">'ligand_regulator'</span>,
        <span class="st">'inhibitory_cofactor'</span>
    <span class="op">)</span>
<span class="op">)</span>
<span class="co"># an overview of the resulted data frame:</span>
<span class="va">pm_omni</span> <span class="op">%&gt;%</span> <span class="fu">glimpse</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Rows: 2,169
## Columns: 17
## $ source                    &lt;chr&gt; "A6NDA9", "A6NI73", "A8K4G0", "A8K4G0", "A8K…
## $ target                    &lt;chr&gt; "Q96LC7", "Q8IYV9", "O15389", "O43699", "O43…
## $ source_genesymbol         &lt;chr&gt; "LRIT2", "LILRA5", "CD300LB", "CD300LB", "CD…
## $ target_genesymbol         &lt;chr&gt; "SIGLEC10", "IZUMO1", "SIGLEC5", "SIGLEC6", …
## $ is_directed               &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…
## $ is_stimulation            &lt;dbl&gt; 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0,…
## $ is_inhibition             &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
## $ consensus_direction       &lt;dbl&gt; 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
## $ consensus_stimulation     &lt;dbl&gt; 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
## $ consensus_inhibition      &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
## $ sources                   &lt;chr&gt; "Cellinker;Wojtowicz2020", "Cellinker;Wojtow…
## $ references                &lt;chr&gt; "Cellinker:32822567", "Cellinker:32822567", …
## $ curation_effort           &lt;dbl&gt; 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 4, 2,…
## $ n_references              &lt;chr&gt; "Cellinker:32822567", "Cellinker:32822567", …
## $ n_resources               &lt;int&gt; 2, 2, 2, 2, 2, 2, 2, 1, 2, 2, 1, 1, 1, 4, 3,…
## $ category_intercell_source &lt;chr&gt; "adhesion", "adhesion", "adhesion", "adhesio…
## $ category_intercell_target &lt;chr&gt; "adhesion", "adhesion", "adhesion", "adhesio…</code></pre>
</div>
<div id="omnipath-resource-as-it-is-in-the-biorxiv" class="section level2">
<h2 class="hasAnchor">
<a href="#omnipath-resource-as-it-is-in-the-biorxiv" class="anchor"></a>OmniPath resource as it is in the Biorxiv</h2>
<p>In the first example, we filter OmniPath as described in the LIANA biorxiv: * Only interactions with literature curation * Interactions only where the receiver protein is plasma membrane transmembrane or peripheral, according to at least 30% of the localisation annotations * Only interactions between single proteins (interactions between complexes were included in that version of OmniPath).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># generate_omni returns a tibble with CCC OmniPath</span>
<span class="va">cust_omni</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_omni.html">generate_omni</a></span><span class="op">(</span>
    loc_consensus_percentile <span class="op">=</span> <span class="fl">30</span>,
    consensus_percentile <span class="op">=</span> <span class="cn">NULL</span>,
    transmitter_topology <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
        <span class="st">'secreted'</span>,
        <span class="st">'plasma_membrane_transmembrane'</span>,
        <span class="st">'plasma_membrane_peripheral'</span>
    <span class="op">)</span>,
    receiver_topology <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
        <span class="st">'plasma_membrane_transmembrane'</span>,
        <span class="st">'plasma_membrane_peripheral'</span>
    <span class="op">)</span>,
    min_curation_effort <span class="op">=</span> <span class="fl">1</span>,
    ligrecextra <span class="op">=</span> <span class="cn">FALSE</span>,
    remove_complexes <span class="op">=</span> <span class="cn">TRUE</span>,
    simplify <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<p>Note that this resource was subsequently improved by manually excluding certain duplicated or ambiguous interactions, and later it was finally deprecated all together. LIANA currently uses the curated resource presented above.</p>
<ul>
<li>Also, available via <code><a href="../reference/select_resource.html">select_resource("OmniPath")</a></code>
</li>
</ul>
</div>
<div id="run-liana-with-a-custom-ccc-resource" class="section level2">
<h2 class="hasAnchor">
<a href="#run-liana-with-a-custom-ccc-resource" class="anchor"></a>Run LIANA with a custom CCC resource</h2>
<div id="run-liana" class="section level3">
<h3 class="hasAnchor">
<a href="#run-liana" class="anchor"></a>RUN liana</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run liana with the custom resource</span>
<span class="co"># liana Wrap</span>
<span class="va">liana_test</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="../reference/liana_wrap.html">liana_wrap</a></span><span class="op">(</span>
        <span class="va">testdata</span>,
        resource<span class="op">=</span><span class="st">'custom'</span>,
        external_resource <span class="op">=</span> <span class="va">pm_omni</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/liana_aggregate.html">liana_aggregate</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in exec(output, ...): 3465 genes and/or 0 cells were removed as they had
## no counts!</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">liana_test</span> <span class="op">%&gt;%</span> <span class="va">glimpse</span></code></pre></div>
<pre><code>## Rows: 929
## Columns: 16
## $ source                 &lt;chr&gt; "B", "B", "B", "B", "B", "B", "B", "B", "B", "B…
## $ target                 &lt;chr&gt; "B", "NK", "B", "B", "B", "B", "B", "NK", "B", …
## $ ligand.complex         &lt;chr&gt; "HLA-DRA", "HLA-DRA", "HLA-DRB1", "HLA-DMA", "H…
## $ receptor.complex       &lt;chr&gt; "CD37", "CD63", "CD37", "CD74", "CD74", "CD82",…
## $ aggregate_rank         &lt;dbl&gt; 2.088734e-08, 1.243603e-07, 4.867595e-07, 1.488…
## $ mean_rank              &lt;dbl&gt; 12.6, 7.8, 19.6, 14.2, 19.8, 38.2, 34.8, 32.6, …
## $ natmi.edge_specificity &lt;dbl&gt; 0.3655873, 0.4806804, 0.3380952, 0.3961174, 0.3…
## $ natmi.rank             &lt;dbl&gt; 54, 28, 68, 46, 47, 11, 17, 56, 21, 9, 26, 19, …
## $ connectome.weight_sc   &lt;dbl&gt; 0.9928272, 1.0496186, 0.9158637, 1.0355778, 0.9…
## $ connectome.rank        &lt;dbl&gt; 5, 1, 11, 2, 7, 13, 18, 27, 12, 3, 21, 23, 4, 2…
## $ logfc.logfc_comb       &lt;dbl&gt; 2.0796425, 2.2218222, 1.6107930, 1.7186762, 1.5…
## $ logfc.rank             &lt;dbl&gt; 2, 1, 9, 6, 13, 16, 5, 29, 4, 3, 22, 28, 7, 8, …
## $ sca.LRscore            &lt;dbl&gt; 0.9442782, 0.9348366, 0.9344024, 0.9293454, 0.9…
## $ sca.rank               &lt;dbl&gt; 1, 8, 9, 16, 31, 150, 133, 50, 77, 73, 113, 181…
## $ cellphonedb.pvalue     &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
## $ cellphonedb.rank       &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…</code></pre>
</div>
</div>
<div id="under-the-hood" class="section level2">
<h2 class="hasAnchor">
<a href="#under-the-hood" class="anchor"></a>Under the hood</h2>
<p>In reality, <code>generate_omni</code> is just a wrapper function that calls the appropriate <code>OmnipathR</code> functions, provide a convenient interface and ensures a liana-appropriate format. Here we give an insight into this process.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reproduce cust_op as from above with OmniPathR alone</span>

<span class="co"># import the OmniPathR intercell network component</span>
<span class="va">ligrec</span> <span class="op">&lt;-</span> <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/import_intercell_network.html">import_intercell_network</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># filter out the complexes</span>
<span class="va">ligrec</span> <span class="op">%&lt;&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span>
    <span class="va">entity_type_intercell_source</span> <span class="op">!=</span> <span class="st">'complex'</span> <span class="op">&amp;</span>
    <span class="va">entity_type_intercell_target</span> <span class="op">!=</span> <span class="st">'complex'</span>
<span class="op">)</span>


<span class="co"># apply filtering according to curation and localisation</span>
<span class="va">ligrec</span> <span class="op">%&lt;&gt;%</span>
    <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/filter_intercell_network.html">filter_intercell_network</a></span><span class="op">(</span>
        loc_consensus_percentile <span class="op">=</span> <span class="fl">30</span>,
        consensus_percentile <span class="op">=</span> <span class="cn">NULL</span>,
        transmitter_topology <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
            <span class="st">'secreted'</span>,
            <span class="st">'plasma_membrane_transmembrane'</span>,
            <span class="st">'plasma_membrane_peripheral'</span>
        <span class="op">)</span>,
        receiver_topology <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
            <span class="st">'plasma_membrane_transmembrane'</span>,
            <span class="st">'plasma_membrane_peripheral'</span>
        <span class="op">)</span>,
        min_curation_effort <span class="op">=</span> <span class="fl">1</span>,
        ligrecextra <span class="op">=</span> <span class="cn">FALSE</span>
    <span class="op">)</span>

<span class="co"># remove duplicate LRs</span>
<span class="va">ligrec</span> <span class="op">%&lt;&gt;%</span>
    <span class="fu">distinct_at</span><span class="op">(</span>
        .vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
            <span class="st">'source_genesymbol'</span>,
            <span class="st">'target_genesymbol'</span>
        <span class="op">)</span>,
        .keep_all <span class="op">=</span> <span class="cn">TRUE</span>
    <span class="op">)</span>

<span class="fu">all_equal</span><span class="op">(</span><span class="va">ligrec</span>, <span class="va">cust_omni</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="further-info" class="section level2">
<h2 class="hasAnchor">
<a href="#further-info" class="anchor"></a>Further Info</h2>
<p>To make use of the full potential of <code>OmnipathR</code> we kindly refer the user to <a href="https://saezlab.github.io/OmnipathR">its documentation</a>.</p>
<p>This tutorial only presents the intercellular component of OmniPath, in the future we will add more details about <a href="https://omnipathdb.org/#about">intracellular signaling and gene regulation</a>.</p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">120</span><span class="op">)</span>
<span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.3 (2020-10-10)
##  os       Ubuntu 20.04.4 LTS          
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language en_US                       
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Europe/Brussels             
##  date     2022-05-16                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date       lib source        
##  abind                  1.4-5    2016-07-21 [2] CRAN (R 4.0.2)
##  assertthat             0.2.1    2019-03-21 [2] CRAN (R 4.0.2)
##  backports              1.2.1    2020-12-09 [2] CRAN (R 4.0.2)
##  beachmat               2.6.4    2020-12-20 [2] Bioconductor  
##  beeswarm               0.2.3    2016-04-25 [2] CRAN (R 4.0.2)
##  Biobase                2.50.0   2020-10-27 [2] Bioconductor  
##  BiocGenerics           0.36.0   2020-10-27 [2] Bioconductor  
##  BiocManager            1.30.10  2019-11-16 [2] CRAN (R 4.0.2)
##  BiocNeighbors          1.8.2    2020-12-07 [2] Bioconductor  
##  BiocParallel           1.24.1   2020-11-06 [2] Bioconductor  
##  BiocSingular           1.6.0    2020-10-27 [2] Bioconductor  
##  BiocStyle            * 2.18.1   2020-11-24 [2] Bioconductor  
##  bit                    4.0.4    2020-08-04 [2] CRAN (R 4.0.2)
##  bit64                  4.0.5    2020-08-30 [2] CRAN (R 4.0.2)
##  bitops                 1.0-6    2013-08-17 [2] CRAN (R 4.0.2)
##  bluster                1.0.0    2020-10-27 [2] Bioconductor  
##  bookdown               0.21     2020-10-13 [2] CRAN (R 4.0.2)
##  broom                  0.7.8    2021-06-24 [2] CRAN (R 4.0.3)
##  bslib                  0.2.4    2021-01-25 [2] CRAN (R 4.0.3)
##  cachem                 1.0.4    2021-02-13 [2] CRAN (R 4.0.3)
##  Cairo                  1.5-12.2 2020-07-07 [2] CRAN (R 4.0.3)
##  cellranger             1.1.0    2016-07-27 [2] CRAN (R 4.0.2)
##  checkmate              2.0.0    2020-02-06 [2] CRAN (R 4.0.3)
##  circlize               0.4.15   2022-05-10 [2] CRAN (R 4.0.3)
##  cli                    2.5.0    2021-04-26 [2] CRAN (R 4.0.3)
##  clue                   0.3-58   2020-12-03 [2] CRAN (R 4.0.3)
##  cluster                2.1.0    2019-06-19 [5] CRAN (R 4.0.0)
##  codetools              0.2-16   2018-12-24 [5] CRAN (R 4.0.0)
##  colorspace             2.0-0    2020-11-11 [2] CRAN (R 4.0.2)
##  ComplexHeatmap         2.6.2    2020-11-12 [2] Bioconductor  
##  cowplot                1.1.1    2020-12-30 [2] CRAN (R 4.0.3)
##  crayon                 1.4.1    2021-02-08 [2] CRAN (R 4.0.3)
##  curl                   4.3.2    2021-06-23 [2] CRAN (R 4.0.3)
##  data.table             1.14.0   2021-02-21 [2] CRAN (R 4.0.3)
##  DBI                    1.1.1    2021-01-15 [2] CRAN (R 4.0.3)
##  dbplyr                 2.1.0    2021-02-03 [2] CRAN (R 4.0.3)
##  DelayedArray           0.16.1   2021-01-22 [2] Bioconductor  
##  DelayedMatrixStats     1.12.3   2021-02-03 [2] Bioconductor  
##  deldir                 0.2-10   2021-02-16 [2] CRAN (R 4.0.3)
##  desc                   1.2.0    2018-05-01 [2] CRAN (R 4.0.2)
##  digest                 0.6.27   2020-10-24 [2] CRAN (R 4.0.2)
##  dplyr                * 1.0.7    2021-06-18 [2] CRAN (R 4.0.3)
##  dqrng                  0.2.1    2019-05-17 [2] CRAN (R 4.0.2)
##  edgeR                  3.32.1   2021-01-14 [2] Bioconductor  
##  ellipsis               0.3.2    2021-04-29 [2] CRAN (R 4.0.3)
##  evaluate               0.15     2022-02-18 [2] CRAN (R 4.0.3)
##  fansi                  0.4.2    2021-01-15 [2] CRAN (R 4.0.3)
##  fastmap                1.1.0    2021-01-25 [2] CRAN (R 4.0.3)
##  fitdistrplus           1.1-3    2020-12-05 [2] CRAN (R 4.0.2)
##  forcats              * 0.5.1    2021-01-27 [2] CRAN (R 4.0.3)
##  fs                     1.5.0    2020-07-31 [2] CRAN (R 4.0.2)
##  future                 1.21.0   2020-12-10 [2] CRAN (R 4.0.3)
##  future.apply           1.7.0    2021-01-04 [2] CRAN (R 4.0.3)
##  generics               0.1.0    2020-10-31 [2] CRAN (R 4.0.2)
##  GenomeInfoDb           1.26.2   2020-12-08 [2] Bioconductor  
##  GenomeInfoDbData       1.2.4    2020-12-14 [2] Bioconductor  
##  GenomicRanges          1.42.0   2020-10-27 [2] Bioconductor  
##  GetoptLong             1.0.5    2020-12-15 [2] CRAN (R 4.0.3)
##  ggbeeswarm             0.6.0    2017-08-07 [2] CRAN (R 4.0.2)
##  ggplot2              * 3.3.5    2021-06-25 [2] CRAN (R 4.0.3)
##  ggrepel                0.9.1    2021-01-15 [2] CRAN (R 4.0.3)
##  ggridges               0.5.3    2021-01-08 [2] CRAN (R 4.0.3)
##  GlobalOptions          0.1.2    2020-06-10 [2] CRAN (R 4.0.3)
##  globals                0.14.0   2020-11-22 [2] CRAN (R 4.0.2)
##  glue                   1.4.2    2020-08-27 [2] CRAN (R 4.0.2)
##  goftest                1.2-2    2019-12-02 [2] CRAN (R 4.0.2)
##  gridExtra              2.3      2017-09-09 [2] CRAN (R 4.0.2)
##  gtable                 0.3.0    2019-03-25 [2] CRAN (R 4.0.2)
##  haven                  2.3.1    2020-06-01 [2] CRAN (R 4.0.2)
##  hms                    1.0.0    2021-01-13 [2] CRAN (R 4.0.3)
##  htmltools              0.5.1.1  2021-01-22 [2] CRAN (R 4.0.3)
##  htmlwidgets            1.5.3    2020-12-10 [2] CRAN (R 4.0.2)
##  httpuv                 1.5.5    2021-01-13 [2] CRAN (R 4.0.3)
##  httr                   1.4.2    2020-07-20 [2] CRAN (R 4.0.2)
##  ica                    1.0-2    2018-05-24 [2] CRAN (R 4.0.2)
##  igraph                 1.3.0    2022-04-01 [2] CRAN (R 4.0.3)
##  IRanges                2.24.1   2020-12-12 [2] Bioconductor  
##  irlba                  2.3.3    2019-02-05 [2] CRAN (R 4.0.2)
##  jquerylib              0.1.3    2020-12-17 [2] CRAN (R 4.0.3)
##  jsonlite               1.7.2    2020-12-09 [2] CRAN (R 4.0.2)
##  KernSmooth             2.23-17  2020-04-26 [5] CRAN (R 4.0.0)
##  knitr                  1.39     2022-04-26 [2] CRAN (R 4.0.3)
##  later                  1.1.0.1  2020-06-05 [2] CRAN (R 4.0.2)
##  lattice                0.20-41  2020-04-02 [5] CRAN (R 4.0.0)
##  lazyeval               0.2.2    2019-03-15 [2] CRAN (R 4.0.2)
##  leiden                 0.3.7    2021-01-26 [2] CRAN (R 4.0.3)
##  liana                * 0.1.3    2022-05-16 [1] Bioconductor  
##  lifecycle              1.0.0    2021-02-15 [2] CRAN (R 4.0.3)
##  limma                  3.46.0   2020-10-27 [2] Bioconductor  
##  listenv                0.8.0    2019-12-05 [2] CRAN (R 4.0.2)
##  lmtest                 0.9-38   2020-09-09 [2] CRAN (R 4.0.2)
##  locfit                 1.5-9.4  2020-03-25 [2] CRAN (R 4.0.2)
##  logger                 0.2.0    2021-03-04 [2] CRAN (R 4.0.3)
##  lubridate              1.7.9.2  2020-11-13 [2] CRAN (R 4.0.2)
##  magrittr             * 2.0.1    2020-11-17 [2] CRAN (R 4.0.2)
##  MASS                   7.3-53   2020-09-09 [5] CRAN (R 4.0.2)
##  Matrix                 1.4-1    2022-03-23 [2] CRAN (R 4.0.3)
##  MatrixGenerics         1.2.1    2021-01-30 [2] Bioconductor  
##  matrixStats            0.58.0   2021-01-29 [2] CRAN (R 4.0.3)
##  memoise                2.0.0    2021-01-26 [2] CRAN (R 4.0.3)
##  mgcv                   1.8-33   2020-08-27 [5] CRAN (R 4.0.2)
##  mime                   0.10     2021-02-13 [2] CRAN (R 4.0.3)
##  miniUI                 0.1.1.1  2018-05-18 [2] CRAN (R 4.0.2)
##  modelr                 0.1.8    2020-05-19 [2] CRAN (R 4.0.2)
##  munsell                0.5.0    2018-06-12 [2] CRAN (R 4.0.2)
##  nlme                   3.1-149  2020-08-23 [5] CRAN (R 4.0.2)
##  OmnipathR            * 3.3.21   2022-04-07 [2] Bioconductor  
##  parallelly             1.24.0   2021-03-14 [2] CRAN (R 4.0.3)
##  patchwork              1.1.1    2020-12-17 [2] CRAN (R 4.0.3)
##  pbapply                1.4-3    2020-08-18 [2] CRAN (R 4.0.2)
##  pillar                 1.6.1    2021-05-16 [2] CRAN (R 4.0.3)
##  pkgconfig              2.0.3    2019-09-22 [2] CRAN (R 4.0.2)
##  pkgdown                1.6.1    2020-09-12 [2] CRAN (R 4.0.3)
##  plotly                 4.9.3    2021-01-10 [2] CRAN (R 4.0.3)
##  plyr                   1.8.6    2020-03-03 [2] CRAN (R 4.0.2)
##  png                    0.1-7    2013-12-03 [2] CRAN (R 4.0.2)
##  polyclip               1.10-0   2019-03-14 [2] CRAN (R 4.0.2)
##  prettyunits            1.1.1    2020-01-24 [2] CRAN (R 4.0.2)
##  progress               1.2.2    2019-05-16 [2] CRAN (R 4.0.2)
##  promises               1.2.0.1  2021-02-11 [2] CRAN (R 4.0.3)
##  purrr                * 0.3.4    2020-04-17 [2] CRAN (R 4.0.2)
##  R6                     2.5.0    2020-10-28 [2] CRAN (R 4.0.2)
##  ragg                   1.1.0    2021-02-15 [2] CRAN (R 4.0.3)
##  RANN                   2.6.1    2019-01-08 [2] CRAN (R 4.0.2)
##  rappdirs               0.3.3    2021-01-31 [2] CRAN (R 4.0.3)
##  RColorBrewer           1.1-2    2014-12-07 [2] CRAN (R 4.0.2)
##  Rcpp                   1.0.6    2021-01-15 [2] CRAN (R 4.0.3)
##  RcppAnnoy              0.0.18   2020-12-15 [2] CRAN (R 4.0.3)
##  RCurl                  1.98-1.3 2021-03-16 [2] CRAN (R 4.0.3)
##  readr                * 2.1.0    2021-11-11 [2] CRAN (R 4.0.3)
##  readxl                 1.3.1    2019-03-13 [2] CRAN (R 4.0.2)
##  reprex                 1.0.0    2021-01-27 [2] CRAN (R 4.0.3)
##  reshape2               1.4.4    2020-04-09 [2] CRAN (R 4.0.2)
##  reticulate             1.20     2021-05-03 [2] CRAN (R 4.0.3)
##  rjson                  0.2.20   2018-06-08 [2] CRAN (R 4.0.3)
##  rlang                  0.4.11   2021-04-30 [2] CRAN (R 4.0.3)
##  rmarkdown              2.7      2021-02-19 [2] CRAN (R 4.0.3)
##  RobustRankAggreg       1.1      2013-06-03 [2] CRAN (R 4.0.3)
##  ROCR                   1.0-11   2020-05-02 [2] CRAN (R 4.0.2)
##  rpart                  4.1-15   2019-04-12 [5] CRAN (R 4.0.0)
##  rprojroot              2.0.2    2020-11-15 [2] CRAN (R 4.0.2)
##  rstudioapi             0.13     2020-11-12 [2] CRAN (R 4.0.2)
##  rsvd                   1.0.3    2020-02-17 [2] CRAN (R 4.0.2)
##  Rtsne                  0.15     2018-11-10 [2] CRAN (R 4.0.2)
##  rvest                  0.3.6    2020-07-25 [2] CRAN (R 4.0.2)
##  S4Vectors              0.28.1   2020-12-09 [2] Bioconductor  
##  sass                   0.3.1    2021-01-24 [2] CRAN (R 4.0.3)
##  scales                 1.1.1    2020-05-11 [2] CRAN (R 4.0.2)
##  scater                 1.18.5   2021-02-16 [2] Bioconductor  
##  scattermore            0.7      2020-11-24 [2] CRAN (R 4.0.3)
##  scran                  1.18.5   2021-02-04 [2] Bioconductor  
##  sctransform            0.3.2    2020-12-16 [2] CRAN (R 4.0.3)
##  scuttle                1.0.4    2020-12-17 [2] Bioconductor  
##  sessioninfo            1.1.1    2018-11-05 [2] CRAN (R 4.0.2)
##  Seurat                 4.0.3    2021-06-10 [2] CRAN (R 4.0.3)
##  SeuratObject           4.0.2    2021-06-09 [2] CRAN (R 4.0.3)
##  shape                  1.4.5    2020-09-13 [2] CRAN (R 4.0.3)
##  shiny                  1.6.0    2021-01-25 [2] CRAN (R 4.0.3)
##  SingleCellExperiment   1.12.0   2020-10-27 [2] Bioconductor  
##  sparseMatrixStats      1.2.1    2021-02-02 [2] Bioconductor  
##  spatstat.core          2.0-0    2021-03-23 [2] CRAN (R 4.0.3)
##  spatstat.data          2.1-0    2021-03-21 [2] CRAN (R 4.0.3)
##  spatstat.geom          2.0-1    2021-03-22 [2] CRAN (R 4.0.3)
##  spatstat.sparse        2.0-0    2021-03-16 [2] CRAN (R 4.0.3)
##  spatstat.utils         2.1-0    2021-03-15 [2] CRAN (R 4.0.3)
##  statmod                1.4.35   2020-10-19 [2] CRAN (R 4.0.2)
##  stringi                1.5.3    2020-09-09 [2] CRAN (R 4.0.2)
##  stringr              * 1.4.0    2019-02-10 [2] CRAN (R 4.0.2)
##  SummarizedExperiment   1.20.0   2020-10-27 [2] Bioconductor  
##  survival               3.2-7    2020-09-28 [2] CRAN (R 4.0.2)
##  systemfonts            1.0.1    2021-02-09 [2] CRAN (R 4.0.3)
##  tensor                 1.5      2012-05-05 [2] CRAN (R 4.0.2)
##  textshaping            0.3.0    2021-02-10 [2] CRAN (R 4.0.3)
##  tibble               * 3.1.2    2021-05-16 [2] CRAN (R 4.0.3)
##  tidyr                * 1.1.3    2021-03-03 [2] CRAN (R 4.0.3)
##  tidyselect             1.1.0    2020-05-11 [2] CRAN (R 4.0.2)
##  tidyverse            * 1.3.0    2019-11-21 [2] CRAN (R 4.0.3)
##  tzdb                   0.1.2    2021-07-20 [2] CRAN (R 4.0.3)
##  utf8                   1.1.4    2018-05-24 [2] CRAN (R 4.0.2)
##  uwot                   0.1.10   2020-12-15 [2] CRAN (R 4.0.3)
##  vctrs                  0.3.8    2021-04-29 [2] CRAN (R 4.0.3)
##  vipor                  0.4.5    2017-03-22 [2] CRAN (R 4.0.2)
##  viridis                0.5.1    2018-03-29 [2] CRAN (R 4.0.2)
##  viridisLite            0.3.0    2018-02-01 [2] CRAN (R 4.0.2)
##  vroom                  1.5.6    2021-11-10 [2] CRAN (R 4.0.3)
##  withr                  2.4.1    2021-01-26 [2] CRAN (R 4.0.3)
##  xfun                   0.30     2022-03-02 [2] CRAN (R 4.0.3)
##  xml2                   1.3.2    2020-04-23 [2] CRAN (R 4.0.2)
##  xtable                 1.8-4    2019-04-21 [2] CRAN (R 4.0.2)
##  XVector                0.30.0   2020-10-27 [2] Bioconductor  
##  yaml                   2.2.1    2020-02-01 [2] CRAN (R 4.0.2)
##  zlibbioc               1.36.0   2020-10-27 [2] Bioconductor  
##  zoo                    1.8-8    2020-05-02 [2] CRAN (R 4.0.2)
## 
## [1] /tmp/Rtmp8eL3Yz/temp_libpath1a12c49e51fb9
## [2] /home/dbdimitrov/R/x86_64-pc-linux-gnu-library/4.0
## [3] /usr/local/lib/R/site-library
## [4] /usr/lib/R/site-library
## [5] /usr/lib/R/library</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Dimitrov, Charlotte Boys, Martin Garrido Rodríguez, Denes Turei, James Nagai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
